{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "aeb5b25e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dickeya_zeae_ec1                                            \tDickeya_zeae_ec1                                            \t          \t                              \thttps://snpeff.blob.core.windows.net/databases/v5_0/snpEff_v5_0_Dickeya_zeae_ec1.zip\n",
      "Dickeya_zeae_ech586                                         \tDickeya_zeae_ech586                                         \t          \t                              \thttps://snpeff.blob.core.windows.net/databases/v5_0/snpEff_v5_0_Dickeya_zeae_ech586.zip\n",
      "Lactobacillus_zeae_dsm_20178_kctc_3804                      \tLactobacillus_zeae_dsm_20178_kctc_3804                      \t          \t                              \thttps://snpeff.blob.core.windows.net/databases/v5_0/snpEff_v5_0_Lactobacillus_zeae_dsm_20178_kctc_3804.zip\n",
      "Lentzea_guizhouensis                                        \tLentzea_guizhouensis                                        \t          \t                              \thttps://snpeff.blob.core.windows.net/databases/v5_0/snpEff_v5_0_Lentzea_guizhouensis.zip\n",
      "Siansivirga_zeaxanthinifaciens_cc_samt_1                    \tSiansivirga_zeaxanthinifaciens_cc_samt_1                    \t          \t                              \thttps://snpeff.blob.core.windows.net/databases/v5_0/snpEff_v5_0_Siansivirga_zeaxanthinifaciens_cc_samt_1.zip\n",
      "Zea_mays                                                    \tZea_mays                                                    \t          \t                              \thttps://snpeff.blob.core.windows.net/databases/v5_0/snpEff_v5_0_Zea_mays.zip\n",
      "Zea_mays_mo17_CAU.1.0                                       \tZea_mays_mo17_CAU.1.0                                       \tOK        \t                              \thttps://snpeff.blob.core.windows.net/databases/v5_0/snpEff_v5_0_Zea_mays_mo17_CAU.1.0.zip\n"
     ]
    }
   ],
   "source": [
    "#To run Snpeff requires a database of annotations. Need to first find the database for the NAM5 genome\n",
    "java -jar /Users/aricha14/Documents/Tools/snpEff/snpEff.jar databases | grep -i zea\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "570c05c4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?1049h\u001b[>4;2m\u001b[?1h\u001b=\u001b[?2004h\u001b[1;24r\u001b[?12h\u001b[?12l\u001b[22;2t\u001b[22;1t\u001b[27m\u001b[29m\u001b[m\u001b[H\u001b[2J\u001b[?25l\u001b[24;1H\"~/Documents/Tools/snpEff/snpEff.config\" 183665L, 10773295B\u001b[2;1H▽\u001b[6n\u001b[2;1H  \u001b[3;1H\u001bPzz\u001b\\\u001b[0%m\u001b[6n\u001b[3;1H           \u001b[1;1H\u001b[>c\u001b]10;?\u0007\u001b]11;?\u0007\u001b[2;1H#-------------------------------------------------------------------------------\u001b[3;1H#\u001b[3;2H\u001b[K\u001b[4;1H# SnpEff configuration file\n",
      "#\n",
      "#\u001b[78C  \u001b[7;1H \u001b[47CPablo Cingolani\n",
      "#-------------------------------------------------------------------------------\u001b[10;1H#---\n",
      "# Databases are stored here\n",
      "# E.g.: Information for 'hg19' is stored in data.dir/hg19/\n",
      "#\n",
      "# You can use tilde ('~') as first character to refer to your home directory.\n",
      "# Also, a non-absolute path will be relative to config's file dir\n",
      "#\n",
      "#---\n",
      "data.dir = ./data/\n",
      "\n",
      "#---\n",
      "# Database repository: A URL to the server where you can download databases (comm\u001b[22;1Hmand: 'snpEff download dbName')\n"
     ]
    }
   ],
   "source": [
    "#Ok so the NAM5 genome is not available, I therefore need to format a new database for snpeff \n",
    "# First in order to tell SnpEff that there is a new genome available, you must update SnpEff's configuration file snpEff.config.\n",
    "\n",
    "\n",
    "vi /Users/aricha14/Documents/Tools/snpEff/snpEff.config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "edbc9fb0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Zea_mays_mo17_CAU.1.0\n"
     ]
    }
   ],
   "source": [
    "cd /Users/aricha14/Documents/Tools/snpEff/data\n",
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d54b1da6",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir Zea_mays_B73_NAM.5.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "13effc5c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-06-21 12:46:29--  https://download.maizegdb.org/Zm-B73-REFERENCE-NAM-5.0/Zm-B73-REFERENCE-NAM-5.0_Zm00001eb.1.gff3.gz\n",
      "Resolving download.maizegdb.org (download.maizegdb.org)... 104.26.11.112, 104.26.10.112, 172.67.74.80\n",
      "Connecting to download.maizegdb.org (download.maizegdb.org)|104.26.11.112|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 12251264 (12M) [application/x-gzip]\n",
      "Saving to: ‘Zm-B73-REFERENCE-NAM-5.0_Zm00001eb.1.gff3.gz’\n",
      "\n",
      "Zm-B73-REFERENCE-NA 100%[===================>]  11.68M  12.3MB/s    in 0.9s    \n",
      "\n",
      "2022-06-21 12:46:30 (12.3 MB/s) - ‘Zm-B73-REFERENCE-NAM-5.0_Zm00001eb.1.gff3.gz’ saved [12251264/12251264]\n",
      "\n",
      "bash: cd: /Users/aricha14/Documents/Tools/snpEff/data/genomes: No such file or directory\n",
      "--2022-06-21 12:46:31--  https://download.maizegdb.org/Zm-B73-REFERENCE-NAM-5.0/Zm-B73-REFERENCE-NAM-5.0.fa.gz\n",
      "Resolving download.maizegdb.org (download.maizegdb.org)... 104.26.10.112, 172.67.74.80, 104.26.11.112\n",
      "Connecting to download.maizegdb.org (download.maizegdb.org)|104.26.10.112|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 633000143 (604M) [application/x-gzip]\n",
      "Saving to: ‘Zm-B73-REFERENCE-NAM-5.0.fa.gz’\n",
      "\n",
      "Zm-B73-REFERENCE-NA 100%[===================>] 603.68M  28.5MB/s    in 26s     \n",
      "\n",
      "2022-06-21 12:46:57 (22.9 MB/s) - ‘Zm-B73-REFERENCE-NAM-5.0.fa.gz’ saved [633000143/633000143]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Get annotation files and put into the snpeff data area\n",
    "cd Zea_mays_B73_NAM.5.0\n",
    "wget https://download.maizegdb.org/Zm-B73-REFERENCE-NAM-5.0/Zm-B73-REFERENCE-NAM-5.0_Zm00001eb.1.gff3.gz\n",
    "mv Zm-B73-REFERENCE-NAM-5.0_Zm00001eb.1.gff3.gz genes.gff.gz\n",
    "\n",
    "# Get the genome reference sequence file\n",
    "cd /Users/aricha14/Documents/Tools/snpEff/data/genomes\n",
    "wget https://download.maizegdb.org/Zm-B73-REFERENCE-NAM-5.0/Zm-B73-REFERENCE-NAM-5.0.fa.gz\n",
    "mv Zm-B73-REFERENCE-NAM-5.0.fa.gz Zea_mays_B73_NAM.5.0.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "69ddacf5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Need to rename the fasta file to sequences.fa\n",
    "cd /Users/aricha14/Documents/Tools/snpEff/data/Zea_mays_B73_NAM.5.0\n",
    "mv Zea_mays_B73_NAM.5.0.fa.gz sequences.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "03e07970",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00:00:00\tSnpEff version SnpEff 5.0c (build 2020-11-25 14:23), by Pablo Cingolani\n",
      "00:00:00\tCommand: 'build'\n",
      "00:00:00\tBuilding database for 'Zea_mays_B73_NAM.5.0'\n",
      "00:00:00\tReading configuration file 'snpEff.config'. Genome: 'Zea_mays_B73_NAM.5.0'\n",
      "00:00:00\tReading config file: /Users/aricha14/Documents/Tools/snpEff/snpEff.config\n",
      "00:00:00\tdone\n",
      "Reading GFF3 data file  : '/Users/aricha14/Documents/Tools/snpEff/./data/Zea_mays_B73_NAM.5.0/genes.gff'\n",
      "....................................................................................................\n",
      "\t\t10000\t....................................................................................................\n",
      "\t\t20000\t....................................................................................................\n",
      "\t\t30000\t....................................................................................................\n",
      "\t\t40000\t....................................................................................................\n",
      "\t\t50000\t....................................................................................................\n",
      "\t\t60000\t....................................................................................................\n",
      "\t\t70000\t....................................................................................................\n",
      "\t\t80000\t....................................................................................................\n",
      "\t\t90000\t....................................................................................................\n",
      "\t\t100000\t....................................................................................................\n",
      "\t\t110000\t....................................................................................................\n",
      "\t\t120000\t....................................................................................................\n",
      "\t\t130000\t....................................................................................................\n",
      "\t\t140000\t....................................................................................................\n",
      "\t\t150000\t....................................................................................................\n",
      "\t\t160000\t....................................................................................................\n",
      "\t\t170000\t....................................................................................................\n",
      "\t\t180000\t....................................................................................................\n",
      "\t\t190000\t....................................................................................................\n",
      "\t\t200000\t....................................................................................................\n",
      "\t\t210000\t....................................................................................................\n",
      "\t\t220000\t....................................................................................................\n",
      "\t\t230000\t....................................................................................................\n",
      "\t\t240000\t....................................................................................................\n",
      "\t\t250000\t....................................................................................................\n",
      "\t\t260000\t....................................................................................................\n",
      "\t\t270000\t....................................................................................................\n",
      "\t\t280000\t....................................................................................................\n",
      "\t\t290000\t....................................................................................................\n",
      "\t\t300000\t....................................................................................................\n",
      "\t\t310000\t....................................................................................................\n",
      "\t\t320000\t....................................................................................................\n",
      "\t\t330000\t....................................................................................................\n",
      "\t\t340000\t....................................................................................................\n",
      "\t\t350000\t....................................................................................................\n",
      "\t\t360000\t....................................................................................................\n",
      "\t\t370000\t....................................................................................................\n",
      "\t\t380000\t....................................................................................................\n",
      "\t\t390000\t....................................................................................................\n",
      "\t\t400000\t....................................................................................................\n",
      "\t\t410000\t....................................................................................................\n",
      "\t\t420000\t....................................................................................................\n",
      "\t\t430000\t....................................................................................................\n",
      "\t\t440000\t....................................................................................................\n",
      "\t\t450000\t....................................................................................................\n",
      "\t\t460000\t....................................................................................................\n",
      "\t\t470000\t....................................................................................................\n",
      "\t\t480000\t....................................................................................................\n",
      "\t\t490000\t....................................................................................................\n",
      "\t\t500000\t....................................................................................................\n",
      "\t\t510000\t....................................................................................................\n",
      "\t\t520000\t....................................................................................................\n",
      "\t\t530000\t....................................................................................................\n",
      "\t\t540000\t....................................................................................................\n",
      "\t\t550000\t....................................................................................................\n",
      "\t\t560000\t....................................................................................................\n",
      "\t\t570000\t....................................................................................................\n",
      "\t\t580000\t....................................................................................................\n",
      "\t\t590000\t....................................................................................................\n",
      "\t\t600000\t....................................................................................................\n",
      "\t\t610000\t....................................................................................................\n",
      "\t\t620000\t....................................................................................................\n",
      "\t\t630000\t....................................................................................................\n",
      "\t\t640000\t....................................................................................................\n",
      "\t\t650000\t....................................................................................................\n",
      "\t\t660000\t....................................................................................................\n",
      "\t\t670000\t....................................................................................................\n",
      "\t\t680000\t....................................................................................................\n",
      "\t\t690000\t....................................................................................................\n",
      "\t\t700000\t....................................................................................................\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t\t710000\t....................................................................................................\n",
      "\t\t720000\t....................................................................................................\n",
      "\t\t730000\t....................................................................................................\n",
      "\t\t740000\t..\n",
      "\tTotal: 740246 markers added.\n",
      "\n",
      "\tCreate exons from CDS (if needed): \n",
      "\tExons created for 0 transcripts.\n",
      "\n",
      "\tDeleting redundant exons (if needed): .\n",
      "\t\t0\t....................................................................................................\n",
      "\t\t10000\t....................................................................................................\n",
      "\t\t20000\t....................................................................................................\n",
      "\t\t30000\t....................................................................................................\n",
      "\t\t40000\t....................................................................................................\n",
      "\t\t50000\t....................................................................................................\n",
      "\t\t60000\t.............................................\n",
      "\t\tTotal transcripts with deleted exons: 64561\n",
      "\n",
      "\tCollapsing zero length introns (if needed): \n",
      "\t\tTotal collapsed transcripts: 0\n",
      "\tReading sequences   :\n",
      "\tFASTA file: '/Users/aricha14/Documents/Tools/snpEff/./data/genomes/Zea_mays_B73_NAM.5.0.fa' not found.\n",
      "\tReading FASTA file: '/Users/aricha14/Documents/Tools/snpEff/./data/Zea_mays_B73_NAM.5.0/sequences.fa'\n",
      "\t\tReading sequence '1', length: 308452471\n",
      "\t\tAdding genomic sequences to exons: \tDone (68813 sequences added, 0 ignored).\n",
      "\t\tReading sequence '2', length: 243675191\n",
      "\t\tAdding genomic sequences to exons: \tDone (52140 sequences added, 0 ignored).\n",
      "\t\tReading sequence '3', length: 238017767\n",
      "\t\tAdding genomic sequences to exons: \tDone (50871 sequences added, 0 ignored).\n",
      "\t\tReading sequence '4', length: 250330460\n",
      "\t\tAdding genomic sequences to exons: \tDone (47163 sequences added, 0 ignored).\n",
      "\t\tReading sequence '5', length: 226353449\n",
      "\t\tAdding genomic sequences to exons: \tDone (52326 sequences added, 0 ignored).\n",
      "\t\tReading sequence '6', length: 181357234\n",
      "\t\tAdding genomic sequences to exons: \tDone (38295 sequences added, 0 ignored).\n",
      "\t\tReading sequence '7', length: 185808916\n",
      "\t\tAdding genomic sequences to exons: \tDone (34486 sequences added, 0 ignored).\n",
      "\t\tReading sequence '8', length: 182411202\n",
      "\t\tAdding genomic sequences to exons: \tDone (41119 sequences added, 0 ignored).\n",
      "\t\tReading sequence '9', length: 163004744\n",
      "\t\tAdding genomic sequences to exons: \tDone (34839 sequences added, 0 ignored).\n",
      "\t\tReading sequence '10', length: 152435371\n",
      "\t\tAdding genomic sequences to exons: \tDone (30390 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_21', length: 712123\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_22', length: 570058\n",
      "\t\tAdding genomic sequences to exons: \tDone (32 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_23', length: 435198\n",
      "\t\tAdding genomic sequences to exons: \tDone (12 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_24', length: 387292\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_25', length: 264983\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_26', length: 254485\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_27', length: 247825\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_28', length: 239868\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_29', length: 217815\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_30', length: 207305\n",
      "\t\tAdding genomic sequences to exons: \tDone (8 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_31', length: 203238\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_32', length: 197712\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_33', length: 196160\n",
      "\t\tAdding genomic sequences to exons: \tDone (29 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_34', length: 190448\n",
      "\t\tAdding genomic sequences to exons: \tDone (36 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_35', length: 190269\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_36', length: 187880\n",
      "\t\tAdding genomic sequences to exons: \tDone (3 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_37', length: 173911\n",
      "\t\tAdding genomic sequences to exons: \tDone (28 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_38', length: 173111\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_39', length: 172167\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_40', length: 171701\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_41', length: 171519\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_42', length: 169347\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_43', length: 168052\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_44', length: 164231\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_45', length: 164060\n",
      "\t\tAdding genomic sequences to exons: \tDone (21 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_46', length: 160926\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_47', length: 160502\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_48', length: 159682\n",
      "\t\tAdding genomic sequences to exons: \tDone (29 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_49', length: 153763\n",
      "\t\tAdding genomic sequences to exons: \tDone (5 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_50', length: 153744\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_51', length: 152095\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_52', length: 151820\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_53', length: 148690\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_54', length: 144791\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_55', length: 144328\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_56', length: 142977\n",
      "\t\tAdding genomic sequences to exons: \tDone (10 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_57', length: 142290\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_58', length: 141577\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_59', length: 138583\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_60', length: 137147\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_61', length: 134092\n",
      "\t\tAdding genomic sequences to exons: \tDone (6 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_62', length: 133633\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_63', length: 132531\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_64', length: 130230\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_65', length: 129299\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_66', length: 128904\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t\tAdding genomic sequences to exons: \tDone (3 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_67', length: 128482\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_68', length: 127882\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_69', length: 127290\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_70', length: 126443\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_71', length: 125047\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_72', length: 124860\n",
      "\t\tAdding genomic sequences to exons: \tDone (7 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_73', length: 123043\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_74', length: 120982\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_75', length: 120904\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_76', length: 120818\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_77', length: 120549\n",
      "\t\tAdding genomic sequences to exons: \tDone (18 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_78', length: 120088\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_79', length: 119143\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_80', length: 118700\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_81', length: 118576\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_82', length: 118500\n",
      "\t\tAdding genomic sequences to exons: \tDone (26 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_83', length: 117266\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_84', length: 117233\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_85', length: 115765\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_86', length: 115550\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_87', length: 114887\n",
      "\t\tAdding genomic sequences to exons: \tDone (14 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_88', length: 114023\n",
      "\t\tAdding genomic sequences to exons: \tDone (21 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_89', length: 113862\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_90', length: 113779\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_91', length: 113002\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_92', length: 112596\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_93', length: 112550\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_94', length: 111637\n",
      "\t\tAdding genomic sequences to exons: \tDone (3 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_95', length: 110562\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_96', length: 110121\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_97', length: 109951\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_98', length: 109947\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_99', length: 109352\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_100', length: 109197\n",
      "\t\tAdding genomic sequences to exons: \tDone (8 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_101', length: 109031\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_102', length: 108697\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_103', length: 108378\n",
      "\t\tAdding genomic sequences to exons: \tDone (18 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_104', length: 108339\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_105', length: 107956\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_106', length: 107526\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_107', length: 107524\n",
      "\t\tAdding genomic sequences to exons: \tDone (25 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_108', length: 106690\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_109', length: 106305\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_110', length: 106098\n",
      "\t\tAdding genomic sequences to exons: \tDone (4 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_111', length: 105289\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_112', length: 105250\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_113', length: 105226\n",
      "\t\tAdding genomic sequences to exons: \tDone (4 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_114', length: 105163\n",
      "\t\tAdding genomic sequences to exons: \tDone (24 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_115', length: 104537\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_116', length: 104499\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_117', length: 103808\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_118', length: 103806\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_119', length: 103210\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_120', length: 102624\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_121', length: 102406\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_122', length: 102397\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_123', length: 102349\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_124', length: 101678\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_125', length: 101641\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_126', length: 100851\n",
      "\t\tAdding genomic sequences to exons: \tDone (16 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_127', length: 100600\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_128', length: 100537\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_129', length: 100364\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_130', length: 99893\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_131', length: 99674\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_132', length: 98130\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_133', length: 97996\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_134', length: 97918\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_135', length: 96911\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_136', length: 96690\n",
      "\t\tAdding genomic sequences to exons: \tDone (6 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_137', length: 96496\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_138', length: 96036\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_139', length: 95934\n",
      "\t\tAdding genomic sequences to exons: \tDone (16 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_140', length: 95894\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_141', length: 95890\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_142', length: 95521\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_143', length: 95419\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_144', length: 95245\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_145', length: 95129\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_146', length: 94989\n",
      "\t\tAdding genomic sequences to exons: \tDone (3 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_147', length: 94793\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_148', length: 94647\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_149', length: 94334\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_150', length: 94333\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_151', length: 94036\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_152', length: 93636\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_153', length: 93514\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_154', length: 93405\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_155', length: 93298\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_156', length: 93084\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_157', length: 93056\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_158', length: 92553\n",
      "\t\tAdding genomic sequences to exons: \tDone (8 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_159', length: 91995\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_160', length: 91958\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_161', length: 91897\n",
      "\t\tAdding genomic sequences to exons: \tDone (8 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_162', length: 91766\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_163', length: 91439\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_164', length: 91076\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_165', length: 91072\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_166', length: 91034\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_167', length: 90901\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_168', length: 90815\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_169', length: 90613\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_170', length: 90547\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_171', length: 89502\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_172', length: 89377\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_173', length: 89242\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_174', length: 89235\n",
      "\t\tAdding genomic sequences to exons: \tDone (10 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_175', length: 89155\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_176', length: 88783\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_177', length: 88032\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_178', length: 87897\n",
      "\t\tAdding genomic sequences to exons: \tDone (9 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_179', length: 87788\n",
      "\t\tAdding genomic sequences to exons: \tDone (17 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_180', length: 87559\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_181', length: 87401\n",
      "\t\tAdding genomic sequences to exons: \tDone (20 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_182', length: 87195\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_183', length: 86681\n",
      "\t\tAdding genomic sequences to exons: \tDone (6 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_184', length: 86649\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_185', length: 86437\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_186', length: 86423\n",
      "\t\tAdding genomic sequences to exons: \tDone (4 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_187', length: 86029\n",
      "\t\tAdding genomic sequences to exons: \tDone (6 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_188', length: 85958\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_189', length: 85939\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_190', length: 85707\n",
      "\t\tAdding genomic sequences to exons: \tDone (12 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_191', length: 85696\n",
      "\t\tAdding genomic sequences to exons: \tDone (8 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_192', length: 85577\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_193', length: 85542\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_194', length: 85512\n",
      "\t\tAdding genomic sequences to exons: \tDone (14 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_195', length: 84529\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_196', length: 84483\n",
      "\t\tAdding genomic sequences to exons: \tDone (25 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_197', length: 84480\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_198', length: 84315\n",
      "\t\tAdding genomic sequences to exons: \tDone (3 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_199', length: 84283\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_200', length: 84277\n",
      "\t\tAdding genomic sequences to exons: \tDone (12 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_201', length: 83765\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_202', length: 83195\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t\tReading sequence 'scaf_203', length: 83150\n",
      "\t\tAdding genomic sequences to exons: \tDone (3 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_204', length: 83142\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_205', length: 83098\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_206', length: 82836\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_207', length: 82698\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_208', length: 82580\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_209', length: 82439\n",
      "\t\tAdding genomic sequences to exons: \tDone (12 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_210', length: 82354\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_211', length: 82347\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_212', length: 82331\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_213', length: 81960\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_214', length: 81777\n",
      "\t\tAdding genomic sequences to exons: \tDone (21 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_215', length: 81747\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_216', length: 81457\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_217', length: 81340\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_218', length: 81233\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_219', length: 80936\n",
      "\t\tAdding genomic sequences to exons: \tDone (19 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_220', length: 80837\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_221', length: 80814\n",
      "\t\tAdding genomic sequences to exons: \tDone (25 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_222', length: 80283\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_223', length: 79710\n",
      "\t\tAdding genomic sequences to exons: \tDone (14 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_224', length: 79539\n",
      "\t\tAdding genomic sequences to exons: \tDone (16 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_225', length: 79526\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_226', length: 79449\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_227', length: 79237\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_228', length: 78875\n",
      "\t\tAdding genomic sequences to exons: \tDone (17 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_229', length: 78242\n",
      "\t\tAdding genomic sequences to exons: \tDone (14 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_230', length: 78216\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_231', length: 78109\n",
      "\t\tAdding genomic sequences to exons: \tDone (47 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_232', length: 77884\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_233', length: 77858\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_234', length: 77850\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_235', length: 77518\n",
      "\t\tAdding genomic sequences to exons: \tDone (4 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_236', length: 77375\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_237', length: 77233\n",
      "\t\tAdding genomic sequences to exons: \tDone (15 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_238', length: 77062\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_239', length: 77005\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_240', length: 76990\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_241', length: 76896\n",
      "\t\tAdding genomic sequences to exons: \tDone (10 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_242', length: 76850\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_243', length: 76719\n",
      "\t\tAdding genomic sequences to exons: \tDone (13 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_244', length: 76711\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_245', length: 76434\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_246', length: 76402\n",
      "\t\tAdding genomic sequences to exons: \tDone (14 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_247', length: 76349\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_248', length: 76347\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_249', length: 76253\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_250', length: 76174\n",
      "\t\tAdding genomic sequences to exons: \tDone (4 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_251', length: 76113\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_252', length: 76030\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_253', length: 75901\n",
      "\t\tAdding genomic sequences to exons: \tDone (4 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_254', length: 75487\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_255', length: 75327\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_256', length: 75278\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_257', length: 75264\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_258', length: 75157\n",
      "\t\tAdding genomic sequences to exons: \tDone (18 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_259', length: 74994\n",
      "\t\tAdding genomic sequences to exons: \tDone (10 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_260', length: 74954\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_261', length: 74875\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_262', length: 74862\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_263', length: 74687\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_264', length: 74566\n",
      "\t\tAdding genomic sequences to exons: \tDone (6 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_265', length: 74453\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_266', length: 73951\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_267', length: 73767\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_268', length: 73692\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_269', length: 73531\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_270', length: 73359\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_271', length: 73251\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_272', length: 73066\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_273', length: 73001\n",
      "\t\tAdding genomic sequences to exons: \tDone (10 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_274', length: 72796\n",
      "\t\tAdding genomic sequences to exons: \tDone (9 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_275', length: 72665\n",
      "\t\tAdding genomic sequences to exons: \tDone (12 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_276', length: 72399\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_277', length: 72053\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_278', length: 71983\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_279', length: 71870\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_280', length: 71434\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_281', length: 71227\n",
      "\t\tAdding genomic sequences to exons: \tDone (6 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_282', length: 71196\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_283', length: 71104\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_284', length: 70953\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_285', length: 70845\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_286', length: 70600\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_287', length: 70547\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_288', length: 70410\n",
      "\t\tAdding genomic sequences to exons: \tDone (43 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_289', length: 70226\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_290', length: 70070\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_291', length: 69844\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_292', length: 69796\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_293', length: 69747\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_294', length: 69542\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_295', length: 69529\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_296', length: 69442\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_297', length: 69369\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_298', length: 69364\n",
      "\t\tAdding genomic sequences to exons: \tDone (14 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_299', length: 69243\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_300', length: 69101\n",
      "\t\tAdding genomic sequences to exons: \tDone (9 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_301', length: 69023\n",
      "\t\tAdding genomic sequences to exons: \tDone (3 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_302', length: 68967\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_303', length: 68912\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_304', length: 68810\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_305', length: 68803\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_306', length: 68764\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_307', length: 68750\n",
      "\t\tAdding genomic sequences to exons: \tDone (42 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_308', length: 68697\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_309', length: 68464\n",
      "\t\tAdding genomic sequences to exons: \tDone (4 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_310', length: 68461\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_311', length: 68394\n",
      "\t\tAdding genomic sequences to exons: \tDone (4 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_312', length: 68386\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_313', length: 68340\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_314', length: 68200\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_315', length: 68185\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_316', length: 68157\n",
      "\t\tAdding genomic sequences to exons: \tDone (19 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_317', length: 68122\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_318', length: 67773\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_319', length: 67715\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_320', length: 67453\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_321', length: 67423\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_322', length: 67253\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_323', length: 67217\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_324', length: 67121\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_325', length: 67094\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_326', length: 66933\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_327', length: 66933\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_328', length: 66728\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_329', length: 66653\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_330', length: 66505\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_331', length: 66412\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_332', length: 66353\n",
      "\t\tAdding genomic sequences to exons: \tDone (9 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_333', length: 66282\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_334', length: 66272\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_335', length: 66081\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_336', length: 66016\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_337', length: 65790\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_338', length: 65540\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_339', length: 65537\n",
      "\t\tAdding genomic sequences to exons: \tDone (5 sequences added, 0 ignored).\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t\tReading sequence 'scaf_340', length: 65427\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_341', length: 65327\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_342', length: 65280\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_343', length: 65236\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_344', length: 65094\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_345', length: 64929\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_346', length: 64906\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_347', length: 64744\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_348', length: 64714\n",
      "\t\tAdding genomic sequences to exons: \tDone (14 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_349', length: 64685\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_350', length: 64593\n",
      "\t\tAdding genomic sequences to exons: \tDone (6 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_351', length: 64495\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_352', length: 64265\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_353', length: 64155\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_354', length: 64072\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_355', length: 64028\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_356', length: 63985\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_357', length: 63908\n",
      "\t\tAdding genomic sequences to exons: \tDone (12 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_358', length: 63880\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_359', length: 63432\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_360', length: 63266\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_361', length: 63177\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_362', length: 63148\n",
      "\t\tAdding genomic sequences to exons: \tDone (10 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_363', length: 63113\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_364', length: 63084\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_365', length: 63080\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_366', length: 62556\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_367', length: 62541\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_368', length: 62533\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_369', length: 62516\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_370', length: 62509\n",
      "\t\tAdding genomic sequences to exons: \tDone (6 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_371', length: 62284\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_372', length: 62204\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_373', length: 62066\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_374', length: 62047\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_375', length: 61971\n",
      "\t\tAdding genomic sequences to exons: \tDone (6 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_376', length: 61863\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_377', length: 61815\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_378', length: 61663\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_379', length: 61251\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_380', length: 61109\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_381', length: 61002\n",
      "\t\tAdding genomic sequences to exons: \tDone (7 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_382', length: 60996\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_383', length: 60990\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_384', length: 60988\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_385', length: 60794\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_386', length: 60618\n",
      "\t\tAdding genomic sequences to exons: \tDone (8 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_387', length: 60414\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_388', length: 60261\n",
      "\t\tAdding genomic sequences to exons: \tDone (35 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_389', length: 60203\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_390', length: 60132\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_391', length: 60044\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_392', length: 59970\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_393', length: 59921\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_394', length: 59838\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_395', length: 59635\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_396', length: 59584\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_397', length: 59572\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_398', length: 59286\n",
      "\t\tAdding genomic sequences to exons: \tDone (12 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_399', length: 59208\n",
      "\t\tAdding genomic sequences to exons: \tDone (3 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_400', length: 59116\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_401', length: 59113\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_402', length: 58978\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_403', length: 58950\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_404', length: 58762\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_405', length: 58605\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_406', length: 58599\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_407', length: 58456\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_408', length: 58405\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_409', length: 58344\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_410', length: 57998\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_411', length: 57932\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_412', length: 57849\n",
      "\t\tAdding genomic sequences to exons: \tDone (12 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_413', length: 57812\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_414', length: 57501\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_415', length: 57472\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_416', length: 57371\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_417', length: 57210\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_418', length: 57042\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_419', length: 57011\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_420', length: 56837\n",
      "\t\tAdding genomic sequences to exons: \tDone (23 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_421', length: 56819\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_422', length: 56707\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_423', length: 56705\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_424', length: 56693\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_425', length: 56600\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_426', length: 56555\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_427', length: 56516\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_428', length: 56435\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_429', length: 56416\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_430', length: 56249\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_431', length: 56149\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_432', length: 56140\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_433', length: 55942\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_434', length: 55813\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_435', length: 55746\n",
      "\t\tAdding genomic sequences to exons: \tDone (7 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_436', length: 55605\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_437', length: 55516\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_438', length: 55489\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_439', length: 55303\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_440', length: 55289\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_441', length: 55099\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_442', length: 55048\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_443', length: 54923\n",
      "\t\tAdding genomic sequences to exons: \tDone (4 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_444', length: 54898\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_445', length: 54817\n",
      "\t\tAdding genomic sequences to exons: \tDone (6 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_446', length: 54746\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_447', length: 54638\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_448', length: 54628\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_449', length: 54545\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_450', length: 54531\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_451', length: 54430\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_452', length: 54309\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_453', length: 54288\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_454', length: 54132\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_455', length: 54093\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_456', length: 53967\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_457', length: 53894\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_458', length: 53843\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_459', length: 53481\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_460', length: 53434\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_461', length: 53417\n",
      "\t\tAdding genomic sequences to exons: \tDone (10 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_462', length: 53195\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_463', length: 53086\n",
      "\t\tAdding genomic sequences to exons: \tDone (8 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_464', length: 52993\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_465', length: 52955\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_466', length: 52889\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_467', length: 52720\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_468', length: 52562\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_469', length: 52527\n",
      "\t\tAdding genomic sequences to exons: \tDone (6 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_470', length: 52467\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_471', length: 52001\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_472', length: 51792\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_473', length: 51791\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_474', length: 51790\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_475', length: 51498\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_476', length: 51442\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t\tReading sequence 'scaf_477', length: 51437\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_478', length: 51436\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_479', length: 51400\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_480', length: 51317\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_481', length: 51285\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_482', length: 51226\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_483', length: 51166\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_484', length: 51159\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_485', length: 51097\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_486', length: 51050\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_487', length: 51005\n",
      "\t\tAdding genomic sequences to exons: \tDone (4 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_488', length: 50467\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_489', length: 50303\n",
      "\t\tAdding genomic sequences to exons: \tDone (6 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_490', length: 50230\n",
      "\t\tAdding genomic sequences to exons: \tDone (8 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_491', length: 50119\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_492', length: 50112\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_493', length: 50095\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_494', length: 50090\n",
      "\t\tAdding genomic sequences to exons: \tDone (3 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_495', length: 49999\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_496', length: 49900\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_497', length: 49811\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_498', length: 49759\n",
      "\t\tAdding genomic sequences to exons: \tDone (23 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_499', length: 49573\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_500', length: 49572\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_501', length: 49558\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_502', length: 49534\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_503', length: 49321\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_504', length: 49133\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_505', length: 49076\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_506', length: 48960\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_507', length: 48845\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_508', length: 48796\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_509', length: 48793\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_510', length: 48788\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_511', length: 48698\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_512', length: 48411\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_513', length: 48321\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_514', length: 48292\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_515', length: 48059\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_516', length: 48018\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_517', length: 47885\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_518', length: 47771\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_519', length: 47732\n",
      "\t\tAdding genomic sequences to exons: \tDone (15 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_520', length: 47726\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_521', length: 47702\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_522', length: 47686\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_523', length: 47658\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_524', length: 47519\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_525', length: 47504\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_526', length: 47428\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_527', length: 47425\n",
      "\t\tAdding genomic sequences to exons: \tDone (28 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_528', length: 47422\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_529', length: 47062\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_530', length: 47056\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_531', length: 47032\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_532', length: 46775\n",
      "\t\tAdding genomic sequences to exons: \tDone (26 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_533', length: 46694\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_534', length: 46575\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_535', length: 46567\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_536', length: 46533\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_537', length: 46395\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_538', length: 46312\n",
      "\t\tAdding genomic sequences to exons: \tDone (4 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_539', length: 46276\n",
      "\t\tAdding genomic sequences to exons: \tDone (23 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_540', length: 46195\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_541', length: 46105\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_542', length: 45928\n",
      "\t\tAdding genomic sequences to exons: \tDone (13 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_543', length: 45898\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_544', length: 45870\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_545', length: 45852\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t\tAdding genomic sequences to exons: \tDone (13 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_546', length: 45599\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_547', length: 45570\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_548', length: 45567\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_549', length: 45460\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_550', length: 45408\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_551', length: 45391\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_552', length: 45290\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_553', length: 45286\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_554', length: 45262\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_555', length: 45165\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_556', length: 44942\n",
      "\t\tAdding genomic sequences to exons: \tDone (9 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_557', length: 44777\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_558', length: 44734\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_559', length: 44719\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_560', length: 44617\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_561', length: 44557\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_562', length: 44525\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_563', length: 44435\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_564', length: 44391\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_565', length: 44377\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_566', length: 44336\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_567', length: 44149\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_568', length: 43798\n",
      "\t\tAdding genomic sequences to exons: \tDone (5 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_569', length: 43689\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_570', length: 43564\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_571', length: 43505\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_572', length: 43361\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_573', length: 43335\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_574', length: 43270\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_575', length: 43186\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_576', length: 43039\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_577', length: 42914\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_578', length: 42890\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_579', length: 42824\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_580', length: 42746\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_581', length: 42734\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_582', length: 42695\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_583', length: 42670\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_584', length: 42655\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_585', length: 42653\n",
      "\t\tAdding genomic sequences to exons: \tDone (8 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_586', length: 42361\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_587', length: 42140\n",
      "\t\tAdding genomic sequences to exons: \tDone (3 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_588', length: 42065\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_589', length: 42059\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_590', length: 42054\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_591', length: 42047\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_592', length: 42042\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_593', length: 41905\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_594', length: 41863\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_595', length: 41770\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_596', length: 41630\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_597', length: 41600\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_598', length: 41539\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_599', length: 41383\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_600', length: 41377\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_601', length: 41329\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_602', length: 41219\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_603', length: 41109\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_604', length: 41107\n",
      "\t\tAdding genomic sequences to exons: \tDone (8 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_605', length: 41057\n",
      "\t\tAdding genomic sequences to exons: \tDone (5 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_606', length: 41030\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_607', length: 40971\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_608', length: 40905\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_609', length: 40876\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_610', length: 40847\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_611', length: 40814\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_612', length: 40576\n",
      "\t\tAdding genomic sequences to exons: \tDone (4 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_613', length: 40560\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t\tReading sequence 'scaf_614', length: 40479\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_615', length: 40402\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_616', length: 40389\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_617', length: 40165\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_618', length: 40145\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_619', length: 40064\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_620', length: 40006\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_621', length: 39904\n",
      "\t\tAdding genomic sequences to exons: \tDone (4 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_622', length: 39697\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_623', length: 39553\n",
      "\t\tAdding genomic sequences to exons: \tDone (4 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_624', length: 39481\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_625', length: 39460\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_626', length: 39077\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_627', length: 39038\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_628', length: 38891\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_629', length: 38764\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_630', length: 38610\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_631', length: 38558\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_632', length: 38463\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_633', length: 38456\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_634', length: 38214\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_635', length: 37995\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_636', length: 37985\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_637', length: 37569\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_638', length: 37477\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_639', length: 37416\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_640', length: 37393\n",
      "\t\tAdding genomic sequences to exons: \tDone (4 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_641', length: 37334\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_642', length: 37315\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_643', length: 37184\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_644', length: 37110\n",
      "\t\tAdding genomic sequences to exons: \tDone (19 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_645', length: 37080\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_646', length: 36930\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_647', length: 36782\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_648', length: 36641\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_649', length: 36626\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_650', length: 36526\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_651', length: 36525\n",
      "\t\tAdding genomic sequences to exons: \tDone (31 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_652', length: 36369\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_653', length: 36282\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_654', length: 36267\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_655', length: 36266\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_656', length: 36217\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_657', length: 35909\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_658', length: 35909\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_659', length: 35884\n",
      "\t\tAdding genomic sequences to exons: \tDone (4 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_660', length: 35784\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_661', length: 35720\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_662', length: 35668\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_663', length: 35599\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_664', length: 35539\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_665', length: 35297\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_666', length: 35285\n",
      "\t\tAdding genomic sequences to exons: \tDone (6 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_667', length: 35263\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_668', length: 35100\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_669', length: 34989\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_670', length: 34940\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_671', length: 34852\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_672', length: 34684\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_673', length: 34626\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_674', length: 34568\n",
      "\t\tAdding genomic sequences to exons: \tDone (9 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_675', length: 34427\n",
      "\t\tAdding genomic sequences to exons: \tDone (2 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_676', length: 34391\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_677', length: 33997\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_678', length: 33938\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_679', length: 33821\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_680', length: 33689\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_681', length: 33683\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_682', length: 33634\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_683', length: 33496\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_684', length: 33264\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_685', length: 32618\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_686', length: 32197\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_687', length: 32019\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_688', length: 31944\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_689', length: 31748\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_690', length: 31498\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_691', length: 31489\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_692', length: 31211\n",
      "\t\tAdding genomic sequences to exons: \tDone (4 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_693', length: 30818\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_694', length: 30512\n",
      "\t\tAdding genomic sequences to exons: \tDone (0 sequences added, 0 ignored).\n",
      "\t\tReading sequence 'scaf_695', length: 30084\n",
      "\t\tAdding genomic sequences to exons: \tDone (1 sequences added, 0 ignored).\n",
      "\tTotal: 452083 sequences added, 0 sequences ignored.\n",
      "\n",
      "\tAdjusting transcripts: \n",
      "\tAdjusting genes: \n",
      "\tAdjusting chromosomes lengths: \n",
      "\tRanking exons: ....................................................................................................\n",
      "\t\t10000\t....................................................................................................\n",
      "\t\t20000\t....................................................................................................\n",
      "\t\t30000\t....................................................................................................\n",
      "\t\t40000\t....................................................................................................\n",
      "\t\t50000\t....................................................................................................\n",
      "\t\t60000\t....................................................................................................\n",
      "\t\t70000\t.........................\n",
      "\tCreate UTRs from CDS (if needed): \n",
      "\tCorrecting exons based on frame information.\n",
      "\t\n",
      "\tRemove empty chromosomes: \n",
      "\n",
      "\tMarking as 'coding' from CDS information: \n",
      "\tDone: 0 transcripts marked\n",
      "#-----------------------------------------------\n",
      "# Genome name                : 'Zea_mays_B73_NAM.5.0'\n",
      "# Genome version             : 'Zea_mays_B73_NAM.5.0'\n",
      "# Genome ID                  : 'Zea_mays_B73_NAM.5.0[0]'\n",
      "# Has protein coding info    : true\n",
      "# Has Tr. Support Level info : true\n",
      "# Genes                      : 39756\n",
      "# Protein coding genes       : 39756\n",
      "#-----------------------------------------------\n",
      "# Transcripts                : 72539\n",
      "# Avg. transcripts per gene  : 1.82\n",
      "# TSL transcripts            : 0\n",
      "#-----------------------------------------------\n",
      "# Checked transcripts        : \n",
      "#               AA sequences :      0 ( 0.00% )\n",
      "#              DNA sequences :      0 ( 0.00% )\n",
      "#-----------------------------------------------\n",
      "# Protein coding transcripts : 72539\n",
      "#              Length errors :      0 ( 0.00% )\n",
      "#  STOP codons in CDS errors :      0 ( 0.00% )\n",
      "#         START codon errors :    844 ( 1.16% )\n",
      "#        STOP codon warnings :   3679 ( 5.07% )\n",
      "#              UTR sequences :  65391 ( 90.15% )\n",
      "#               Total Errors :    844 ( 1.16% )\n",
      "#-----------------------------------------------\n",
      "# Cds                        : 403367\n",
      "# Exons                      : 452083\n",
      "# Exons with sequence        : 452083\n",
      "# Exons without sequence     : 0\n",
      "# Avg. exons per transcript  : 6.23\n",
      "# WARNING                    : No mitochondrion chromosome found\n",
      "#-----------------------------------------------\n",
      "# Number of chromosomes      : 685\n",
      "# Chromosomes                : Format 'chromo_name size codon_table'\n",
      "#\t\t'1'\t308452471\tStandard\n",
      "#\t\t'4'\t250330460\tStandard\n",
      "#\t\t'2'\t243675191\tStandard\n",
      "#\t\t'3'\t238017767\tStandard\n",
      "#\t\t'5'\t226353449\tStandard\n",
      "#\t\t'7'\t185808916\tStandard\n",
      "#\t\t'8'\t182411202\tStandard\n",
      "#\t\t'6'\t181357234\tStandard\n",
      "#\t\t'9'\t163004744\tStandard\n",
      "#\t\t'10'\t152435371\tStandard\n",
      "#\t\t'scaf_21'\t712123\tStandard\n",
      "#\t\t'scaf_22'\t570058\tStandard\n",
      "#\t\t'scaf_23'\t435198\tStandard\n",
      "#\t\t'scaf_24'\t387292\tStandard\n",
      "#\t\t'scaf_25'\t264983\tStandard\n",
      "#\t\t'scaf_26'\t254485\tStandard\n",
      "#\t\t'scaf_27'\t247825\tStandard\n",
      "#\t\t'scaf_28'\t239868\tStandard\n",
      "#\t\t'scaf_29'\t217815\tStandard\n",
      "#\t\t'scaf_30'\t207305\tStandard\n",
      "#\t\t'scaf_31'\t203238\tStandard\n",
      "#\t\t'scaf_32'\t197712\tStandard\n",
      "#\t\t'scaf_33'\t196160\tStandard\n",
      "#\t\t'scaf_34'\t190448\tStandard\n",
      "#\t\t'scaf_35'\t190269\tStandard\n",
      "#\t\t'scaf_36'\t187880\tStandard\n",
      "#\t\t'scaf_37'\t173911\tStandard\n",
      "#\t\t'scaf_38'\t173111\tStandard\n",
      "#\t\t'scaf_39'\t172167\tStandard\n",
      "#\t\t'scaf_40'\t171701\tStandard\n",
      "#\t\t'scaf_41'\t171519\tStandard\n",
      "#\t\t'scaf_42'\t169347\tStandard\n",
      "#\t\t'scaf_43'\t168052\tStandard\n",
      "#\t\t'scaf_44'\t164231\tStandard\n",
      "#\t\t'scaf_45'\t164060\tStandard\n",
      "#\t\t'scaf_46'\t160926\tStandard\n",
      "#\t\t'scaf_47'\t160502\tStandard\n",
      "#\t\t'scaf_48'\t159682\tStandard\n",
      "#\t\t'scaf_49'\t153763\tStandard\n",
      "#\t\t'scaf_50'\t153744\tStandard\n",
      "#\t\t'scaf_51'\t152095\tStandard\n",
      "#\t\t'scaf_52'\t151820\tStandard\n",
      "#\t\t'scaf_53'\t148690\tStandard\n",
      "#\t\t'scaf_54'\t144791\tStandard\n",
      "#\t\t'scaf_55'\t144328\tStandard\n",
      "#\t\t'scaf_56'\t142977\tStandard\n",
      "#\t\t'scaf_57'\t142290\tStandard\n",
      "#\t\t'scaf_58'\t141577\tStandard\n",
      "#\t\t'scaf_59'\t138583\tStandard\n",
      "#\t\t'scaf_60'\t137147\tStandard\n",
      "#\t\t'scaf_61'\t134092\tStandard\n",
      "#\t\t'scaf_62'\t133633\tStandard\n",
      "#\t\t'scaf_63'\t132531\tStandard\n",
      "#\t\t'scaf_64'\t130230\tStandard\n",
      "#\t\t'scaf_65'\t129299\tStandard\n",
      "#\t\t'scaf_66'\t128904\tStandard\n",
      "#\t\t'scaf_67'\t128482\tStandard\n",
      "#\t\t'scaf_68'\t127882\tStandard\n",
      "#\t\t'scaf_69'\t127290\tStandard\n",
      "#\t\t'scaf_70'\t126443\tStandard\n",
      "#\t\t'scaf_71'\t125047\tStandard\n",
      "#\t\t'scaf_72'\t124860\tStandard\n",
      "#\t\t'scaf_73'\t123043\tStandard\n",
      "#\t\t'scaf_74'\t120982\tStandard\n",
      "#\t\t'scaf_75'\t120904\tStandard\n",
      "#\t\t'scaf_76'\t120818\tStandard\n",
      "#\t\t'scaf_77'\t120549\tStandard\n",
      "#\t\t'scaf_78'\t120088\tStandard\n",
      "#\t\t'scaf_79'\t119143\tStandard\n",
      "#\t\t'scaf_80'\t118700\tStandard\n",
      "#\t\t'scaf_81'\t118576\tStandard\n",
      "#\t\t'scaf_82'\t118500\tStandard\n",
      "#\t\t'scaf_83'\t117266\tStandard\n",
      "#\t\t'scaf_84'\t117233\tStandard\n",
      "#\t\t'scaf_85'\t115765\tStandard\n",
      "#\t\t'scaf_86'\t115550\tStandard\n",
      "#\t\t'scaf_87'\t114887\tStandard\n",
      "#\t\t'scaf_88'\t114023\tStandard\n",
      "#\t\t'scaf_89'\t113862\tStandard\n",
      "#\t\t'scaf_90'\t113779\tStandard\n",
      "#\t\t'scaf_91'\t113002\tStandard\n",
      "#\t\t'scaf_92'\t112596\tStandard\n",
      "#\t\t'scaf_93'\t112550\tStandard\n",
      "#\t\t'scaf_94'\t111637\tStandard\n",
      "#\t\t'scaf_95'\t110562\tStandard\n",
      "#\t\t'scaf_96'\t110121\tStandard\n",
      "#\t\t'scaf_97'\t109951\tStandard\n",
      "#\t\t'scaf_98'\t109947\tStandard\n",
      "#\t\t'scaf_99'\t109352\tStandard\n",
      "#\t\t'scaf_100'\t109197\tStandard\n",
      "#\t\t'scaf_101'\t109031\tStandard\n",
      "#\t\t'scaf_102'\t108697\tStandard\n",
      "#\t\t'scaf_103'\t108378\tStandard\n",
      "#\t\t'scaf_104'\t108339\tStandard\n",
      "#\t\t'scaf_105'\t107956\tStandard\n",
      "#\t\t'scaf_106'\t107526\tStandard\n",
      "#\t\t'scaf_107'\t107524\tStandard\n",
      "#\t\t'scaf_108'\t106690\tStandard\n",
      "#\t\t'scaf_109'\t106305\tStandard\n",
      "#\t\t'scaf_110'\t106098\tStandard\n",
      "#\t\t'scaf_111'\t105289\tStandard\n",
      "#\t\t'scaf_112'\t105250\tStandard\n",
      "#\t\t'scaf_113'\t105226\tStandard\n",
      "#\t\t'scaf_114'\t105163\tStandard\n",
      "#\t\t'scaf_115'\t104537\tStandard\n",
      "#\t\t'scaf_116'\t104499\tStandard\n",
      "#\t\t'scaf_117'\t103808\tStandard\n",
      "#\t\t'scaf_118'\t103806\tStandard\n",
      "#\t\t'scaf_119'\t103210\tStandard\n",
      "#\t\t'scaf_120'\t102624\tStandard\n",
      "#\t\t'scaf_121'\t102406\tStandard\n",
      "#\t\t'scaf_122'\t102397\tStandard\n",
      "#\t\t'scaf_123'\t102349\tStandard\n",
      "#\t\t'scaf_124'\t101678\tStandard\n",
      "#\t\t'scaf_125'\t101641\tStandard\n",
      "#\t\t'scaf_126'\t100851\tStandard\n",
      "#\t\t'scaf_127'\t100600\tStandard\n",
      "#\t\t'scaf_128'\t100537\tStandard\n",
      "#\t\t'scaf_129'\t100364\tStandard\n",
      "#\t\t'scaf_130'\t99893\tStandard\n",
      "#\t\t'scaf_131'\t99674\tStandard\n",
      "#\t\t'scaf_132'\t98130\tStandard\n",
      "#\t\t'scaf_133'\t97996\tStandard\n",
      "#\t\t'scaf_134'\t97918\tStandard\n",
      "#\t\t'scaf_135'\t96911\tStandard\n",
      "#\t\t'scaf_136'\t96690\tStandard\n",
      "#\t\t'scaf_137'\t96496\tStandard\n",
      "#\t\t'scaf_138'\t96036\tStandard\n",
      "#\t\t'scaf_139'\t95934\tStandard\n",
      "#\t\t'scaf_140'\t95894\tStandard\n",
      "#\t\t'scaf_141'\t95890\tStandard\n",
      "#\t\t'scaf_142'\t95521\tStandard\n",
      "#\t\t'scaf_143'\t95419\tStandard\n",
      "#\t\t'scaf_144'\t95245\tStandard\n",
      "#\t\t'scaf_145'\t95129\tStandard\n",
      "#\t\t'scaf_146'\t94989\tStandard\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#\t\t'scaf_147'\t94793\tStandard\n",
      "#\t\t'scaf_148'\t94647\tStandard\n",
      "#\t\t'scaf_149'\t94334\tStandard\n",
      "#\t\t'scaf_150'\t94333\tStandard\n",
      "#\t\t'scaf_151'\t94036\tStandard\n",
      "#\t\t'scaf_152'\t93636\tStandard\n",
      "#\t\t'scaf_153'\t93514\tStandard\n",
      "#\t\t'scaf_154'\t93405\tStandard\n",
      "#\t\t'scaf_155'\t93298\tStandard\n",
      "#\t\t'scaf_156'\t93084\tStandard\n",
      "#\t\t'scaf_157'\t93056\tStandard\n",
      "#\t\t'scaf_158'\t92553\tStandard\n",
      "#\t\t'scaf_159'\t91995\tStandard\n",
      "#\t\t'scaf_160'\t91958\tStandard\n",
      "#\t\t'scaf_161'\t91897\tStandard\n",
      "#\t\t'scaf_162'\t91766\tStandard\n",
      "#\t\t'scaf_163'\t91439\tStandard\n",
      "#\t\t'scaf_164'\t91076\tStandard\n",
      "#\t\t'scaf_165'\t91072\tStandard\n",
      "#\t\t'scaf_166'\t91034\tStandard\n",
      "#\t\t'scaf_167'\t90901\tStandard\n",
      "#\t\t'scaf_168'\t90815\tStandard\n",
      "#\t\t'scaf_169'\t90613\tStandard\n",
      "#\t\t'scaf_170'\t90547\tStandard\n",
      "#\t\t'scaf_171'\t89502\tStandard\n",
      "#\t\t'scaf_172'\t89377\tStandard\n",
      "#\t\t'scaf_173'\t89242\tStandard\n",
      "#\t\t'scaf_174'\t89235\tStandard\n",
      "#\t\t'scaf_175'\t89155\tStandard\n",
      "#\t\t'scaf_176'\t88783\tStandard\n",
      "#\t\t'scaf_177'\t88032\tStandard\n",
      "#\t\t'scaf_178'\t87897\tStandard\n",
      "#\t\t'scaf_179'\t87788\tStandard\n",
      "#\t\t'scaf_180'\t87559\tStandard\n",
      "#\t\t'scaf_181'\t87401\tStandard\n",
      "#\t\t'scaf_182'\t87195\tStandard\n",
      "#\t\t'scaf_183'\t86681\tStandard\n",
      "#\t\t'scaf_184'\t86649\tStandard\n",
      "#\t\t'scaf_185'\t86437\tStandard\n",
      "#\t\t'scaf_186'\t86423\tStandard\n",
      "#\t\t'scaf_187'\t86029\tStandard\n",
      "#\t\t'scaf_188'\t85958\tStandard\n",
      "#\t\t'scaf_189'\t85939\tStandard\n",
      "#\t\t'scaf_190'\t85707\tStandard\n",
      "#\t\t'scaf_191'\t85696\tStandard\n",
      "#\t\t'scaf_192'\t85577\tStandard\n",
      "#\t\t'scaf_193'\t85542\tStandard\n",
      "#\t\t'scaf_194'\t85512\tStandard\n",
      "#\t\t'scaf_195'\t84529\tStandard\n",
      "#\t\t'scaf_196'\t84483\tStandard\n",
      "#\t\t'scaf_197'\t84480\tStandard\n",
      "#\t\t'scaf_198'\t84315\tStandard\n",
      "#\t\t'scaf_199'\t84283\tStandard\n",
      "#\t\t'scaf_200'\t84277\tStandard\n",
      "#\t\t'scaf_201'\t83765\tStandard\n",
      "#\t\t'scaf_202'\t83195\tStandard\n",
      "#\t\t'scaf_203'\t83150\tStandard\n",
      "#\t\t'scaf_204'\t83142\tStandard\n",
      "#\t\t'scaf_205'\t83098\tStandard\n",
      "#\t\t'scaf_206'\t82836\tStandard\n",
      "#\t\t'scaf_207'\t82698\tStandard\n",
      "#\t\t'scaf_208'\t82580\tStandard\n",
      "#\t\t'scaf_209'\t82439\tStandard\n",
      "#\t\t'scaf_210'\t82354\tStandard\n",
      "#\t\t'scaf_211'\t82347\tStandard\n",
      "#\t\t'scaf_212'\t82331\tStandard\n",
      "#\t\t'scaf_213'\t81960\tStandard\n",
      "#\t\t'scaf_214'\t81777\tStandard\n",
      "#\t\t'scaf_215'\t81747\tStandard\n",
      "#\t\t'scaf_216'\t81457\tStandard\n",
      "#\t\t'scaf_217'\t81340\tStandard\n",
      "#\t\t'scaf_218'\t81233\tStandard\n",
      "#\t\t'scaf_219'\t80936\tStandard\n",
      "#\t\t'scaf_220'\t80837\tStandard\n",
      "#\t\t'scaf_221'\t80814\tStandard\n",
      "#\t\t'scaf_222'\t80283\tStandard\n",
      "#\t\t'scaf_223'\t79710\tStandard\n",
      "#\t\t'scaf_224'\t79539\tStandard\n",
      "#\t\t'scaf_225'\t79526\tStandard\n",
      "#\t\t'scaf_226'\t79449\tStandard\n",
      "#\t\t'scaf_227'\t79237\tStandard\n",
      "#\t\t'scaf_228'\t78875\tStandard\n",
      "#\t\t'scaf_229'\t78242\tStandard\n",
      "#\t\t'scaf_230'\t78216\tStandard\n",
      "#\t\t'scaf_231'\t78109\tStandard\n",
      "#\t\t'scaf_232'\t77884\tStandard\n",
      "#\t\t'scaf_233'\t77858\tStandard\n",
      "#\t\t'scaf_234'\t77850\tStandard\n",
      "#\t\t'scaf_235'\t77518\tStandard\n",
      "#\t\t'scaf_236'\t77375\tStandard\n",
      "#\t\t'scaf_237'\t77233\tStandard\n",
      "#\t\t'scaf_238'\t77062\tStandard\n",
      "#\t\t'scaf_239'\t77005\tStandard\n",
      "#\t\t'scaf_240'\t76990\tStandard\n",
      "#\t\t'scaf_241'\t76896\tStandard\n",
      "#\t\t'scaf_242'\t76850\tStandard\n",
      "#\t\t'scaf_243'\t76719\tStandard\n",
      "#\t\t'scaf_244'\t76711\tStandard\n",
      "#\t\t'scaf_245'\t76434\tStandard\n",
      "#\t\t'scaf_246'\t76402\tStandard\n",
      "#\t\t'scaf_247'\t76349\tStandard\n",
      "#\t\t'scaf_248'\t76347\tStandard\n",
      "#\t\t'scaf_249'\t76253\tStandard\n",
      "#\t\t'scaf_250'\t76174\tStandard\n",
      "#\t\t'scaf_251'\t76113\tStandard\n",
      "#\t\t'scaf_252'\t76030\tStandard\n",
      "#\t\t'scaf_253'\t75901\tStandard\n",
      "#\t\t'scaf_254'\t75487\tStandard\n",
      "#\t\t'scaf_255'\t75327\tStandard\n",
      "#\t\t'scaf_256'\t75278\tStandard\n",
      "#\t\t'scaf_257'\t75264\tStandard\n",
      "#\t\t'scaf_258'\t75157\tStandard\n",
      "#\t\t'scaf_259'\t74994\tStandard\n",
      "#\t\t'scaf_260'\t74954\tStandard\n",
      "#\t\t'scaf_261'\t74875\tStandard\n",
      "#\t\t'scaf_262'\t74862\tStandard\n",
      "#\t\t'scaf_263'\t74687\tStandard\n",
      "#\t\t'scaf_264'\t74566\tStandard\n",
      "#\t\t'scaf_265'\t74453\tStandard\n",
      "#\t\t'scaf_266'\t73951\tStandard\n",
      "#\t\t'scaf_267'\t73767\tStandard\n",
      "#\t\t'scaf_268'\t73692\tStandard\n",
      "#\t\t'scaf_269'\t73531\tStandard\n",
      "#\t\t'scaf_270'\t73359\tStandard\n",
      "#\t\t'scaf_271'\t73251\tStandard\n",
      "#\t\t'scaf_272'\t73066\tStandard\n",
      "#\t\t'scaf_273'\t73001\tStandard\n",
      "#\t\t'scaf_274'\t72796\tStandard\n",
      "#\t\t'scaf_275'\t72665\tStandard\n",
      "#\t\t'scaf_276'\t72399\tStandard\n",
      "#\t\t'scaf_277'\t72053\tStandard\n",
      "#\t\t'scaf_278'\t71983\tStandard\n",
      "#\t\t'scaf_279'\t71870\tStandard\n",
      "#\t\t'scaf_280'\t71434\tStandard\n",
      "#\t\t'scaf_281'\t71227\tStandard\n",
      "#\t\t'scaf_282'\t71196\tStandard\n",
      "#\t\t'scaf_283'\t71104\tStandard\n",
      "#\t\t'scaf_284'\t70953\tStandard\n",
      "#\t\t'scaf_285'\t70845\tStandard\n",
      "#\t\t'scaf_286'\t70600\tStandard\n",
      "#\t\t'scaf_287'\t70547\tStandard\n",
      "#\t\t'scaf_288'\t70410\tStandard\n",
      "#\t\t'scaf_289'\t70226\tStandard\n",
      "#\t\t'scaf_290'\t70070\tStandard\n",
      "#\t\t'scaf_291'\t69844\tStandard\n",
      "#\t\t'scaf_292'\t69796\tStandard\n",
      "#\t\t'scaf_293'\t69747\tStandard\n",
      "#\t\t'scaf_294'\t69542\tStandard\n",
      "#\t\t'scaf_295'\t69529\tStandard\n",
      "#\t\t'scaf_296'\t69442\tStandard\n",
      "#\t\t'scaf_297'\t69369\tStandard\n",
      "#\t\t'scaf_298'\t69364\tStandard\n",
      "#\t\t'scaf_299'\t69243\tStandard\n",
      "#\t\t'scaf_300'\t69101\tStandard\n",
      "#\t\t'scaf_301'\t69023\tStandard\n",
      "#\t\t'scaf_302'\t68967\tStandard\n",
      "#\t\t'scaf_303'\t68912\tStandard\n",
      "#\t\t'scaf_304'\t68810\tStandard\n",
      "#\t\t'scaf_305'\t68803\tStandard\n",
      "#\t\t'scaf_306'\t68764\tStandard\n",
      "#\t\t'scaf_307'\t68750\tStandard\n",
      "#\t\t'scaf_308'\t68697\tStandard\n",
      "#\t\t'scaf_309'\t68464\tStandard\n",
      "#\t\t'scaf_310'\t68461\tStandard\n",
      "#\t\t'scaf_311'\t68394\tStandard\n",
      "#\t\t'scaf_312'\t68386\tStandard\n",
      "#\t\t'scaf_313'\t68340\tStandard\n",
      "#\t\t'scaf_314'\t68200\tStandard\n",
      "#\t\t'scaf_315'\t68185\tStandard\n",
      "#\t\t'scaf_316'\t68157\tStandard\n",
      "#\t\t'scaf_317'\t68122\tStandard\n",
      "#\t\t'scaf_318'\t67773\tStandard\n",
      "#\t\t'scaf_319'\t67715\tStandard\n",
      "#\t\t'scaf_320'\t67453\tStandard\n",
      "#\t\t'scaf_321'\t67423\tStandard\n",
      "#\t\t'scaf_322'\t67253\tStandard\n",
      "#\t\t'scaf_323'\t67217\tStandard\n",
      "#\t\t'scaf_324'\t67121\tStandard\n",
      "#\t\t'scaf_325'\t67094\tStandard\n",
      "#\t\t'scaf_326'\t66933\tStandard\n",
      "#\t\t'scaf_327'\t66933\tStandard\n",
      "#\t\t'scaf_328'\t66728\tStandard\n",
      "#\t\t'scaf_329'\t66653\tStandard\n",
      "#\t\t'scaf_330'\t66505\tStandard\n",
      "#\t\t'scaf_331'\t66412\tStandard\n",
      "#\t\t'scaf_332'\t66353\tStandard\n",
      "#\t\t'scaf_333'\t66282\tStandard\n",
      "#\t\t'scaf_334'\t66272\tStandard\n",
      "#\t\t'scaf_335'\t66081\tStandard\n",
      "#\t\t'scaf_336'\t66016\tStandard\n",
      "#\t\t'scaf_337'\t65790\tStandard\n",
      "#\t\t'scaf_338'\t65540\tStandard\n",
      "#\t\t'scaf_339'\t65537\tStandard\n",
      "#\t\t'scaf_340'\t65427\tStandard\n",
      "#\t\t'scaf_341'\t65327\tStandard\n",
      "#\t\t'scaf_342'\t65280\tStandard\n",
      "#\t\t'scaf_343'\t65236\tStandard\n",
      "#\t\t'scaf_344'\t65094\tStandard\n",
      "#\t\t'scaf_345'\t64929\tStandard\n",
      "#\t\t'scaf_346'\t64906\tStandard\n",
      "#\t\t'scaf_347'\t64744\tStandard\n",
      "#\t\t'scaf_348'\t64714\tStandard\n",
      "#\t\t'scaf_349'\t64685\tStandard\n",
      "#\t\t'scaf_350'\t64593\tStandard\n",
      "#\t\t'scaf_351'\t64495\tStandard\n",
      "#\t\t'scaf_352'\t64265\tStandard\n",
      "#\t\t'scaf_353'\t64155\tStandard\n",
      "#\t\t'scaf_354'\t64072\tStandard\n",
      "#\t\t'scaf_355'\t64028\tStandard\n",
      "#\t\t'scaf_356'\t63985\tStandard\n",
      "#\t\t'scaf_357'\t63908\tStandard\n",
      "#\t\t'scaf_358'\t63880\tStandard\n",
      "#\t\t'scaf_359'\t63432\tStandard\n",
      "#\t\t'scaf_360'\t63266\tStandard\n",
      "#\t\t'scaf_361'\t63177\tStandard\n",
      "#\t\t'scaf_362'\t63148\tStandard\n",
      "#\t\t'scaf_363'\t63113\tStandard\n",
      "#\t\t'scaf_364'\t63084\tStandard\n",
      "#\t\t'scaf_365'\t63080\tStandard\n",
      "#\t\t'scaf_366'\t62556\tStandard\n",
      "#\t\t'scaf_367'\t62541\tStandard\n",
      "#\t\t'scaf_368'\t62533\tStandard\n",
      "#\t\t'scaf_369'\t62516\tStandard\n",
      "#\t\t'scaf_370'\t62509\tStandard\n",
      "#\t\t'scaf_371'\t62284\tStandard\n",
      "#\t\t'scaf_372'\t62204\tStandard\n",
      "#\t\t'scaf_373'\t62066\tStandard\n",
      "#\t\t'scaf_374'\t62047\tStandard\n",
      "#\t\t'scaf_375'\t61971\tStandard\n",
      "#\t\t'scaf_376'\t61863\tStandard\n",
      "#\t\t'scaf_377'\t61815\tStandard\n",
      "#\t\t'scaf_378'\t61663\tStandard\n",
      "#\t\t'scaf_379'\t61251\tStandard\n",
      "#\t\t'scaf_380'\t61109\tStandard\n",
      "#\t\t'scaf_381'\t61002\tStandard\n",
      "#\t\t'scaf_382'\t60996\tStandard\n",
      "#\t\t'scaf_383'\t60990\tStandard\n",
      "#\t\t'scaf_384'\t60988\tStandard\n",
      "#\t\t'scaf_385'\t60794\tStandard\n",
      "#\t\t'scaf_386'\t60618\tStandard\n",
      "#\t\t'scaf_387'\t60414\tStandard\n",
      "#\t\t'scaf_388'\t60261\tStandard\n",
      "#\t\t'scaf_389'\t60203\tStandard\n",
      "#\t\t'scaf_390'\t60132\tStandard\n",
      "#\t\t'scaf_391'\t60044\tStandard\n",
      "#\t\t'scaf_392'\t59970\tStandard\n",
      "#\t\t'scaf_393'\t59921\tStandard\n",
      "#\t\t'scaf_394'\t59838\tStandard\n",
      "#\t\t'scaf_395'\t59635\tStandard\n",
      "#\t\t'scaf_396'\t59584\tStandard\n",
      "#\t\t'scaf_397'\t59572\tStandard\n",
      "#\t\t'scaf_398'\t59286\tStandard\n",
      "#\t\t'scaf_399'\t59208\tStandard\n",
      "#\t\t'scaf_400'\t59116\tStandard\n",
      "#\t\t'scaf_401'\t59113\tStandard\n",
      "#\t\t'scaf_402'\t58978\tStandard\n",
      "#\t\t'scaf_403'\t58950\tStandard\n",
      "#\t\t'scaf_404'\t58762\tStandard\n",
      "#\t\t'scaf_405'\t58605\tStandard\n",
      "#\t\t'scaf_406'\t58599\tStandard\n",
      "#\t\t'scaf_407'\t58456\tStandard\n",
      "#\t\t'scaf_408'\t58405\tStandard\n",
      "#\t\t'scaf_409'\t58344\tStandard\n",
      "#\t\t'scaf_410'\t57998\tStandard\n",
      "#\t\t'scaf_411'\t57932\tStandard\n",
      "#\t\t'scaf_412'\t57849\tStandard\n",
      "#\t\t'scaf_413'\t57812\tStandard\n",
      "#\t\t'scaf_414'\t57501\tStandard\n",
      "#\t\t'scaf_415'\t57472\tStandard\n",
      "#\t\t'scaf_416'\t57371\tStandard\n",
      "#\t\t'scaf_417'\t57210\tStandard\n",
      "#\t\t'scaf_418'\t57042\tStandard\n",
      "#\t\t'scaf_419'\t57011\tStandard\n",
      "#\t\t'scaf_420'\t56837\tStandard\n",
      "#\t\t'scaf_421'\t56819\tStandard\n",
      "#\t\t'scaf_422'\t56707\tStandard\n",
      "#\t\t'scaf_423'\t56705\tStandard\n",
      "#\t\t'scaf_424'\t56693\tStandard\n",
      "#\t\t'scaf_425'\t56600\tStandard\n",
      "#\t\t'scaf_426'\t56555\tStandard\n",
      "#\t\t'scaf_427'\t56516\tStandard\n",
      "#\t\t'scaf_428'\t56435\tStandard\n",
      "#\t\t'scaf_429'\t56416\tStandard\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#\t\t'scaf_430'\t56249\tStandard\n",
      "#\t\t'scaf_431'\t56149\tStandard\n",
      "#\t\t'scaf_432'\t56140\tStandard\n",
      "#\t\t'scaf_433'\t55942\tStandard\n",
      "#\t\t'scaf_434'\t55813\tStandard\n",
      "#\t\t'scaf_435'\t55746\tStandard\n",
      "#\t\t'scaf_436'\t55605\tStandard\n",
      "#\t\t'scaf_437'\t55516\tStandard\n",
      "#\t\t'scaf_438'\t55489\tStandard\n",
      "#\t\t'scaf_439'\t55303\tStandard\n",
      "#\t\t'scaf_440'\t55289\tStandard\n",
      "#\t\t'scaf_441'\t55099\tStandard\n",
      "#\t\t'scaf_442'\t55048\tStandard\n",
      "#\t\t'scaf_443'\t54923\tStandard\n",
      "#\t\t'scaf_444'\t54898\tStandard\n",
      "#\t\t'scaf_445'\t54817\tStandard\n",
      "#\t\t'scaf_446'\t54746\tStandard\n",
      "#\t\t'scaf_447'\t54638\tStandard\n",
      "#\t\t'scaf_448'\t54628\tStandard\n",
      "#\t\t'scaf_449'\t54545\tStandard\n",
      "#\t\t'scaf_450'\t54531\tStandard\n",
      "#\t\t'scaf_451'\t54430\tStandard\n",
      "#\t\t'scaf_452'\t54309\tStandard\n",
      "#\t\t'scaf_453'\t54288\tStandard\n",
      "#\t\t'scaf_454'\t54132\tStandard\n",
      "#\t\t'scaf_455'\t54093\tStandard\n",
      "#\t\t'scaf_456'\t53967\tStandard\n",
      "#\t\t'scaf_457'\t53894\tStandard\n",
      "#\t\t'scaf_458'\t53843\tStandard\n",
      "#\t\t'scaf_459'\t53481\tStandard\n",
      "#\t\t'scaf_460'\t53434\tStandard\n",
      "#\t\t'scaf_461'\t53417\tStandard\n",
      "#\t\t'scaf_462'\t53195\tStandard\n",
      "#\t\t'scaf_463'\t53086\tStandard\n",
      "#\t\t'scaf_464'\t52993\tStandard\n",
      "#\t\t'scaf_465'\t52955\tStandard\n",
      "#\t\t'scaf_466'\t52889\tStandard\n",
      "#\t\t'scaf_467'\t52720\tStandard\n",
      "#\t\t'scaf_468'\t52562\tStandard\n",
      "#\t\t'scaf_469'\t52527\tStandard\n",
      "#\t\t'scaf_470'\t52467\tStandard\n",
      "#\t\t'scaf_471'\t52001\tStandard\n",
      "#\t\t'scaf_472'\t51792\tStandard\n",
      "#\t\t'scaf_473'\t51791\tStandard\n",
      "#\t\t'scaf_474'\t51790\tStandard\n",
      "#\t\t'scaf_475'\t51498\tStandard\n",
      "#\t\t'scaf_476'\t51442\tStandard\n",
      "#\t\t'scaf_477'\t51437\tStandard\n",
      "#\t\t'scaf_478'\t51436\tStandard\n",
      "#\t\t'scaf_479'\t51400\tStandard\n",
      "#\t\t'scaf_480'\t51317\tStandard\n",
      "#\t\t'scaf_481'\t51285\tStandard\n",
      "#\t\t'scaf_482'\t51226\tStandard\n",
      "#\t\t'scaf_483'\t51166\tStandard\n",
      "#\t\t'scaf_484'\t51159\tStandard\n",
      "#\t\t'scaf_485'\t51097\tStandard\n",
      "#\t\t'scaf_486'\t51050\tStandard\n",
      "#\t\t'scaf_487'\t51005\tStandard\n",
      "#\t\t'scaf_488'\t50467\tStandard\n",
      "#\t\t'scaf_489'\t50303\tStandard\n",
      "#\t\t'scaf_490'\t50230\tStandard\n",
      "#\t\t'scaf_491'\t50119\tStandard\n",
      "#\t\t'scaf_492'\t50112\tStandard\n",
      "#\t\t'scaf_493'\t50095\tStandard\n",
      "#\t\t'scaf_494'\t50090\tStandard\n",
      "#\t\t'scaf_495'\t49999\tStandard\n",
      "#\t\t'scaf_496'\t49900\tStandard\n",
      "#\t\t'scaf_497'\t49811\tStandard\n",
      "#\t\t'scaf_498'\t49759\tStandard\n",
      "#\t\t'scaf_499'\t49573\tStandard\n",
      "#\t\t'scaf_500'\t49572\tStandard\n",
      "#\t\t'scaf_501'\t49558\tStandard\n",
      "#\t\t'scaf_502'\t49534\tStandard\n",
      "#\t\t'scaf_503'\t49321\tStandard\n",
      "#\t\t'scaf_504'\t49133\tStandard\n",
      "#\t\t'scaf_505'\t49076\tStandard\n",
      "#\t\t'scaf_506'\t48960\tStandard\n",
      "#\t\t'scaf_507'\t48845\tStandard\n",
      "#\t\t'scaf_508'\t48796\tStandard\n",
      "#\t\t'scaf_509'\t48793\tStandard\n",
      "#\t\t'scaf_510'\t48788\tStandard\n",
      "#\t\t'scaf_511'\t48698\tStandard\n",
      "#\t\t'scaf_512'\t48411\tStandard\n",
      "#\t\t'scaf_513'\t48321\tStandard\n",
      "#\t\t'scaf_514'\t48292\tStandard\n",
      "#\t\t'scaf_515'\t48059\tStandard\n",
      "#\t\t'scaf_516'\t48018\tStandard\n",
      "#\t\t'scaf_517'\t47885\tStandard\n",
      "#\t\t'scaf_518'\t47771\tStandard\n",
      "#\t\t'scaf_519'\t47732\tStandard\n",
      "#\t\t'scaf_520'\t47726\tStandard\n",
      "#\t\t'scaf_521'\t47702\tStandard\n",
      "#\t\t'scaf_522'\t47686\tStandard\n",
      "#\t\t'scaf_523'\t47658\tStandard\n",
      "#\t\t'scaf_524'\t47519\tStandard\n",
      "#\t\t'scaf_525'\t47504\tStandard\n",
      "#\t\t'scaf_526'\t47428\tStandard\n",
      "#\t\t'scaf_527'\t47425\tStandard\n",
      "#\t\t'scaf_528'\t47422\tStandard\n",
      "#\t\t'scaf_529'\t47062\tStandard\n",
      "#\t\t'scaf_530'\t47056\tStandard\n",
      "#\t\t'scaf_531'\t47032\tStandard\n",
      "#\t\t'scaf_532'\t46775\tStandard\n",
      "#\t\t'scaf_533'\t46694\tStandard\n",
      "#\t\t'scaf_534'\t46575\tStandard\n",
      "#\t\t'scaf_535'\t46567\tStandard\n",
      "#\t\t'scaf_536'\t46533\tStandard\n",
      "#\t\t'scaf_537'\t46395\tStandard\n",
      "#\t\t'scaf_538'\t46312\tStandard\n",
      "#\t\t'scaf_539'\t46276\tStandard\n",
      "#\t\t'scaf_540'\t46195\tStandard\n",
      "#\t\t'scaf_541'\t46105\tStandard\n",
      "#\t\t'scaf_542'\t45928\tStandard\n",
      "#\t\t'scaf_543'\t45898\tStandard\n",
      "#\t\t'scaf_544'\t45870\tStandard\n",
      "#\t\t'scaf_545'\t45852\tStandard\n",
      "#\t\t'scaf_546'\t45599\tStandard\n",
      "#\t\t'scaf_547'\t45570\tStandard\n",
      "#\t\t'scaf_548'\t45567\tStandard\n",
      "#\t\t'scaf_549'\t45460\tStandard\n",
      "#\t\t'scaf_550'\t45408\tStandard\n",
      "#\t\t'scaf_551'\t45391\tStandard\n",
      "#\t\t'scaf_552'\t45290\tStandard\n",
      "#\t\t'scaf_553'\t45286\tStandard\n",
      "#\t\t'scaf_554'\t45262\tStandard\n",
      "#\t\t'scaf_555'\t45165\tStandard\n",
      "#\t\t'scaf_556'\t44942\tStandard\n",
      "#\t\t'scaf_557'\t44777\tStandard\n",
      "#\t\t'scaf_558'\t44734\tStandard\n",
      "#\t\t'scaf_559'\t44719\tStandard\n",
      "#\t\t'scaf_560'\t44617\tStandard\n",
      "#\t\t'scaf_561'\t44557\tStandard\n",
      "#\t\t'scaf_562'\t44525\tStandard\n",
      "#\t\t'scaf_563'\t44435\tStandard\n",
      "#\t\t'scaf_564'\t44391\tStandard\n",
      "#\t\t'scaf_565'\t44377\tStandard\n",
      "#\t\t'scaf_566'\t44336\tStandard\n",
      "#\t\t'scaf_567'\t44149\tStandard\n",
      "#\t\t'scaf_568'\t43798\tStandard\n",
      "#\t\t'scaf_569'\t43689\tStandard\n",
      "#\t\t'scaf_570'\t43564\tStandard\n",
      "#\t\t'scaf_571'\t43505\tStandard\n",
      "#\t\t'scaf_572'\t43361\tStandard\n",
      "#\t\t'scaf_573'\t43335\tStandard\n",
      "#\t\t'scaf_574'\t43270\tStandard\n",
      "#\t\t'scaf_575'\t43186\tStandard\n",
      "#\t\t'scaf_576'\t43039\tStandard\n",
      "#\t\t'scaf_577'\t42914\tStandard\n",
      "#\t\t'scaf_578'\t42890\tStandard\n",
      "#\t\t'scaf_579'\t42824\tStandard\n",
      "#\t\t'scaf_580'\t42746\tStandard\n",
      "#\t\t'scaf_581'\t42734\tStandard\n",
      "#\t\t'scaf_582'\t42695\tStandard\n",
      "#\t\t'scaf_583'\t42670\tStandard\n",
      "#\t\t'scaf_584'\t42655\tStandard\n",
      "#\t\t'scaf_585'\t42653\tStandard\n",
      "#\t\t'scaf_586'\t42361\tStandard\n",
      "#\t\t'scaf_587'\t42140\tStandard\n",
      "#\t\t'scaf_588'\t42065\tStandard\n",
      "#\t\t'scaf_589'\t42059\tStandard\n",
      "#\t\t'scaf_590'\t42054\tStandard\n",
      "#\t\t'scaf_591'\t42047\tStandard\n",
      "#\t\t'scaf_592'\t42042\tStandard\n",
      "#\t\t'scaf_593'\t41905\tStandard\n",
      "#\t\t'scaf_594'\t41863\tStandard\n",
      "#\t\t'scaf_595'\t41770\tStandard\n",
      "#\t\t'scaf_596'\t41630\tStandard\n",
      "#\t\t'scaf_597'\t41600\tStandard\n",
      "#\t\t'scaf_598'\t41539\tStandard\n",
      "#\t\t'scaf_599'\t41383\tStandard\n",
      "#\t\t'scaf_600'\t41377\tStandard\n",
      "#\t\t'scaf_601'\t41329\tStandard\n",
      "#\t\t'scaf_602'\t41219\tStandard\n",
      "#\t\t'scaf_603'\t41109\tStandard\n",
      "#\t\t'scaf_604'\t41107\tStandard\n",
      "#\t\t'scaf_605'\t41057\tStandard\n",
      "#\t\t'scaf_606'\t41030\tStandard\n",
      "#\t\t'scaf_607'\t40971\tStandard\n",
      "#\t\t'scaf_608'\t40905\tStandard\n",
      "#\t\t'scaf_609'\t40876\tStandard\n",
      "#\t\t'scaf_610'\t40847\tStandard\n",
      "#\t\t'scaf_611'\t40814\tStandard\n",
      "#\t\t'scaf_612'\t40576\tStandard\n",
      "#\t\t'scaf_613'\t40560\tStandard\n",
      "#\t\t'scaf_614'\t40479\tStandard\n",
      "#\t\t'scaf_615'\t40402\tStandard\n",
      "#\t\t'scaf_616'\t40389\tStandard\n",
      "#\t\t'scaf_617'\t40165\tStandard\n",
      "#\t\t'scaf_618'\t40145\tStandard\n",
      "#\t\t'scaf_619'\t40064\tStandard\n",
      "#\t\t'scaf_620'\t40006\tStandard\n",
      "#\t\t'scaf_621'\t39904\tStandard\n",
      "#\t\t'scaf_622'\t39697\tStandard\n",
      "#\t\t'scaf_623'\t39553\tStandard\n",
      "#\t\t'scaf_624'\t39481\tStandard\n",
      "#\t\t'scaf_625'\t39460\tStandard\n",
      "#\t\t'scaf_626'\t39077\tStandard\n",
      "#\t\t'scaf_627'\t39038\tStandard\n",
      "#\t\t'scaf_628'\t38891\tStandard\n",
      "#\t\t'scaf_629'\t38764\tStandard\n",
      "#\t\t'scaf_630'\t38610\tStandard\n",
      "#\t\t'scaf_631'\t38558\tStandard\n",
      "#\t\t'scaf_632'\t38463\tStandard\n",
      "#\t\t'scaf_633'\t38456\tStandard\n",
      "#\t\t'scaf_634'\t38214\tStandard\n",
      "#\t\t'scaf_635'\t37995\tStandard\n",
      "#\t\t'scaf_636'\t37985\tStandard\n",
      "#\t\t'scaf_637'\t37569\tStandard\n",
      "#\t\t'scaf_638'\t37477\tStandard\n",
      "#\t\t'scaf_639'\t37416\tStandard\n",
      "#\t\t'scaf_640'\t37393\tStandard\n",
      "#\t\t'scaf_641'\t37334\tStandard\n",
      "#\t\t'scaf_642'\t37315\tStandard\n",
      "#\t\t'scaf_643'\t37184\tStandard\n",
      "#\t\t'scaf_644'\t37110\tStandard\n",
      "#\t\t'scaf_645'\t37080\tStandard\n",
      "#\t\t'scaf_646'\t36930\tStandard\n",
      "#\t\t'scaf_647'\t36782\tStandard\n",
      "#\t\t'scaf_648'\t36641\tStandard\n",
      "#\t\t'scaf_649'\t36626\tStandard\n",
      "#\t\t'scaf_650'\t36526\tStandard\n",
      "#\t\t'scaf_651'\t36525\tStandard\n",
      "#\t\t'scaf_652'\t36369\tStandard\n",
      "#\t\t'scaf_653'\t36282\tStandard\n",
      "#\t\t'scaf_654'\t36267\tStandard\n",
      "#\t\t'scaf_655'\t36266\tStandard\n",
      "#\t\t'scaf_656'\t36217\tStandard\n",
      "#\t\t'scaf_657'\t35909\tStandard\n",
      "#\t\t'scaf_658'\t35909\tStandard\n",
      "#\t\t'scaf_659'\t35884\tStandard\n",
      "#\t\t'scaf_660'\t35784\tStandard\n",
      "#\t\t'scaf_661'\t35720\tStandard\n",
      "#\t\t'scaf_662'\t35668\tStandard\n",
      "#\t\t'scaf_663'\t35599\tStandard\n",
      "#\t\t'scaf_664'\t35539\tStandard\n",
      "#\t\t'scaf_665'\t35297\tStandard\n",
      "#\t\t'scaf_666'\t35285\tStandard\n",
      "#\t\t'scaf_667'\t35263\tStandard\n",
      "#\t\t'scaf_668'\t35100\tStandard\n",
      "#\t\t'scaf_669'\t34989\tStandard\n",
      "#\t\t'scaf_670'\t34940\tStandard\n",
      "#\t\t'scaf_671'\t34852\tStandard\n",
      "#\t\t'scaf_672'\t34684\tStandard\n",
      "#\t\t'scaf_673'\t34626\tStandard\n",
      "#\t\t'scaf_674'\t34568\tStandard\n",
      "#\t\t'scaf_675'\t34427\tStandard\n",
      "#\t\t'scaf_676'\t34391\tStandard\n",
      "#\t\t'scaf_677'\t33997\tStandard\n",
      "#\t\t'scaf_678'\t33938\tStandard\n",
      "#\t\t'scaf_679'\t33821\tStandard\n",
      "#\t\t'scaf_680'\t33689\tStandard\n",
      "#\t\t'scaf_681'\t33683\tStandard\n",
      "#\t\t'scaf_682'\t33634\tStandard\n",
      "#\t\t'scaf_683'\t33496\tStandard\n",
      "#\t\t'scaf_684'\t33264\tStandard\n",
      "#\t\t'scaf_685'\t32618\tStandard\n",
      "#\t\t'scaf_686'\t32197\tStandard\n",
      "#\t\t'scaf_687'\t32019\tStandard\n",
      "#\t\t'scaf_688'\t31944\tStandard\n",
      "#\t\t'scaf_689'\t31748\tStandard\n",
      "#\t\t'scaf_690'\t31498\tStandard\n",
      "#\t\t'scaf_691'\t31489\tStandard\n",
      "#\t\t'scaf_692'\t31211\tStandard\n",
      "#\t\t'scaf_693'\t30818\tStandard\n",
      "#\t\t'scaf_694'\t30512\tStandard\n",
      "#\t\t'scaf_695'\t30084\tStandard\n",
      "#-----------------------------------------------\n",
      "\n",
      "00:00:27\tCaracterizing exons by splicing (stage 1) : \n",
      "\t....................................................................................................\n",
      "\t100000\t....................................................................................................\n",
      "\t200000\t....................................................................................................\n",
      "\t300000\t....................................................................................................\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t400000\t....................................................\n",
      "00:00:27\tCaracterizing exons by splicing (stage 2) : \n",
      "\t....................................................................................................\n",
      "\t100000\t....................................................................................................\n",
      "\t200000\t....................................................................................................\n",
      "\t300000\t....................................................................................................\n",
      "\t400000\t....................................................00:00:28\tdone.\n",
      "00:00:28\t[Optional] Rare amino acid annotations\n",
      "00:00:28\tWarning: Cannot read optional protein sequence file '/Users/aricha14/Documents/Tools/snpEff/./data/Zea_mays_B73_NAM.5.0/protein.fa', nothing done.\n",
      "00:00:28\tSaving database\n",
      "00:00:44\t[Optional] Reading regulation elements: GFF\n",
      "00:00:44\tWarning: Cannot read optional regulation file '/Users/aricha14/Documents/Tools/snpEff/./data/Zea_mays_B73_NAM.5.0/regulation.gff', nothing done.\n",
      "00:00:44\t[Optional] Reading regulation elements: BED \n",
      "00:00:44\tCannot find optional regulation dir '/Users/aricha14/Documents/Tools/snpEff/./data/Zea_mays_B73_NAM.5.0/regulation.bed/', nothing done.\n",
      "00:00:44\t[Optional] Reading motifs: GFF\n",
      "00:00:44\tWarning: Cannot open PWMs file /Users/aricha14/Documents/Tools/snpEff/./data/Zea_mays_B73_NAM.5.0/pwms.bin. Nothing done\n",
      "00:00:44\tDone\n",
      "00:00:44\tLogging\n",
      "00:00:45\tChecking for updates...\n",
      "00:00:46\tDone.\n"
     ]
    }
   ],
   "source": [
    "#Build the database \n",
    "cd /Users/aricha14/Documents/Tools/snpEff\n",
    "java -jar snpEff.jar build -gff3 -v Zea_mays_B73_NAM.5.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "f40b5f28",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dickeya_zeae_ec1                                            \tDickeya_zeae_ec1                                            \t          \t                              \thttps://snpeff.blob.core.windows.net/databases/v5_0/snpEff_v5_0_Dickeya_zeae_ec1.zip\n",
      "Dickeya_zeae_ech586                                         \tDickeya_zeae_ech586                                         \t          \t                              \thttps://snpeff.blob.core.windows.net/databases/v5_0/snpEff_v5_0_Dickeya_zeae_ech586.zip\n",
      "Lactobacillus_zeae_dsm_20178_kctc_3804                      \tLactobacillus_zeae_dsm_20178_kctc_3804                      \t          \t                              \thttps://snpeff.blob.core.windows.net/databases/v5_0/snpEff_v5_0_Lactobacillus_zeae_dsm_20178_kctc_3804.zip\n",
      "Lentzea_guizhouensis                                        \tLentzea_guizhouensis                                        \t          \t                              \thttps://snpeff.blob.core.windows.net/databases/v5_0/snpEff_v5_0_Lentzea_guizhouensis.zip\n",
      "Siansivirga_zeaxanthinifaciens_cc_samt_1                    \tSiansivirga_zeaxanthinifaciens_cc_samt_1                    \t          \t                              \thttps://snpeff.blob.core.windows.net/databases/v5_0/snpEff_v5_0_Siansivirga_zeaxanthinifaciens_cc_samt_1.zip\n",
      "Zea_mays                                                    \tZea_mays                                                    \t          \t                              \thttps://snpeff.blob.core.windows.net/databases/v5_0/snpEff_v5_0_Zea_mays.zip\n",
      "Zea_mays_B73_NAM.5.0                                        \tZea_mays_B73_NAM.5.0                                        \tOK        \t                              \thttps://snpeff.blob.core.windows.net/databases/v5_0/snpEff_v5_0_Zea_mays_B73_NAM.5.0.zip\n",
      "Zea_mays_mo17_CAU.1.0                                       \tZea_mays_mo17_CAU.1.0                                       \tOK        \t                              \thttps://snpeff.blob.core.windows.net/databases/v5_0/snpEff_v5_0_Zea_mays_mo17_CAU.1.0.zip\n"
     ]
    }
   ],
   "source": [
    "# Check to see if the correct genome is added \n",
    "java -jar /Users/aricha14/Documents/Tools/snpEff/snpEff.jar databases | grep -i Zea"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ee553140",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: snpeff: File exists\n"
     ]
    }
   ],
   "source": [
    "# Now I have built the snpeff database I can run snpeff on the vcf files \n",
    "#Snpeff code \n",
    "cd /Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022\n",
    "mkdir snpeff\n",
    "java -Xmx4g -jar /Users/aricha14/Documents/Tools/snpEff/snpEff.jar Zea_mays_B73_NAM.5.0 /Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_MAy2022/TrfM6_WGS_NAM5_mpileup_Dec2022.vcf > /Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022/snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022.ann.vcf\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4942477f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\t/Volumes/Elements/Sequence_Data/WGS_data/Trf_mutant_WGS_2022/WGS_Alignment_NAM5_May2022/BAM_reads/TrfM6_aligned_NAM5.bam\n",
      "##fileformat=VCFv4.2\n",
      "##FILTER=<ID=PASS,Description=\"All filters passed\">\n",
      "##samtoolsVersion=1.14+htslib-1.14\n",
      "##samtoolsCommand=samtools mpileup -f /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0_chr.fa -g /Volumes/Elements/Sequence_Data/WGS_data/Trf_mutant_WGS_2022/WGS_Alignment_NAM5_May2022/BAM_reads/TrfM6_aligned_NAM5.bam\n",
      "##reference=file:///Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0_chr.fa\n",
      "##contig=<ID=chr1,length=308452471>\n",
      "##contig=<ID=chr2,length=243675191>\n",
      "##contig=<ID=chr3,length=238017767>\n",
      "##contig=<ID=chr4,length=250330460>\n",
      "##contig=<ID=chr5,length=226353449>\n"
     ]
    }
   ],
   "source": [
    "#Check the output file of SnpEff to see the annotations\n",
    "\n",
    "#Check the output file to make sure that it looks ok\n",
    "\n",
    "awk '{if($1==\"#CHROM\") print $0}' ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022.ann.vcf\n",
    "\n",
    "head -10 ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022.ann.vcf \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d14c2070",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\t/Volumes/Elements/Sequence_Data/WGS_data/Trf_mutant_WGS_2022/WGS_Alignment_NAM5_May2022/BAM_reads/TrfM6_aligned_NAM5.bam\n",
      "chr10\t9899\t.\tA\tG\t3.7766\t.\tDP=11;VDB=0.198166;SGB=-0.670168;MQSB=0.733186;MQ0F=0;AC=2;AN=2;DP4=0,0,3,7;MQ=3;ANN=G|intergenic_region|MODIFIER|CHR_START-Zm00001eb404740|CHR_START-Zm00001eb404740|intergenic_region|CHR_START-Zm00001eb404740|||n.9899A>G||||||\tGT:PL\t1/1:31,30,0\n",
      "chr10\t10791\t.\tA\tG\t5.04598\t.\tDP=18;VDB=0.576628;SGB=-0.691153;MQSB=0.0235981;MQ0F=0.111111;AC=2;AN=2;DP4=0,0,9,9;MQ=2;ANN=G|intergenic_region|MODIFIER|CHR_START-Zm00001eb404740|CHR_START-Zm00001eb404740|intergenic_region|CHR_START-Zm00001eb404740|||n.10791A>G||||||\tGT:PL\t1/1:33,54,0\n",
      "chr10\t10811\t.\tC\tA\t4.38466\t.\tDP=16;VDB=0.0249134;SGB=-0.689466;MQSB=0.0878622;MQ0F=0.125;AC=2;AN=2;DP4=0,0,8,8;MQ=2;ANN=A|intergenic_region|MODIFIER|CHR_START-Zm00001eb404740|CHR_START-Zm00001eb404740|intergenic_region|CHR_START-Zm00001eb404740|||n.10811C>A||||||\tGT:PL\t1/1:32,48,0\n",
      "chr10\t11932\t.\tA\tG\t6.03513\t.\tDP=34;VDB=0.0644318;SGB=-0.693021;RPB=0.367879;MQB=0.23222;MQSB=0.847353;BQB=0.99308;MQ0F=0.0294118;AC=2;AN=2;DP4=4,0,10,17;MQ=4;ANN=G|intergenic_region|MODIFIER|CHR_START-Zm00001eb404740|CHR_START-Zm00001eb404740|intergenic_region|CHR_START-Zm00001eb404740|||n.11932A>G||||||\tGT:PL\t1/1:32,48,0\n",
      "chr10\t13709\t.\tG\tA\t5.75677\t.\tDP=6;VDB=0.226058;SGB=-0.616816;MQSB=1;MQ0F=0;AC=2;AN=2;DP4=0,0,1,5;MQ=8;ANN=A|intergenic_region|MODIFIER|CHR_START-Zm00001eb404740|CHR_START-Zm00001eb404740|intergenic_region|CHR_START-Zm00001eb404740|||n.13709G>A||||||\tGT:PL\t1/1:34,18,0\n",
      "chr10\t14532\t.\tC\tT\t134.416\t.\tDP=10;VDB=0.120448;SGB=-0.670168;MQSB=0.0559863;MQ0F=0;AC=2;AN=2;DP4=0,0,4,6;MQ=23;ANN=T|intergenic_region|MODIFIER|CHR_START-Zm00001eb404740|CHR_START-Zm00001eb404740|intergenic_region|CHR_START-Zm00001eb404740|||n.14532C>T||||||\tGT:PL\t1/1:164,30,0\n",
      "chr10\t14623\t.\tG\tA\t94.4151\t.\tDP=6;VDB=0.0411825;SGB=-0.616816;MQSB=0.28717;MQ0F=0;AC=2;AN=2;DP4=0,0,3,3;MQ=24;ANN=A|intergenic_region|MODIFIER|CHR_START-Zm00001eb404740|CHR_START-Zm00001eb404740|intergenic_region|CHR_START-Zm00001eb404740|||n.14623G>A||||||\tGT:PL\t1/1:124,18,0\n",
      "chr10\t14886\t.\tG\tA\t79.2181\t.\tDP=13;VDB=0.0117892;SGB=-0.662043;RPB=0.222222;MQB=0;MQSB=1;BQB=0.611111;MQ0F=0;AC=2;AN=2;DP4=2,0,8,1;MQ=18;ANN=A|intergenic_region|MODIFIER|CHR_START-Zm00001eb404740|CHR_START-Zm00001eb404740|intergenic_region|CHR_START-Zm00001eb404740|||n.14886G>A||||||\tGT:PL\t1/1:106,15,0\n",
      "chr10\t15025\t.\tG\tA\t49.2219\t.\tDP=8;VDB=0.98214;SGB=-0.616816;RPB=1;MQB=1;MQSB=1;BQB=1;MQ0F=0;AC=2;AN=2;DP4=0,1,1,5;MQ=16;ANN=A|intergenic_region|MODIFIER|CHR_START-Zm00001eb404740|CHR_START-Zm00001eb404740|intergenic_region|CHR_START-Zm00001eb404740|||n.15025G>A||||||\tGT:PL\t1/1:76,10,0\n",
      "chr10\t15258\t.\tT\tC\t8.39671\t.\tDP=19;VDB=0.0684004;SGB=-0.636426;RPB=0.941512;MQB=0.322486;MQSB=0.778801;BQB=0.269146;MQ0F=0.368421;AC=2;AN=2;DP4=5,3,4,3;MQ=6;ANN=C|intergenic_region|MODIFIER|CHR_START-Zm00001eb404740|CHR_START-Zm00001eb404740|intergenic_region|CHR_START-Zm00001eb404740|||n.15258T>C||||||\tGT:PL\t1/1:35,3,0\n"
     ]
    }
   ],
   "source": [
    "awk '{if($1==\"#CHROM\") print $0}' ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022.ann.vcf  \n",
    "\n",
    "awk '{if($1==\"chr10\") print $0}' ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022.ann.vcf   |  head -10\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "20437ec4",
   "metadata": {},
   "outputs": [],
   "source": [
    " #Filter to just the variants on chromosome 10 \n",
    "java -jar /Users/aricha14/Documents/Tools/snpEff/SnpSift.jar filter \"(CHROM == 'chr10')\" ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022.ann.vcf    > ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022_snpeff_chr10.ann.vcf \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "c62755f3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\t/Volumes/Elements/Sequence_Data/WGS_data/Trf_mutant_WGS_2022/WGS_Alignment_NAM5_May2022/BAM_reads/TrfM6_aligned_NAM5.bam\n",
      "chr10\t13496880\t.\tTAACAC\tTAACACCACACCCACGATCAAACAC\t117.415\t.\tINDEL;IDV=26;IMF=0.8125;DP=32;VDB=0.731354;SGB=-0.693097;MQSB=0.336578;MQ0F=0.125;AC=2;AN=2;DP4=0,0,16,14;MQ=11;ANN=TAACACCACACCCACGATCAAACAC|downstream_gene_variant|MODIFIER|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding||c.*4279_*4280insGTGTTTGATCGTGGGTGTG|||||3982|,TAACACCACACCCACGATCAAACAC|intron_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding|3/4|c.694-218_694-217insCACACCCACGATCAAACAC||||||\tGT:PL\t1/1:147,90,0\n",
      "chr10\t13496940\t.\tA\tC\t200.416\t.\tDP=31;VDB=0.46276;SGB=-0.692717;MQSB=0.0326821;MQ0F=0.0645161;AC=2;AN=2;DP4=0,0,11,12;MQ=20;ANN=C|downstream_gene_variant|MODIFIER|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding||c.*4225T>G|||||3928|,C|intron_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding|3/4|c.694-163A>C||||||\tGT:PL\t1/1:230,69,0\n",
      "chr10\t13496952\t.\tG\tT\t225.417\t.\tDP=32;VDB=0.749015;SGB=-0.692976;MQSB=0.00789441;MQ0F=0.0625;AC=2;AN=2;DP4=0,0,14,12;MQ=23;ANN=T|downstream_gene_variant|MODIFIER|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding||c.*4213C>A|||||3916|,T|intron_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding|3/4|c.694-151G>T||||||\tGT:PL\t1/1:255,78,0\n",
      "chr10\t13496978\t.\tC\tA\t225.417\t.\tDP=33;VDB=0.943304;SGB=-0.693021;MQSB=0.0762218;MQ0F=0;AC=2;AN=2;DP4=0,0,15,12;MQ=26;ANN=A|downstream_gene_variant|MODIFIER|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding||c.*4187G>T|||||3890|,A|intron_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding|3/4|c.694-125C>A||||||\tGT:PL\t1/1:255,81,0\n",
      "chr10\t13497236\t.\tATTTTTTTTT\tATTTTTTT\t156.416\t.\tINDEL;IDV=18;IMF=0.9;DP=20;VDB=0.0820542;SGB=-0.690438;MQSB=0.762744;MQ0F=0;AC=2;AN=2;DP4=0,0,4,13;MQ=41;ANN=ATTTTTTT|downstream_gene_variant|MODIFIER|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding||c.*3920_*3921delAA|||||3624|,ATTTTTTT|intron_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding|4/4|c.822+13_822+14delTT||||||\tGT:PL\t1/1:186,51,0\n",
      "chr10\t13497396\t.\tT\tC\t225.417\t.\tDP=24;VDB=0.933383;SGB=-0.691153;MQSB=0.0982563;MQ0F=0;AC=2;AN=2;DP4=0,0,12,6;MQ=36;ANN=C|downstream_gene_variant|MODIFIER|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding||c.*3769A>G|||||3472|,C|intron_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding|4/4|c.822+165T>C||||||\tGT:PL\t1/1:255,54,0\n",
      "chr10\t13497405\t.\tT\tA\t225.417\t.\tDP=26;VDB=0.970247;SGB=-0.691153;MQSB=0.0982563;MQ0F=0;AC=2;AN=2;DP4=0,0,12,6;MQ=36;ANN=A|downstream_gene_variant|MODIFIER|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding||c.*3760A>T|||||3463|,A|intron_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding|4/4|c.822+174T>A||||||\tGT:PL\t1/1:255,54,0\n",
      "chr10\t13497411\t.\tC\tT\t225.417\t.\tDP=27;VDB=0.972572;SGB=-0.692352;MQSB=0.00840937;MQ0F=0;AC=2;AN=2;DP4=0,0,12,9;MQ=36;ANN=T|downstream_gene_variant|MODIFIER|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding||c.*3754G>A|||||3457|,T|intron_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding|4/4|c.822+180C>T||||||\tGT:PL\t1/1:255,63,0\n",
      "chr10\t13497446\t.\tA\tC\t225.417\t.\tDP=23;VDB=0.720365;SGB=-0.692352;MQSB=0.164074;MQ0F=0;AC=2;AN=2;DP4=0,0,10,11;MQ=36;ANN=C|downstream_gene_variant|MODIFIER|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding||c.*3719T>G|||||3422|,C|intron_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding|4/4|c.822+215A>C||||||\tGT:PL\t1/1:255,63,0\n",
      "chr10\t13497459\t.\tG\tA\t225.417\t.\tDP=21;VDB=0.349823;SGB=-0.692067;MQSB=0.396248;MQ0F=0;AC=2;AN=2;DP4=0,0,10,10;MQ=36;ANN=A|downstream_gene_variant|MODIFIER|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding||c.*3706C>T|||||3409|,A|intron_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding|4/4|c.822+228G>A||||||\tGT:PL\t1/1:255,60,0\n"
     ]
    }
   ],
   "source": [
    "# Then filter for the intreval region 13.5Mb to 14.0Mb on chr1 \n",
    "java -jar /Users/aricha14/Documents/Tools/snpEff/SnpSift.jar filter \"(CHROM == 'chr10') && (POS >= 13496844) && (POS < 14026688)\" ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022.ann.vcf  > ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3.ann.vcf\n",
    "\n",
    "awk '{if($1==\"#CHROM\") print $0}' ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3.ann.vcf\n",
    "\n",
    "awk '{if($1==\"chr10\") print $0}' ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3.ann.vcf  |  head -10\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "79ac2a50",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\t/Volumes/Elements/Sequence_Data/WGS_data/Trf_mutant_WGS_2022/WGS_Alignment_NAM5_May2022/BAM_reads/TrfM6_aligned_NAM5.bam\n",
      "chr10\t13498464\t.\tA\tT\t225.417\t.\tDP=17;VDB=0.79373;SGB=-0.686358;MQSB=0.927041;MQ0F=0;AC=2;AN=2;DP4=0,0,3,11;MQ=40;ANN=T|missense_variant|MODERATE|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding|5/5|c.876A>T|p.Glu292Asp|1013/1527|876/954|292/317||,T|downstream_gene_variant|MODIFIER|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding||c.*2701T>A|||||2404|\tGT:PL\t1/1:255,42,0\n",
      "chr10\t13498471\t.\tA\tC\t225.417\t.\tDP=17;VDB=0.960592;SGB=-0.689466;MQSB=0.607143;MQ0F=0;AC=2;AN=2;DP4=0,0,2,14;MQ=40;ANN=C|missense_variant|MODERATE|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding|5/5|c.883A>C|p.Met295Leu|1020/1527|883/954|295/317||,C|downstream_gene_variant|MODIFIER|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding||c.*2694T>G|||||2397|\tGT:PL\t1/1:255,48,0\n",
      "chr10\t13501412\t.\tC\tT\t74.9597\t.\tDP=29;VDB=0.415179;SGB=-0.680642;RPB=0.941765;MQB=0.000703108;MQSB=0.20662;BQB=0.430095;MQ0F=0.0689655;AC=1;AN=2;DP4=9,3,7,5;MQ=30;ANN=T|missense_variant|MODERATE|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding|2/2|c.992G>A|p.Arg331Lys|1270/1814|992/1239|331/412||,T|downstream_gene_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding||c.*2870C>T|||||2434|\tGT:PL\t0/1:108,0,231\n",
      "chr10\t13501546\t.\tGAGCTGCAGCTGCAGCTGCAGCTGC\tGAGCTGCAGCTGC\t106.321\t.\tINDEL;IDV=12;IMF=0.521739;DP=23;VDB=0.809172;SGB=-0.680642;MQSB=0.977933;MQ0F=0;AC=1;AN=2;DP4=4,5,6,6;MQ=31;ANN=GAGCTGCAGCTGC|conservative_inframe_deletion|MODERATE|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding|2/2|c.834_845delGCAGCTGCAGCT|p.Gln279_Leu282del|1123/1814|834/1239|278/412||,GAGCTGCAGCTGC|downstream_gene_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding||c.*3017_*3028delAGCTGCAGCTGC|||||2581|\tGT:PL\t0/1:139,0,231\n",
      "chr10\t13501725\t.\tG\tT\t104.295\t.\tDP=15;VDB=0.816978;SGB=-0.662043;RPB=0.486752;MQB=0.0657285;MQSB=0.0657285;BQB=0.980199;MQ0F=0;AC=1;AN=2;DP4=3,2,2,7;MQ=34;ANN=T|missense_variant|MODERATE|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding|2/2|c.679C>A|p.Leu227Met|957/1814|679/1239|227/412||,T|downstream_gene_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding||c.*3183G>T|||||2747|\tGT:PL\t0/1:137,0,117\n",
      "chr10\t13501946\t.\tC\tT\t221.973\t.\tDP=24;VDB=0.529691;SGB=-0.686358;RPB=0.945204;MQB=1;MQSB=1;BQB=0.293091;MQ0F=0;AC=1;AN=2;DP4=3,4,9,5;MQ=42;ANN=T|missense_variant|MODERATE|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding|2/2|c.458G>A|p.Cys153Tyr|736/1814|458/1239|153/412||,T|downstream_gene_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding||c.*3404C>T|||||2968|\tGT:PL\t0/1:255,0,166\n",
      "chr10\t13502011\t.\tC\tG\t131.277\t.\tDP=20;VDB=0.802329;SGB=-0.636426;RPB=0.679965;MQB=0.887766;MQSB=0.943335;BQB=0.621145;MQ0F=0;AC=1;AN=2;DP4=5,5,5,2;MQ=41;ANN=G|missense_variant|MODERATE|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding|2/2|c.393G>C|p.Glu131Asp|671/1814|393/1239|131/412||,G|downstream_gene_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding||c.*3469C>G|||||3033|\tGT:PL\t0/1:164,0,224\n",
      "chr10\t13503120\t.\tC\tG\t39.5694\t.\tDP=13;VDB=0.0846253;SGB=-0.556411;RPB=0.826565;MQB=0.236815;MQSB=0.0209031;BQB=0.953497;MQ0F=0;AC=1;AN=2;DP4=7,2,1,3;MQ=35;ANN=G|missense_variant|MODERATE|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding|1/2|c.278G>C|p.Gly93Ala|556/1814|278/1239|93/412||,G|downstream_gene_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding||c.*4578C>G|||||4142|\tGT:PL\t0/1:73,0,168\n",
      "chr10\t13503206\t.\tGGCAGCAGCAG\tGGCAGCAG\t15.7038\t.\tINDEL;IDV=3;IMF=0.2;DP=15;VDB=0.112589;SGB=-0.453602;MQSB=0.924584;MQ0F=0;AC=1;AN=2;DP4=2,5,1,1;MQ=41;ANN=GGCAGCAG|conservative_inframe_deletion|MODERATE|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding|1/2|c.182_184delCTG|p.Ala62del|462/1814|182/1239|61/412||,GGCAGCAG|downstream_gene_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding||c.*4672_*4674delCAG|||||4236|\tGT:PL\t0/1:50,0,204\n",
      "chr10\t13503212\t.\tAG\tAGAGGAGG\t47.4795\t.\tINDEL;IDV=3;IMF=0.2;DP=15;VDB=0.0579073;SGB=-0.511536;MQSB=0.643095;MQ0F=0;AC=1;AN=2;DP4=2,6,2,1;MQ=41;ANN=AGAGGAGG|conservative_inframe_insertion|MODERATE|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding|1/2|c.184_185insCCTCCT|p.Ser61_Ala62dup|462/1814|184/1239|62/412||,AGAGGAGG|downstream_gene_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding||c.*4671_*4672insAGGAGG|||||4236|\tGT:PL\t0/1:81,0,203\n"
     ]
    }
   ],
   "source": [
    "#Use SNPSift to filter for high impact variants, in the AD2 to AD3 interval\n",
    "\n",
    "java -jar /Users/aricha14/Documents/Tools/snpEff/SnpSift.jar filter \"ANN[*].IMPACT has 'HIGH' | ANN[*].IMPACT has 'MODERATE'\" ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3.ann.vcf  > ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3_HIGHorMOD.vcf\n",
    "\n",
    "awk '{if($1==\"#CHROM\") print $0}' ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3_HIGHorMOD.vcf\n",
    "\n",
    "awk '{if($1==\"chr10\") print $0}' ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3_HIGHorMOD.vcf  |  head -10\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "f05110af",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Repeat the process with the N pool so that we can compare which ones are present/ absent \n",
    "java -Xmx4g -jar /Users/aricha14/Documents/Tools/snpEff/snpEff.jar Zea_mays_B73_NAM.5.0 /Volumes/Elements/Sequence_Data/WGS_data/Trf_mutant_WGS_2022/TrfN5_WGS_NAM5_mpileup_Dec2022.vcf > /Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022/snpeff/TrfN5_WGS_NAM5_mpileup_Dec2022.ann.vcf\n",
    "\n",
    " #Filter to just the variants on chromosome 10 \n",
    "java -jar /Users/aricha14/Documents/Tools/snpEff/SnpSift.jar filter \"(CHROM == 'chr10')\" ./snpeff/TrfN5_WGS_NAM5_mpileup_Dec2022.ann.vcf    > ./snpeff/TrfN5_WGS_NAM5_mpileup_Dec2022_snpeff_chr10.ann.vcf \n",
    "\n",
    "# Then filter for the interval region 110Mb to 165Mb on chr1 \n",
    "java -jar /Users/aricha14/Documents/Tools/snpEff/SnpSift.jar filter \"(CHROM == 'chr10') && (POS >= 13496844) && (POS < 14026688)\" ./snpeff/TrfN5_WGS_NAM5_mpileup_Dec2022.ann.vcf  > ./snpeff/TrfN5_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3.ann.vcf\n",
    "\n",
    "#Use SNPSift to filter for high impact variants, in the AD2 to AD3 interval\n",
    "\n",
    "java -jar /Users/aricha14/Documents/Tools/snpEff/SnpSift.jar filter \"ANN[*].IMPACT has 'HIGH' | ANN[*].IMPACT has 'MODERATE'\" ./snpeff/TrfN5_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3.ann.vcf  > ./snpeff/TrfN5_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3_HIGHorMOD.vcf\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "ff1a721f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "659e5cae",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Finally filter the high/ moderate effect variant list to just EMS type\n",
    "\n",
    "#Use SNPSift to filter for high impact variants, in the AD2 to AD3 interval\n",
    "\n",
    "java -jar /Users/aricha14/Documents/Tools/snpEff/SnpSift.jar filter \"((REF == 'C') | (REF == 'G')) && ((ALT == 'T')| (ALT == 'A'))\" ./snpeff/TrfN5_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3_HIGHorMOD.vcf  > ./snpeff/TrfN5_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3_HIGHorMOD_ems.vcf\n",
    "\n",
    "java -jar /Users/aricha14/Documents/Tools/snpEff/SnpSift.jar filter \"((REF == 'C') | (REF == 'G')) && ((ALT == 'T')| (ALT == 'A'))\" ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3_HIGHorMOD.vcf  > ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3_HIGHorMOD_ems.vcf\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "aa1effa9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\t/Volumes/Elements/Sequence_Data/WGS_data/Trf_mutant_WGS_2022/WGS_Alignment_NAM5_May2022/BAM_reads/TrfM6_aligned_NAM5.bam\n",
      "chr10\t13501412\t.\tC\tT\t74.9597\t.\tDP=29;VDB=0.415179;SGB=-0.680642;RPB=0.941765;MQB=0.000703108;MQSB=0.20662;BQB=0.430095;MQ0F=0.0689655;AC=1;AN=2;DP4=9,3,7,5;MQ=30;ANN=T|missense_variant|MODERATE|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding|2/2|c.992G>A|p.Arg331Lys|1270/1814|992/1239|331/412||,T|downstream_gene_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding||c.*2870C>T|||||2434|\tGT:PL\t0/1:108,0,231\n",
      "chr10\t13501725\t.\tG\tT\t104.295\t.\tDP=15;VDB=0.816978;SGB=-0.662043;RPB=0.486752;MQB=0.0657285;MQSB=0.0657285;BQB=0.980199;MQ0F=0;AC=1;AN=2;DP4=3,2,2,7;MQ=34;ANN=T|missense_variant|MODERATE|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding|2/2|c.679C>A|p.Leu227Met|957/1814|679/1239|227/412||,T|downstream_gene_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding||c.*3183G>T|||||2747|\tGT:PL\t0/1:137,0,117\n",
      "chr10\t13501946\t.\tC\tT\t221.973\t.\tDP=24;VDB=0.529691;SGB=-0.686358;RPB=0.945204;MQB=1;MQSB=1;BQB=0.293091;MQ0F=0;AC=1;AN=2;DP4=3,4,9,5;MQ=42;ANN=T|missense_variant|MODERATE|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding|2/2|c.458G>A|p.Cys153Tyr|736/1814|458/1239|153/412||,T|downstream_gene_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding||c.*3404C>T|||||2968|\tGT:PL\t0/1:255,0,166\n",
      "chr10\t13503214\t.\tC\tA\t162.416\t.\tDP=18;VDB=0.311328;SGB=-0.636426;MQSB=1;MQ0F=0;AC=2;AN=2;DP4=0,0,2,5;MQ=42;ANN=A|missense_variant|MODERATE|Zm00001eb408750|Zm00001eb408750|transcript|Zm00001eb408750_T001|protein_coding|1/2|c.184G>T|p.Ala62Ser|462/1814|184/1239|62/412||,A|downstream_gene_variant|MODIFIER|Zm00001eb408730|Zm00001eb408730|transcript|Zm00001eb408730_T001|protein_coding||c.*4672C>A|||||4236|\tGT:PL\t1/1:192,21,0\n",
      "chr10\t13676774\t.\tG\tA\t33.0854\t.\tDP=14;VDB=0.807564;SGB=-0.511536;RPB=0.699673;MQB=0.447727;MQSB=0.623736;BQB=0.968368;MQ0F=0;AC=1;AN=2;DP4=6,4,1,2;MQ=38;ANN=A|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T005|protein_coding|8/15|c.839G>A|p.Ser280Asn|1108/3099|839/2529|280/842||,A|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T002|protein_coding|8/14|c.842G>A|p.Ser281Asn|1111/3193|842/2442|281/813||,A|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T003|protein_coding|8/15|c.839G>A|p.Ser280Asn|1108/3095|839/2478|280/825||,A|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T004|protein_coding|8/15|c.842G>A|p.Ser281Asn|1111/3098|842/2481|281/826||,A|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T006|protein_coding|8/15|c.842G>A|p.Ser281Asn|1111/3098|842/2481|281/826||,A|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T001|protein_coding|8/15|c.842G>A|p.Ser281Asn|1111/3137|842/2442|281/813||\tGT:PL\t0/1:67,0,209\n",
      "chr10\t13678064\t.\tC\tT\t220.27\t.\tDP=26;VDB=0.989925;SGB=-0.683931;RPB=0.428345;MQB=0.955854;MQSB=0.947103;BQB=0.834768;MQ0F=0;AC=1;AN=2;DP4=6,3,4,9;MQ=41;ANN=T|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T005|protein_coding|12/15|c.1673C>T|p.Thr558Ile|1942/3099|1673/2529|558/842||,T|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T002|protein_coding|12/14|c.1676C>T|p.Thr559Ile|1945/3193|1676/2442|559/813||,T|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T003|protein_coding|12/15|c.1673C>T|p.Thr558Ile|1942/3095|1673/2478|558/825||,T|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T004|protein_coding|12/15|c.1676C>T|p.Thr559Ile|1945/3098|1676/2481|559/826||,T|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T006|protein_coding|12/15|c.1676C>T|p.Thr559Ile|1945/3098|1676/2481|559/826||,T|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T001|protein_coding|12/15|c.1676C>T|p.Thr559Ile|1945/3137|1676/2442|559/813||\tGT:PL\t0/1:253,0,187\n",
      "chr10\t13679068\t.\tG\tA\t194.607\t.\tDP=20;VDB=0.913441;SGB=-0.680642;RPB=0.85394;MQB=0.302983;MQSB=0.938795;BQB=0.875749;MQ0F=0;AC=1;AN=2;DP4=5,1,3,9;MQ=37;ANN=A|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T005|protein_coding|13/15|c.2098G>A|p.Ala700Thr|2367/3099|2098/2529|700/842||,A|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T002|protein_coding|13/14|c.2101G>A|p.Ala701Thr|2370/3193|2101/2442|701/813||,A|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T003|protein_coding|13/15|c.2098G>A|p.Ala700Thr|2367/3095|2098/2478|700/825||,A|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T004|protein_coding|13/15|c.2101G>A|p.Ala701Thr|2370/3098|2101/2481|701/826||,A|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T006|protein_coding|13/15|c.2101G>A|p.Ala701Thr|2370/3098|2101/2481|701/826||,A|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T001|protein_coding|13/15|c.2101G>A|p.Ala701Thr|2370/3137|2101/2442|701/813||\tGT:PL\t0/1:228,0,96\n",
      "chr10\t13679378\t.\tG\tT\t225.417\t.\tDP=16;VDB=0.390925;SGB=-0.683931;MQSB=0.826565;MQ0F=0;AC=2;AN=2;DP4=0,0,4,9;MQ=41;ANN=T|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T005|protein_coding|14/15|c.2317G>T|p.Ala773Ser|2586/3099|2317/2529|773/842||,T|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T002|protein_coding|14/14|c.2320G>T|p.Ala774Ser|2589/3193|2320/2442|774/813||,T|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T003|protein_coding|14/15|c.2317G>T|p.Ala773Ser|2586/3095|2317/2478|773/825||,T|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T004|protein_coding|14/15|c.2320G>T|p.Ala774Ser|2589/3098|2320/2481|774/826||,T|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T006|protein_coding|14/15|c.2320G>T|p.Ala774Ser|2589/3098|2320/2481|774/826||,T|missense_variant|MODERATE|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T001|protein_coding|14/15|c.2320G>T|p.Ala774Ser|2589/3137|2320/2442|774/813||\tGT:PL\t1/1:255,39,0\n",
      "chr10\t13679503\t.\tC\tT\t215.417\t.\tDP=11;VDB=0.2216;SGB=-0.676189;MQSB=0.829029;MQ0F=0;AC=2;AN=2;DP4=0,0,8,3;MQ=40;ANN=T|splice_donor_variant&intron_variant|HIGH|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T001|protein_coding|14/14|c.*1+2C>T||||||,T|splice_region_variant&intron_variant|LOW|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T005|protein_coding|14/14|c.2435+7C>T||||||,T|3_prime_UTR_variant|MODIFIER|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T002|protein_coding|14/14|c.*3C>T|||||3|,T|intron_variant|MODIFIER|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T003|protein_coding|14/14|c.2431+11C>T||||||,T|intron_variant|MODIFIER|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T004|protein_coding|14/14|c.2434+11C>T||||||,T|intron_variant|MODIFIER|Zm00001eb408800|Zm00001eb408800|transcript|Zm00001eb408800_T006|protein_coding|14/14|c.2434+11C>T||||||;LOF=(Zm00001eb408800|Zm00001eb408800|6|0.17)\tGT:PL\t1/1:245,33,0\n",
      "chr10\t13953240\t.\tC\tA\t9.07957\t.\tDP=18;VDB=0.319981;SGB=-0.556411;RPB=0.140858;MQB=0.99005;MQSB=0.486752;BQB=0.612626;MQ0F=0;AC=1;AN=2;DP4=4,6,1,3;MQ=23;ANN=A|missense_variant|MODERATE|Zm00001eb408820|Zm00001eb408820|transcript|Zm00001eb408820_T001|protein_coding|1/3|c.297C>A|p.Asp99Glu|776/2354|297/906|99/301||\tGT:PL\t0/1:42,0,135\n",
      "chr10\t13953392\t.\tC\tT\t122.415\t.\tDP=23;VDB=0.370777;SGB=-0.692067;MQSB=0.825863;MQ0F=0.130435;AC=2;AN=2;DP4=0,0,14,6;MQ=13;ANN=T|missense_variant|MODERATE|Zm00001eb408820|Zm00001eb408820|transcript|Zm00001eb408820_T001|protein_coding|1/3|c.449C>T|p.Ala150Val|928/2354|449/906|150/301||\tGT:PL\t1/1:152,60,0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr10\t13954038\t.\tG\tT\t182.416\t.\tDP=20;VDB=0.0870779;SGB=-0.689466;MQSB=0.0390604;MQ0F=0;AC=2;AN=2;DP4=0,0,12,4;MQ=23;ANN=T|missense_variant|MODERATE|Zm00001eb408820|Zm00001eb408820|transcript|Zm00001eb408820_T001|protein_coding|3/3|c.767G>T|p.Cys256Phe|1246/2354|767/906|256/301||\tGT:PL\t1/1:212,48,0\n"
     ]
    }
   ],
   "source": [
    "awk '{if($1==\"#CHROM\") print $0}' ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3_HIGHorMOD_ems.vcf\n",
    "awk '{if($1==\"chr10\") print $0}' ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3_HIGHorMOD_ems.vcf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "b47f47ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "#Use SNPSift to filter for high impact variants, in the AD2 to AD3 interval\n",
    "\n",
    "java -jar /Users/aricha14/Documents/Tools/snpEff/SnpSift.jar filter \"ANN[*].IMPACT has 'HIGH'\" ./snpeff/TrfN5_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3_HIGHorMOD_ems.vcf > ./snpeff/TrfN5_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3_HIGH_ems.vcf\n",
    "\n",
    "java -jar /Users/aricha14/Documents/Tools/snpEff/SnpSift.jar filter \"ANN[*].IMPACT has 'HIGH'\" ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3_HIGHorMOD_ems.vcf > ./snpeff/TrfM6_WGS_NAM5_mpileup_Dec2022_snpeff_chr10_AD2toAD3_HIGH_ems.vcf\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
