{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0/1 -  0.00% of files done   0.000/0.144 MB -  0.00% of file sizes done\n",
      "Processing WGS_May2019_LG2.ipynb - 0.144 MB   2019-06-05.12:18:11\n",
      "0/1 -  0.00% of files done   0.000/0.196 MB -  0.00% of file sizes done\n",
      "Processing WGS_HojaI.ipynb - 0.196 MB   2019-06-05.12:18:13\n",
      "0/5 -  0.00% of files done   0.000/26231.851 MB -  0.00% of file sizes done\n",
      "Processing MD5.txt - 0.000 MB   2019-06-05.12:18:16\n",
      "1/5 - 20.00% of files done   0.000/26231.851 MB -  0.00% of file sizes done\n",
      "Processing TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 10571.418 MB   2019-06-05.12:18:17\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 199.995/10571.418 MB -  1.89% done   2019-06-05.12:18:23\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 399.991/10571.418 MB -  3.78% done   2019-06-05.12:18:24\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 639.985/10571.418 MB -  6.05% done   2019-06-05.12:18:26\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 799.982/10571.418 MB -  7.57% done   2019-06-05.12:18:27\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 999.977/10571.418 MB -  9.46% done   2019-06-05.12:18:28\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 1199.973/10571.418 MB - 11.35% done   2019-06-05.12:18:30\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 1399.968/10571.418 MB - 13.24% done   2019-06-05.12:18:31\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 1559.964/10571.418 MB - 14.76% done   2019-06-05.12:18:34\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 1719.961/10571.418 MB - 16.27% done   2019-06-05.12:18:35\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 1879.957/10571.418 MB - 17.78% done   2019-06-05.12:18:36\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 2039.953/10571.418 MB - 19.30% done   2019-06-05.12:18:37\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 2199.950/10571.418 MB - 20.81% done   2019-06-05.12:18:38\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 2359.946/10571.418 MB - 22.32% done   2019-06-05.12:18:39\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 2559.941/10571.418 MB - 24.22% done   2019-06-05.12:18:41\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 2839.935/10571.418 MB - 26.86% done   2019-06-05.12:18:43\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 3039.930/10571.418 MB - 28.76% done   2019-06-05.12:18:45\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 3199.927/10571.418 MB - 30.27% done   2019-06-05.12:18:47\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 3439.921/10571.418 MB - 32.54% done   2019-06-05.12:18:48\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 3519.919/10571.418 MB - 33.30% done   2019-06-05.12:18:49\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 3679.916/10571.418 MB - 34.81% done   2019-06-05.12:18:50\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 3879.911/10571.418 MB - 36.70% done   2019-06-05.12:18:51\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 4039.908/10571.418 MB - 38.22% done   2019-06-05.12:18:53\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 4399.899/10571.418 MB - 41.62% done   2019-06-05.12:18:55\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 4599.895/10571.418 MB - 43.51% done   2019-06-05.12:18:57\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 4759.891/10571.418 MB - 45.03% done   2019-06-05.12:18:59\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 4879.888/10571.418 MB - 46.16% done   2019-06-05.12:19:00\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 5039.885/10571.418 MB - 47.67% done   2019-06-05.12:19:01\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 5199.881/10571.418 MB - 49.19% done   2019-06-05.12:19:02\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 5399.876/10571.418 MB - 51.08% done   2019-06-05.12:19:03\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 5519.874/10571.418 MB - 52.22% done   2019-06-05.12:19:06\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 5679.870/10571.418 MB - 53.73% done   2019-06-05.12:19:07\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 5879.865/10571.418 MB - 55.62% done   2019-06-05.12:19:09\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 6039.862/10571.418 MB - 57.13% done   2019-06-05.12:19:10\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 6239.857/10571.418 MB - 59.03% done   2019-06-05.12:19:12\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 6319.855/10571.418 MB - 59.78% done   2019-06-05.12:19:14\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 6479.852/10571.418 MB - 61.30% done   2019-06-05.12:19:15\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 6679.847/10571.418 MB - 63.19% done   2019-06-05.12:19:17\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 6959.841/10571.418 MB - 65.84% done   2019-06-05.12:19:19\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 7119.837/10571.418 MB - 67.35% done   2019-06-05.12:19:20\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 7239.834/10571.418 MB - 68.48% done   2019-06-05.12:19:22\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 7399.831/10571.418 MB - 70.00% done   2019-06-05.12:19:23\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 7559.827/10571.418 MB - 71.51% done   2019-06-05.12:19:24\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 7719.823/10571.418 MB - 73.03% done   2019-06-05.12:19:25\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 7879.820/10571.418 MB - 74.54% done   2019-06-05.12:19:26\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 7919.819/10571.418 MB - 74.92% done   2019-06-05.12:19:27\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 8039.816/10571.418 MB - 76.05% done   2019-06-05.12:19:28\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 8159.813/10571.418 MB - 77.19% done   2019-06-05.12:19:29\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 8319.810/10571.418 MB - 78.70% done   2019-06-05.12:19:30\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 8479.806/10571.418 MB - 80.21% done   2019-06-05.12:19:31\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 8639.802/10571.418 MB - 81.73% done   2019-06-05.12:19:32\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 8759.800/10571.418 MB - 82.86% done   2019-06-05.12:19:33\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 8959.795/10571.418 MB - 84.75% done   2019-06-05.12:19:35\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 9119.791/10571.418 MB - 86.27% done   2019-06-05.12:19:36\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 9319.787/10571.418 MB - 88.16% done   2019-06-05.12:19:39\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 9399.785/10571.418 MB - 88.92% done   2019-06-05.12:19:40\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 9639.779/10571.418 MB - 91.19% done   2019-06-05.12:19:41\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 9799.776/10571.418 MB - 92.70% done   2019-06-05.12:19:42\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 9962.687/10571.418 MB - 94.24% done   2019-06-05.12:19:43\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 10085.599/10571.418 MB - 95.40% done   2019-06-05.12:19:44\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 10125.598/10571.418 MB - 95.78% done   2019-06-05.12:19:44\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 10205.596/10571.418 MB - 96.54% done   2019-06-05.12:19:45\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 10285.595/10571.418 MB - 97.30% done   2019-06-05.12:19:46\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 10365.593/10571.418 MB - 98.05% done   2019-06-05.12:19:47\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 10445.591/10571.418 MB - 98.81% done   2019-06-05.12:19:48\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 10528.504/10571.418 MB - 99.59% done   2019-06-05.12:19:49\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 10568.503/10571.418 MB - 99.97% done   2019-06-05.12:19:49\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_1.fq.gz - 10571.418/10571.418 MB - 100.00% done   2019-06-05.12:19:49\n",
      "2/5 - 40.00% of files done   10571.418/26231.851 MB - 40.30% of file sizes done\n",
      "Processing TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 10931.675 MB   2019-06-05.12:20:11\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 39.999/10931.675 MB -  0.37% done   2019-06-05.12:20:13\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 119.997/10931.675 MB -  1.10% done   2019-06-05.12:20:14\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 279.994/10931.675 MB -  2.56% done   2019-06-05.12:20:15\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 399.991/10931.675 MB -  3.66% done   2019-06-05.12:20:16\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 519.988/10931.675 MB -  4.76% done   2019-06-05.12:20:16\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 639.985/10931.675 MB -  5.85% done   2019-06-05.12:20:17\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 799.982/10931.675 MB -  7.32% done   2019-06-05.12:20:18\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 919.979/10931.675 MB -  8.42% done   2019-06-05.12:20:19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 1039.976/10931.675 MB -  9.51% done   2019-06-05.12:20:20\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 1199.973/10931.675 MB - 10.98% done   2019-06-05.12:20:21\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 1279.971/10931.675 MB - 11.71% done   2019-06-05.12:20:21\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 1439.967/10931.675 MB - 13.17% done   2019-06-05.12:20:22\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 1599.963/10931.675 MB - 14.64% done   2019-06-05.12:20:23\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 1719.961/10931.675 MB - 15.73% done   2019-06-05.12:20:24\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 1879.957/10931.675 MB - 17.20% done   2019-06-05.12:20:25\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 1999.954/10931.675 MB - 18.30% done   2019-06-05.12:20:26\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 2159.951/10931.675 MB - 19.76% done   2019-06-05.12:20:27\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 2279.948/10931.675 MB - 20.86% done   2019-06-05.12:20:28\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 2439.944/10931.675 MB - 22.32% done   2019-06-05.12:20:29\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 2599.940/10931.675 MB - 23.78% done   2019-06-05.12:20:30\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 2759.937/10931.675 MB - 25.25% done   2019-06-05.12:20:31\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 2879.934/10931.675 MB - 26.34% done   2019-06-05.12:20:32\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 2999.931/10931.675 MB - 27.44% done   2019-06-05.12:20:33\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 3159.928/10931.675 MB - 28.91% done   2019-06-05.12:20:34\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 3319.924/10931.675 MB - 30.37% done   2019-06-05.12:20:35\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 3519.919/10931.675 MB - 32.20% done   2019-06-05.12:20:36\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 3679.916/10931.675 MB - 33.66% done   2019-06-05.12:20:37\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 3799.913/10931.675 MB - 34.76% done   2019-06-05.12:20:38\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 3959.909/10931.675 MB - 36.22% done   2019-06-05.12:20:39\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 4159.905/10931.675 MB - 38.05% done   2019-06-05.12:20:41\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 4319.901/10931.675 MB - 39.52% done   2019-06-05.12:20:42\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 4479.897/10931.675 MB - 40.98% done   2019-06-05.12:20:43\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 4679.893/10931.675 MB - 42.81% done   2019-06-05.12:20:44\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 4839.889/10931.675 MB - 44.27% done   2019-06-05.12:20:45\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 4999.886/10931.675 MB - 45.74% done   2019-06-05.12:20:46\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 5239.880/10931.675 MB - 47.93% done   2019-06-05.12:20:48\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 5359.877/10931.675 MB - 49.03% done   2019-06-05.12:20:49\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 5479.875/10931.675 MB - 50.13% done   2019-06-05.12:20:50\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 5599.872/10931.675 MB - 51.23% done   2019-06-05.12:20:50\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 5759.868/10931.675 MB - 52.69% done   2019-06-05.12:20:51\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 5879.865/10931.675 MB - 53.79% done   2019-06-05.12:20:53\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 6039.862/10931.675 MB - 55.25% done   2019-06-05.12:20:55\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 6199.858/10931.675 MB - 56.71% done   2019-06-05.12:20:56\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 6359.854/10931.675 MB - 58.18% done   2019-06-05.12:20:57\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 6439.853/10931.675 MB - 58.91% done   2019-06-05.12:20:58\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 6559.850/10931.675 MB - 60.01% done   2019-06-05.12:20:58\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 6679.847/10931.675 MB - 61.11% done   2019-06-05.12:20:59\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 6759.845/10931.675 MB - 61.84% done   2019-06-05.12:21:00\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 6919.842/10931.675 MB - 63.30% done   2019-06-05.12:21:01\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 6999.840/10931.675 MB - 64.03% done   2019-06-05.12:21:01\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 7159.836/10931.675 MB - 65.50% done   2019-06-05.12:21:02\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 7319.832/10931.675 MB - 66.96% done   2019-06-05.12:21:04\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 7359.832/10931.675 MB - 67.33% done   2019-06-05.12:21:06\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 7559.827/10931.675 MB - 69.16% done   2019-06-05.12:21:07\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 7679.824/10931.675 MB - 70.25% done   2019-06-05.12:21:08\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 7839.821/10931.675 MB - 71.72% done   2019-06-05.12:21:09\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 7999.817/10931.675 MB - 73.18% done   2019-06-05.12:21:10\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 8119.814/10931.675 MB - 74.28% done   2019-06-05.12:21:10\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 8319.810/10931.675 MB - 76.11% done   2019-06-05.12:21:11\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 8479.806/10931.675 MB - 77.57% done   2019-06-05.12:21:12\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 8639.802/10931.675 MB - 79.03% done   2019-06-05.12:21:14\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 8799.799/10931.675 MB - 80.50% done   2019-06-05.12:21:17\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 9039.793/10931.675 MB - 82.69% done   2019-06-05.12:21:18\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 9119.791/10931.675 MB - 83.43% done   2019-06-05.12:21:19\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 9319.787/10931.675 MB - 85.25% done   2019-06-05.12:21:20\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 9479.783/10931.675 MB - 86.72% done   2019-06-05.12:21:22\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 9679.778/10931.675 MB - 88.55% done   2019-06-05.12:21:23\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 9919.773/10931.675 MB - 90.74% done   2019-06-05.12:21:25\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 9932.754/10931.675 MB - 90.86% done   2019-06-05.12:21:26\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_2.fq.gz - 10931.675/10931.675 MB - 100.00% done   2019-06-05.12:21:32\n",
      "3/5 - 60.00% of files done   21503.094/26231.851 MB - 81.97% of file sizes done\n",
      "Processing TrfM6_USD16090827L_HKFY3DSXX_L3_1.fq.gz - 2320.076 MB   2019-06-05.12:21:55\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_1.fq.gz - 119.997/2320.076 MB -  5.17% done   2019-06-05.12:21:57\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_1.fq.gz - 239.995/2320.076 MB - 10.34% done   2019-06-05.12:21:58\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_1.fq.gz - 359.992/2320.076 MB - 15.52% done   2019-06-05.12:21:59\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_1.fq.gz - 519.988/2320.076 MB - 22.41% done   2019-06-05.12:22:00\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_1.fq.gz - 679.984/2320.076 MB - 29.31% done   2019-06-05.12:22:02\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_1.fq.gz - 839.981/2320.076 MB - 36.20% done   2019-06-05.12:22:04\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_1.fq.gz - 999.977/2320.076 MB - 43.10% done   2019-06-05.12:22:05\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_1.fq.gz - 1119.974/2320.076 MB - 48.27% done   2019-06-05.12:22:06\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_1.fq.gz - 1239.972/2320.076 MB - 53.45% done   2019-06-05.12:22:07\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_1.fq.gz - 1319.970/2320.076 MB - 56.89% done   2019-06-05.12:22:08\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_1.fq.gz - 1479.966/2320.076 MB - 63.79% done   2019-06-05.12:22:08\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_1.fq.gz - 1639.962/2320.076 MB - 70.69% done   2019-06-05.12:22:09\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_1.fq.gz - 1839.958/2320.076 MB - 79.31% done   2019-06-05.12:22:11\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_1.fq.gz - 1999.954/2320.076 MB - 86.20% done   2019-06-05.12:22:12\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_1.fq.gz - 2139.983/2320.076 MB - 92.24% done   2019-06-05.12:22:13\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_1.fq.gz - 2320.076/2320.076 MB - 100.00% done   2019-06-05.12:22:15\n",
      "4/5 - 80.00% of files done   23823.170/26231.851 MB - 90.82% of file sizes done\n",
      "Processing TrfM6_USD16090827L_HKFY3DSXX_L3_2.fq.gz - 2408.681 MB   2019-06-05.12:22:20\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_2.fq.gz - 199.995/2408.681 MB -  8.30% done   2019-06-05.12:22:23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrfM6_USD16090827L_HKFY3DSXX_L3_2.fq.gz - 359.992/2408.681 MB - 14.95% done   2019-06-05.12:22:24\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_2.fq.gz - 559.987/2408.681 MB - 23.25% done   2019-06-05.12:22:26\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_2.fq.gz - 799.982/2408.681 MB - 33.21% done   2019-06-05.12:22:27\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_2.fq.gz - 959.978/2408.681 MB - 39.85% done   2019-06-05.12:22:28\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_2.fq.gz - 1159.973/2408.681 MB - 48.16% done   2019-06-05.12:22:30\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_2.fq.gz - 1319.970/2408.681 MB - 54.80% done   2019-06-05.12:22:31\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_2.fq.gz - 1479.966/2408.681 MB - 61.44% done   2019-06-05.12:22:32\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_2.fq.gz - 1799.959/2408.681 MB - 74.73% done   2019-06-05.12:22:34\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_2.fq.gz - 2042.137/2408.681 MB - 84.78% done   2019-06-05.12:22:35\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_2.fq.gz - 2204.318/2408.681 MB - 91.52% done   2019-06-05.12:22:36\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_2.fq.gz - 2286.500/2408.681 MB - 94.93% done   2019-06-05.12:22:37\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_2.fq.gz - 2326.499/2408.681 MB - 96.59% done   2019-06-05.12:22:37\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_2.fq.gz - 2366.498/2408.681 MB - 98.25% done   2019-06-05.12:22:37\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_2.fq.gz - 2406.497/2408.681 MB - 99.91% done   2019-06-05.12:22:38\n",
      "TrfM6_USD16090827L_HKFY3DSXX_L3_2.fq.gz - 2408.681/2408.681 MB - 100.00% done   2019-06-05.12:22:38\n"
     ]
    }
   ],
   "source": [
    "#Get the notebooks and data\n",
    "iget -rKP -N 4 /iplant/home/annis/WGS_2019/lg2/WGS_May2019_LG2.ipynb\n",
    "iget -rKP -N 4 /iplant/home/annis/WGS_2019/HojaI/WGS_HojaI.ipynb\n",
    "iget -rKP -N 4 /iplant/home/annis/WGS_2019/TRF/Data_TrfM6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#run QC on the raw data files \n",
    "#Run FASTP on the data with paried end modality, using the compressed files. \n",
    "#LocaM2_USD16090676L_HJLYTDSXX_L1_1.fq.gz\n",
    "mkdir ./fastp_initial\n",
    "find ./Data_TrfM6/ -name \"*.fq.gz\" | while read f; do\n",
    "STEM=$(basename \"${f}\" _1.fq.gz)\n",
    "fastp -w 4 -i ./Data_TrfM6/${STEM}_1.fq.gz \\\n",
    "    -I ./Data_TrfM6/${STEM}_2.fq.gz \\\n",
    "    -o ./fastp_initial/${STEM}_R1.fq.gz \\\n",
    "    -O ./fastp_initial/${STEM}_R2.fq.gz \\\n",
    "    -h ./fastp_initial/${STEM}.html \\\n",
    "    -j ./fastp_initial/${STEM}.json \\\n",
    "    -R \"${STEM} fastp report\"\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cat: './fastp_initial/*L3_R1.fq.gz': No such file or directory\n",
      "cat: './fastp_initial/*L3_R2.fq.gz': No such file or directory\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "#combine the two lanes of sequencing to get single input files \n",
    "mkdir ./trimmed_reads\n",
    "cat ./fastp_initial/*L1_R1.fq.gz ./fastp_initial/*L3_R1.fq.gz> ./trimmed_reads/TrfM6_R1.fastq.gz\n",
    "cat ./fastp_initial/*L1_R2.fq.gz ./fastp_initial/*L3_R2.fq.gz> ./trimmed_reads/TrfM6_R2.fastq.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0/2 -  0.00% of files done   0.000/4688.893 MB -  0.00% of file sizes done\n",
      "Processing TrfM6_USD16090827L_HJLYTDSXX_L1_R1.fq.gz - 2315.947 MB   2019-06-06.12:36:09\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R1.fq.gz - 39.999/2315.947 MB -  1.73% done   2019-06-06.12:36:19\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R1.fq.gz - 239.995/2315.947 MB - 10.36% done   2019-06-06.12:36:23\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R1.fq.gz - 399.991/2315.947 MB - 17.27% done   2019-06-06.12:36:27\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R1.fq.gz - 519.988/2315.947 MB - 22.45% done   2019-06-06.12:36:31\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R1.fq.gz - 679.984/2315.947 MB - 29.36% done   2019-06-06.12:36:34\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R1.fq.gz - 759.983/2315.947 MB - 32.82% done   2019-06-06.12:36:36\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R1.fq.gz - 879.980/2315.947 MB - 38.00% done   2019-06-06.12:36:39\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R1.fq.gz - 999.977/2315.947 MB - 43.18% done   2019-06-06.12:36:42\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R1.fq.gz - 1119.974/2315.947 MB - 48.36% done   2019-06-06.12:36:45\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R1.fq.gz - 1279.971/2315.947 MB - 55.27% done   2019-06-06.12:36:48\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R1.fq.gz - 1319.970/2315.947 MB - 56.99% done   2019-06-06.12:36:50\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R1.fq.gz - 1399.968/2315.947 MB - 60.45% done   2019-06-06.12:36:52\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R1.fq.gz - 1519.965/2315.947 MB - 65.63% done   2019-06-06.12:36:54\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R1.fq.gz - 1599.963/2315.947 MB - 69.08% done   2019-06-06.12:36:56\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R1.fq.gz - 1618.963/2315.947 MB - 69.91% done   2019-06-06.12:36:58\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R1.fq.gz - 2315.947/2315.947 MB - 100.00% done   2019-06-06.12:37:14\n",
      "1/2 - 50.00% of files done   2315.947/4688.893 MB - 49.39% of file sizes done\n",
      "Processing TrfM6_USD16090827L_HJLYTDSXX_L1_R2.fq.gz - 2372.946 MB   2019-06-06.12:37:27\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R2.fq.gz - 39.999/2372.946 MB -  1.69% done   2019-06-06.12:37:35\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R2.fq.gz - 159.996/2372.946 MB -  6.74% done   2019-06-06.12:37:37\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R2.fq.gz - 239.995/2372.946 MB - 10.11% done   2019-06-06.12:37:39\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R2.fq.gz - 359.992/2372.946 MB - 15.17% done   2019-06-06.12:37:42\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R2.fq.gz - 519.988/2372.946 MB - 21.91% done   2019-06-06.12:37:44\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R2.fq.gz - 679.984/2372.946 MB - 28.66% done   2019-06-06.12:37:47\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R2.fq.gz - 879.980/2372.946 MB - 37.08% done   2019-06-06.12:37:51\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R2.fq.gz - 1039.976/2372.946 MB - 43.83% done   2019-06-06.12:37:54\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R2.fq.gz - 1239.972/2372.946 MB - 52.25% done   2019-06-06.12:37:58\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R2.fq.gz - 1439.967/2372.946 MB - 60.68% done   2019-06-06.12:38:02\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R2.fq.gz - 1599.963/2372.946 MB - 67.43% done   2019-06-06.12:38:05\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R2.fq.gz - 1799.959/2372.946 MB - 75.85% done   2019-06-06.12:38:09\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R2.fq.gz - 1999.954/2372.946 MB - 84.28% done   2019-06-06.12:38:13\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R2.fq.gz - 2146.451/2372.946 MB - 90.46% done   2019-06-06.12:38:15\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R2.fq.gz - 2219.699/2372.946 MB - 93.54% done   2019-06-06.12:38:16\n",
      "TrfM6_USD16090827L_HJLYTDSXX_L1_R2.fq.gz - 2372.946/2372.946 MB - 100.00% done   2019-06-06.12:38:18\n",
      "0/2 -  0.00% of files done   0.000/4688.893 MB -  0.00% of file sizes done\n",
      "Processing TrfM6_R2.fastq.gz - 2372.946 MB   2019-06-06.12:38:32\n",
      "TrfM6_R2.fastq.gz - 39.999/2372.946 MB -  1.69% done   2019-06-06.12:38:41\n",
      "TrfM6_R2.fastq.gz - 119.997/2372.946 MB -  5.06% done   2019-06-06.12:38:43\n",
      "TrfM6_R2.fastq.gz - 239.995/2372.946 MB - 10.11% done   2019-06-06.12:38:46\n",
      "TrfM6_R2.fastq.gz - 359.992/2372.946 MB - 15.17% done   2019-06-06.12:38:49\n",
      "TrfM6_R2.fastq.gz - 479.989/2372.946 MB - 20.23% done   2019-06-06.12:38:52\n",
      "TrfM6_R2.fastq.gz - 599.986/2372.946 MB - 25.28% done   2019-06-06.12:38:56\n",
      "TrfM6_R2.fastq.gz - 759.983/2372.946 MB - 32.03% done   2019-06-06.12:38:59\n",
      "TrfM6_R2.fastq.gz - 879.980/2372.946 MB - 37.08% done   2019-06-06.12:39:02\n",
      "TrfM6_R2.fastq.gz - 959.978/2372.946 MB - 40.46% done   2019-06-06.12:39:04\n",
      "TrfM6_R2.fastq.gz - 1039.976/2372.946 MB - 43.83% done   2019-06-06.12:39:07\n",
      "TrfM6_R2.fastq.gz - 1159.973/2372.946 MB - 48.88% done   2019-06-06.12:39:09\n",
      "TrfM6_R2.fastq.gz - 1199.973/2372.946 MB - 50.57% done   2019-06-06.12:39:12\n",
      "TrfM6_R2.fastq.gz - 1359.969/2372.946 MB - 57.31% done   2019-06-06.12:39:15\n",
      "TrfM6_R2.fastq.gz - 1559.964/2372.946 MB - 65.74% done   2019-06-06.12:39:19\n",
      "TrfM6_R2.fastq.gz - 1633.213/2372.946 MB - 68.83% done   2019-06-06.12:39:22\n",
      "TrfM6_R2.fastq.gz - 2372.946/2372.946 MB - 100.00% done   2019-06-06.12:39:40\n",
      "1/2 - 50.00% of files done   2372.946/4688.893 MB - 50.61% of file sizes done\n",
      "Processing TrfM6_R1.fastq.gz - 2315.947 MB   2019-06-06.12:39:53\n",
      "TrfM6_R1.fastq.gz - 119.997/2315.947 MB -  5.18% done   2019-06-06.12:40:03\n",
      "TrfM6_R1.fastq.gz - 359.992/2315.947 MB - 15.54% done   2019-06-06.12:40:08\n",
      "TrfM6_R1.fastq.gz - 559.987/2315.947 MB - 24.18% done   2019-06-06.12:40:13\n",
      "TrfM6_R1.fastq.gz - 759.983/2315.947 MB - 32.82% done   2019-06-06.12:40:19\n",
      "TrfM6_R1.fastq.gz - 1039.976/2315.947 MB - 44.91% done   2019-06-06.12:40:25\n",
      "TrfM6_R1.fastq.gz - 1258.971/2315.947 MB - 54.36% done   2019-06-06.12:40:30\n",
      "TrfM6_R1.fastq.gz - 1338.969/2315.947 MB - 57.82% done   2019-06-06.12:40:33\n",
      "TrfM6_R1.fastq.gz - 1458.967/2315.947 MB - 63.00% done   2019-06-06.12:40:35\n",
      "TrfM6_R1.fastq.gz - 1578.964/2315.947 MB - 68.18% done   2019-06-06.12:40:38\n",
      "TrfM6_R1.fastq.gz - 1698.961/2315.947 MB - 73.36% done   2019-06-06.12:40:41\n"
     ]
    }
   ],
   "source": [
    "#Transfer files to CYVERSE\n",
    "icd /iplant/home/annis/WGS_2019/TRF\n",
    "iput -rKP -N 4 ./fastp_initial\n",
    "iput -rKP -N 4 ./trimmed_reads\n",
    "iput -rKP -N 4 ./WGS_TRF_May2019.ipynb"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The above steps were carried out May 2019 for the B73v3 alignment,\n",
    "The trimmed read files were downloaded from CYVERSE to an iMac and used for the below alignment to the NAM5.0 genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/aricha14/Documents/Research_Projects/TRF_Project/WGS\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022\n"
     ]
    }
   ],
   "source": [
    "cd ./WGS_Alignment_NAM5_May2022\n",
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/local/bin/bowtie2-align-s version 2.4.4\n",
      "64-bit\n",
      "Built on sce-bio-c05491\n",
      "Wed 25 Aug 2021 11:26:53 UTC\n",
      "Compiler: InstalledDir: /Library/Developer/CommandLineTools/usr/bin\n",
      "Options: -O3 -msse2 -funroll-loops -g3 -std=c++11 -DPOPCNT_CAPABILITY -DNO_SPINLOCK -DWITH_QUEUELOCK=1\n",
      "Sizeof {int, long, long long, void*, size_t, off_t}: {4, 8, 8, 8, 8, 8}\n"
     ]
    }
   ],
   "source": [
    "bowtie2 --version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error to read gzip file\n",
      "Error to read gzip file\n",
      "ERROR: file './Data_TrfN5/TrfN5_USD16090826L_HJLYTDSXX_L1_2.fq.gz_1.fq.gz' doesn't exist, quit now\n",
      "Error to read gzip file\n",
      "Error to read gzip file\n",
      "ERROR: file './Data_TrfN5/TrfN5_USD16090826L_HKFY3DSXX_L3_2.fq.gz_1.fq.gz' doesn't exist, quit now\n",
      "Read1 before filtering:\n",
      "total reads: 149387158\n",
      "total bases: 22408073700\n",
      "Q20 bases: 21943083852(97.9249%)\n",
      "Q30 bases: 21184098303(94.5378%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 149387158\n",
      "total bases: 22408073700\n",
      "Q20 bases: 21836814068(97.4507%)\n",
      "Q30 bases: 20917287983(93.3471%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 148369209\n",
      "total bases: 22139496525\n",
      "Q20 bases: 21745151421(98.2188%)\n",
      "Q30 bases: 21017968636(94.9343%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 148369209\n",
      "total bases: 22139824104\n",
      "Q20 bases: 21661525821(97.8396%)\n",
      "Q30 bases: 20778503431(93.8513%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 296738418\n",
      "reads failed due to low quality: 1854010\n",
      "reads failed due to too many N: 71010\n",
      "reads failed due to too short: 110878\n",
      "reads with adapter trimmed: 5444934\n",
      "bases trimmed due to adapters: 222820107\n",
      "\n",
      "Duplication rate: 19.8145%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 269\n",
      "\n",
      "JSON report: ./fastp_initial/TrfN5_USD16090826L_HJLYTDSXX_L1.json\n",
      "HTML report: ./fastp_initial/TrfN5_USD16090826L_HJLYTDSXX_L1.html\n",
      "\n",
      "fastp -w 4 -i ./Data_TrfN5/TrfN5_USD16090826L_HJLYTDSXX_L1_1.fq.gz -I ./Data_TrfN5/TrfN5_USD16090826L_HJLYTDSXX_L1_2.fq.gz -o ./fastp_initial/TrfN5_USD16090826L_HJLYTDSXX_L1_R1.fq.gz -O ./fastp_initial/TrfN5_USD16090826L_HJLYTDSXX_L1_R2.fq.gz -h ./fastp_initial/TrfN5_USD16090826L_HJLYTDSXX_L1.html -j ./fastp_initial/TrfN5_USD16090826L_HJLYTDSXX_L1.json -R TrfN5_USD16090826L_HJLYTDSXX_L1 fastp report \n",
      "fastp v0.22.0, time used: 833 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 31444312\n",
      "total bases: 4716646800\n",
      "Q20 bases: 4597074015(97.4649%)\n",
      "Q30 bases: 4394654882(93.1733%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 31444312\n",
      "total bases: 4716646800\n",
      "Q20 bases: 4569038505(96.8705%)\n",
      "Q30 bases: 4323628691(91.6674%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 31173219\n",
      "total bases: 4654718119\n",
      "Q20 bases: 4549082546(97.7306%)\n",
      "Q30 bases: 4353901360(93.5374%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 31173219\n",
      "total bases: 4654827401\n",
      "Q20 bases: 4526867297(97.251%)\n",
      "Q30 bases: 4289555082(92.1528%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 62346438\n",
      "reads failed due to low quality: 436916\n",
      "reads failed due to too many N: 84242\n",
      "reads failed due to too short: 21028\n",
      "reads with adapter trimmed: 1051172\n",
      "bases trimmed due to adapters: 41282394\n",
      "\n",
      "Duplication rate: 18.5964%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 269\n",
      "\n",
      "JSON report: ./fastp_initial/TrfN5_USD16090826L_HKFY3DSXX_L3.json\n",
      "HTML report: ./fastp_initial/TrfN5_USD16090826L_HKFY3DSXX_L3.html\n",
      "\n",
      "fastp -w 4 -i ./Data_TrfN5/TrfN5_USD16090826L_HKFY3DSXX_L3_1.fq.gz -I ./Data_TrfN5/TrfN5_USD16090826L_HKFY3DSXX_L3_2.fq.gz -o ./fastp_initial/TrfN5_USD16090826L_HKFY3DSXX_L3_R1.fq.gz -O ./fastp_initial/TrfN5_USD16090826L_HKFY3DSXX_L3_R2.fq.gz -h ./fastp_initial/TrfN5_USD16090826L_HKFY3DSXX_L3.html -j ./fastp_initial/TrfN5_USD16090826L_HKFY3DSXX_L3.json -R TrfN5_USD16090826L_HKFY3DSXX_L3 fastp report \n",
      "fastp v0.22.0, time used: 183 seconds\n"
     ]
    }
   ],
   "source": [
    "mkdir ./fastp_initial\n",
    "find ./Data_TrfN5/ -name \"*.fq.gz\" | while read f; do\n",
    "STEM=$(basename \"${f}\" _1.fq.gz)\n",
    "fastp -w 4 -i ./Data_TrfN5/${STEM}_1.fq.gz \\\n",
    "    -I ./Data_TrfN5/${STEM}_2.fq.gz \\\n",
    "    -o ./fastp_initial/${STEM}_R1.fq.gz \\\n",
    "    -O ./fastp_initial/${STEM}_R2.fq.gz \\\n",
    "    -h ./fastp_initial/${STEM}.html \\\n",
    "    -j ./fastp_initial/${STEM}.json \\\n",
    "    -R \"${STEM} fastp report\"\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: ./trimmed_reads: File exists\n"
     ]
    }
   ],
   "source": [
    "#combine the two lanes of sequencing to get single input files \n",
    "mkdir ./trimmed_reads\n",
    "cat ./fastp_initial/*L1_R1.fq.gz ./fastp_initial/*L3_R1.fq.gz> ./trimmed_reads/TrfM6_R1.fastq.gz\n",
    "cat ./fastp_initial/*L1_R2.fq.gz ./fastp_initial/*L3_R2.fq.gz> ./trimmed_reads/TrfM6_R2.fastq.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: ./data_comb: File exists\n"
     ]
    }
   ],
   "source": [
    "# First combine the reads from the two lanes of sequencing to make single files \n",
    "mkdir ./data_comb\n",
    "cat ./Data_TrfN5/*L1_1.fq.gz ./Data_TrfN5/*L3_1.fq.gz> ./data_comb/TrfN5_rawlanescomb_1.fastq.gz\n",
    "cat ./Data_TrfN5/*L1_2.fq.gz ./Data_TrfN5/*L3_2.fq.gz> ./data_comb/TrfN5_rawlanescomb_2.fastq.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error to read gzip file\n",
      "Error to read gzip file\n",
      "ERROR: file './data_comb/TrfN5_rawlanescomb_2.fastq.gz_1.fastq.gz' doesn't exist, quit now\n",
      "Read1 before filtering:\n",
      "total reads: 180831470\n",
      "total bases: 27124720500\n",
      "Q20 bases: 26540157867(97.8449%)\n",
      "Q30 bases: 25578753185(94.3005%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 180831470\n",
      "total bases: 27124720500\n",
      "Q20 bases: 26405852573(97.3498%)\n",
      "Q30 bases: 25240916674(93.055%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 179542428\n",
      "total bases: 26794214644\n",
      "Q20 bases: 26294233967(98.134%)\n",
      "Q30 bases: 25371869996(94.6916%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 179542428\n",
      "total bases: 26794651505\n",
      "Q20 bases: 26188393118(97.7374%)\n",
      "Q30 bases: 25068058513(93.5562%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 359084856\n",
      "reads failed due to low quality: 2290926\n",
      "reads failed due to too many N: 155252\n",
      "reads failed due to too short: 131906\n",
      "reads with adapter trimmed: 6496106\n",
      "bases trimmed due to adapters: 264102501\n",
      "\n",
      "Duplication rate: 19.744%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 269\n",
      "\n",
      "JSON report: ./fastp_initial/TrfN5_rawlanescomb.json\n",
      "HTML report: ./fastp_initial/TrfN5_rawlanescomb.html\n",
      "\n",
      "fastp -w 4 -i ./data_comb/TrfN5_rawlanescomb_1.fastq.gz -I ./Data_comb/TrfN5_rawlanescomb_2.fastq.gz -o ./fastp_initial/TrfN5_rawlanescomb_R1.fastq.gz -O ./fastp_initial/TrfN5_rawlanescomb_R2.fastq.gz -h ./fastp_initial/TrfN5_rawlanescomb.html -j ./fastp_initial/TrfN5_rawlanescomb.json -R TrfN5_rawlanescomb fastp report \n",
      "fastp v0.22.0, time used: 1037 seconds\n"
     ]
    }
   ],
   "source": [
    "# Run fastp analysis and trimming on the combined files using paired end settings \n",
    "find ./data_comb/ -name \"*.fastq.gz\" | while read f; do\n",
    "STEM=$(basename \"${f}\" _1.fastq.gz)\n",
    "fastp -w 4 -i ./data_comb/${STEM}_1.fastq.gz \\\n",
    "    -I ./Data_comb/${STEM}_2.fastq.gz \\\n",
    "    -o ./fastp_initial/${STEM}_R1.fastq.gz \\\n",
    "    -O ./fastp_initial/${STEM}_R2.fastq.gz \\\n",
    "    -h ./fastp_initial/${STEM}.html \\\n",
    "    -j ./fastp_initial/${STEM}.json \\\n",
    "    -R \"${STEM} fastp report\"\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "179542428 reads; of these:\n",
      "  179542428 (100.00%) were paired; of these:\n",
      "    12296586 (6.85%) aligned concordantly 0 times\n",
      "    29115709 (16.22%) aligned concordantly exactly 1 time\n",
      "    138130133 (76.93%) aligned concordantly >1 times\n",
      "    ----\n",
      "    12296586 pairs aligned concordantly 0 times; of these:\n",
      "      237290 (1.93%) aligned discordantly 1 time\n",
      "    ----\n",
      "    12059296 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      24118592 mates make up the pairs; of these:\n",
      "        13580141 (56.31%) aligned 0 times\n",
      "        3086021 (12.80%) aligned exactly 1 time\n",
      "        7452430 (30.90%) aligned >1 times\n",
      "96.22% overall alignment rate\n"
     ]
    }
   ],
   "source": [
    "bowtie2 -L 20 -p 4 \\\n",
    "    -x /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0 \\\n",
    "    -1 ./trimmed_reads/TrfN5_rawlanescomb_R1.fastq.gz \\\n",
    "    -2 ./trimmed_reads/TrfN5_rawlanescomb_R2.fastq.gz \\\n",
    "    -S ./aligned_reads/TrfN5_aligned_NAM5.sam \n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bam_sort_core] merging from 176 files and 4 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "mkdir ./BAM_reads\n",
    "#Convert to .bam file\n",
    "samtools sort -o ./BAM_reads/TrfN5_aligned_NAM5.bam -@ 4 ./aligned_reads/TrfN5_aligned_NAM5.sam \n",
    "#index the bam file\n",
    "samtools index ./BAM_reads/TrfN5_aligned_NAM5.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/iplant/home/annis\n"
     ]
    }
   ],
   "source": [
    "ipwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/iplant/home/annis/WGS/WGS_2019/TRF/Trf_WGS_NAM5_alignment_May2022:\n",
      "  C- /iplant/home/annis/WGS/WGS_2019/TRF/Trf_WGS_NAM5_alignment_May2022/aligned_reads_NAM5\n"
     ]
    }
   ],
   "source": [
    "icd /iplant/home/annis/WGS/WGS_2019/TRF/Trf_WGS_NAM5_alignment_May2022\n",
    "ils"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0/2 -  0.00% of files done   0.000/156883.753 MB -  0.00% of file sizes done\n",
      "Processing TrfN5_aligned_NAM5.sam - 156883.737 MB   2022-05-16.10:04:22\n",
      "TrfN5_aligned_NAM5.sam - 119.997/156883.737 MB -  0.08% done   2022-05-16.10:12:12\n",
      "TrfN5_aligned_NAM5.sam - 319.993/156883.737 MB -  0.20% done   2022-05-16.10:16:26\n",
      "TrfN5_aligned_NAM5.sam - 479.989/156883.737 MB -  0.31% done   2022-05-16.10:20:52\n",
      "TrfN5_aligned_NAM5.sam - 639.985/156883.737 MB -  0.41% done   2022-05-16.10:24:58\n",
      "TrfN5_aligned_NAM5.sam - 839.981/156883.737 MB -  0.54% done   2022-05-16.10:33:09\n",
      "TrfN5_aligned_NAM5.sam - 1039.976/156883.737 MB -  0.66% done   2022-05-16.10:41:51\n",
      "TrfN5_aligned_NAM5.sam - 1199.973/156883.737 MB -  0.76% done   2022-05-16.10:50:04\n",
      "TrfN5_aligned_NAM5.sam - 1359.969/156883.737 MB -  0.87% done   2022-05-16.10:57:02\n",
      "TrfN5_aligned_NAM5.sam - 1519.965/156883.737 MB -  0.97% done   2022-05-16.11:02:17\n",
      "TrfN5_aligned_NAM5.sam - 1679.962/156883.737 MB -  1.07% done   2022-05-16.11:06:03\n",
      "TrfN5_aligned_NAM5.sam - 1839.958/156883.737 MB -  1.17% done   2022-05-16.11:09:43\n",
      "TrfN5_aligned_NAM5.sam - 1999.954/156883.737 MB -  1.27% done   2022-05-16.11:13:39\n",
      "TrfN5_aligned_NAM5.sam - 2159.951/156883.737 MB -  1.38% done   2022-05-16.11:19:49\n",
      "TrfN5_aligned_NAM5.sam - 2319.947/156883.737 MB -  1.48% done   2022-05-16.11:26:52\n",
      "TrfN5_aligned_NAM5.sam - 2479.943/156883.737 MB -  1.58% done   2022-05-16.11:35:48\n",
      "TrfN5_aligned_NAM5.sam - 2639.940/156883.737 MB -  1.68% done   2022-05-16.11:46:02\n",
      "TrfN5_aligned_NAM5.sam - 2839.935/156883.737 MB -  1.81% done   2022-05-16.11:57:49\n",
      "TrfN5_aligned_NAM5.sam - 2999.931/156883.737 MB -  1.91% done   2022-05-16.12:08:14\n",
      "TrfN5_aligned_NAM5.sam - 3159.928/156883.737 MB -  2.01% done   2022-05-16.12:21:10\n",
      "TrfN5_aligned_NAM5.sam - 3319.924/156883.737 MB -  2.12% done   2022-05-16.12:34:17\n",
      "TrfN5_aligned_NAM5.sam - 3479.920/156883.737 MB -  2.22% done   2022-05-16.12:48:17\n",
      "TrfN5_aligned_NAM5.sam - 3639.917/156883.737 MB -  2.32% done   2022-05-16.13:02:55\n",
      "TrfN5_aligned_NAM5.sam - 3799.913/156883.737 MB -  2.42% done   2022-05-16.13:16:39\n",
      "TrfN5_aligned_NAM5.sam - 3959.909/156883.737 MB -  2.52% done   2022-05-16.13:29:59\n",
      "TrfN5_aligned_NAM5.sam - 4159.905/156883.737 MB -  2.65% done   2022-05-16.13:41:55\n",
      "TrfN5_aligned_NAM5.sam - 4359.900/156883.737 MB -  2.78% done   2022-05-16.13:54:07\n",
      "TrfN5_aligned_NAM5.sam - 4519.897/156883.737 MB -  2.88% done   2022-05-16.14:04:41\n",
      "TrfN5_aligned_NAM5.sam - 4679.893/156883.737 MB -  2.98% done   2022-05-16.14:15:05\n",
      "TrfN5_aligned_NAM5.sam - 4839.889/156883.737 MB -  3.09% done   2022-05-16.14:25:37\n",
      "TrfN5_aligned_NAM5.sam - 4999.886/156883.737 MB -  3.19% done   2022-05-16.14:35:16\n",
      "TrfN5_aligned_NAM5.sam - 5159.882/156883.737 MB -  3.29% done   2022-05-16.14:46:11\n",
      "TrfN5_aligned_NAM5.sam - 5319.878/156883.737 MB -  3.39% done   2022-05-16.14:57:35\n",
      "TrfN5_aligned_NAM5.sam - 5479.875/156883.737 MB -  3.49% done   2022-05-16.15:08:22\n",
      "TrfN5_aligned_NAM5.sam - 5639.871/156883.737 MB -  3.59% done   2022-05-16.15:19:04\n",
      "TrfN5_aligned_NAM5.sam - 5799.867/156883.737 MB -  3.70% done   2022-05-16.15:28:53\n",
      "TrfN5_aligned_NAM5.sam - 5959.864/156883.737 MB -  3.80% done   2022-05-16.15:37:42\n",
      "TrfN5_aligned_NAM5.sam - 6119.860/156883.737 MB -  3.90% done   2022-05-16.15:47:46\n",
      "TrfN5_aligned_NAM5.sam - 6279.856/156883.737 MB -  4.00% done   2022-05-16.15:56:24\n",
      "TrfN5_aligned_NAM5.sam - 6439.853/156883.737 MB -  4.10% done   2022-05-16.16:07:58\n",
      "TrfN5_aligned_NAM5.sam - 6599.849/156883.737 MB -  4.21% done   2022-05-16.16:19:17\n",
      "TrfN5_aligned_NAM5.sam - 6759.845/156883.737 MB -  4.31% done   2022-05-16.16:30:25\n",
      "TrfN5_aligned_NAM5.sam - 6919.842/156883.737 MB -  4.41% done   2022-05-16.16:41:19\n",
      "TrfN5_aligned_NAM5.sam - 7079.838/156883.737 MB -  4.51% done   2022-05-16.16:52:23\n",
      "TrfN5_aligned_NAM5.sam - 7239.834/156883.737 MB -  4.61% done   2022-05-16.17:03:42\n",
      "TrfN5_aligned_NAM5.sam - 7399.831/156883.737 MB -  4.72% done   2022-05-16.17:15:40\n",
      "TrfN5_aligned_NAM5.sam - 7559.827/156883.737 MB -  4.82% done   2022-05-16.17:27:40\n",
      "TrfN5_aligned_NAM5.sam - 7679.824/156883.737 MB -  4.90% done   2022-05-16.17:38:59\n",
      "TrfN5_aligned_NAM5.sam - 7839.821/156883.737 MB -  5.00% done   2022-05-16.17:50:41\n",
      "TrfN5_aligned_NAM5.sam - 7999.817/156883.737 MB -  5.10% done   2022-05-16.18:02:51\n",
      "TrfN5_aligned_NAM5.sam - 8199.812/156883.737 MB -  5.23% done   2022-05-16.18:15:02\n",
      "TrfN5_aligned_NAM5.sam - 8359.809/156883.737 MB -  5.33% done   2022-05-16.18:27:33\n",
      "TrfN5_aligned_NAM5.sam - 8519.805/156883.737 MB -  5.43% done   2022-05-16.18:40:25\n",
      "TrfN5_aligned_NAM5.sam - 8719.800/156883.737 MB -  5.56% done   2022-05-16.18:52:45\n",
      "TrfN5_aligned_NAM5.sam - 8879.797/156883.737 MB -  5.66% done   2022-05-16.19:05:24\n",
      "TrfN5_aligned_NAM5.sam - 9039.793/156883.737 MB -  5.76% done   2022-05-16.19:16:34\n",
      "TrfN5_aligned_NAM5.sam - 9199.789/156883.737 MB -  5.86% done   2022-05-16.19:27:48\n",
      "TrfN5_aligned_NAM5.sam - 9359.786/156883.737 MB -  5.97% done   2022-05-16.19:38:45\n",
      "TrfN5_aligned_NAM5.sam - 9519.782/156883.737 MB -  6.07% done   2022-05-16.19:48:43\n",
      "TrfN5_aligned_NAM5.sam - 9679.778/156883.737 MB -  6.17% done   2022-05-16.19:58:47\n",
      "TrfN5_aligned_NAM5.sam - 9839.775/156883.737 MB -  6.27% done   2022-05-16.20:08:34\n",
      "TrfN5_aligned_NAM5.sam - 9999.771/156883.737 MB -  6.37% done   2022-05-16.20:19:04\n",
      "TrfN5_aligned_NAM5.sam - 10159.767/156883.737 MB -  6.48% done   2022-05-16.20:28:45\n",
      "TrfN5_aligned_NAM5.sam - 10279.765/156883.737 MB -  6.55% done   2022-05-16.20:37:54\n",
      "TrfN5_aligned_NAM5.sam - 10439.761/156883.737 MB -  6.65% done   2022-05-16.20:46:59\n",
      "TrfN5_aligned_NAM5.sam - 10599.757/156883.737 MB -  6.76% done   2022-05-16.20:56:17\n",
      "TrfN5_aligned_NAM5.sam - 10759.754/156883.737 MB -  6.86% done   2022-05-16.21:04:52\n",
      "TrfN5_aligned_NAM5.sam - 10919.750/156883.737 MB -  6.96% done   2022-05-16.21:13:00\n",
      "TrfN5_aligned_NAM5.sam - 11079.746/156883.737 MB -  7.06% done   2022-05-16.21:21:34\n",
      "TrfN5_aligned_NAM5.sam - 11279.742/156883.737 MB -  7.19% done   2022-05-16.21:31:21\n",
      "TrfN5_aligned_NAM5.sam - 11439.738/156883.737 MB -  7.29% done   2022-05-16.21:39:13\n",
      "TrfN5_aligned_NAM5.sam - 11559.735/156883.737 MB -  7.37% done   2022-05-16.21:47:35\n",
      "TrfN5_aligned_NAM5.sam - 11719.732/156883.737 MB -  7.47% done   2022-05-16.21:55:01\n",
      "TrfN5_aligned_NAM5.sam - 11879.728/156883.737 MB -  7.57% done   2022-05-16.22:05:15\n",
      "TrfN5_aligned_NAM5.sam - 12039.724/156883.737 MB -  7.67% done   2022-05-16.22:15:32\n",
      "TrfN5_aligned_NAM5.sam - 12199.721/156883.737 MB -  7.78% done   2022-05-16.22:27:26\n",
      "TrfN5_aligned_NAM5.sam - 12359.717/156883.737 MB -  7.88% done   2022-05-16.22:37:58\n",
      "TrfN5_aligned_NAM5.sam - 12519.713/156883.737 MB -  7.98% done   2022-05-16.22:48:05\n",
      "TrfN5_aligned_NAM5.sam - 12679.710/156883.737 MB -  8.08% done   2022-05-16.23:00:16\n",
      "TrfN5_aligned_NAM5.sam - 12839.706/156883.737 MB -  8.18% done   2022-05-16.23:11:14\n",
      "TrfN5_aligned_NAM5.sam - 12999.702/156883.737 MB -  8.29% done   2022-05-16.23:21:57\n",
      "TrfN5_aligned_NAM5.sam - 13159.699/156883.737 MB -  8.39% done   2022-05-16.23:34:05\n",
      "TrfN5_aligned_NAM5.sam - 13279.696/156883.737 MB -  8.46% done   2022-05-16.23:46:05\n",
      "TrfN5_aligned_NAM5.sam - 13439.692/156883.737 MB -  8.57% done   2022-05-16.23:57:45\n",
      "TrfN5_aligned_NAM5.sam - 13599.689/156883.737 MB -  8.67% done   2022-05-17.00:10:07\n",
      "TrfN5_aligned_NAM5.sam - 13719.686/156883.737 MB -  8.75% done   2022-05-17.00:22:15\n",
      "TrfN5_aligned_NAM5.sam - 13879.682/156883.737 MB -  8.85% done   2022-05-17.00:35:02\n",
      "TrfN5_aligned_NAM5.sam - 14039.679/156883.737 MB -  8.95% done   2022-05-17.00:47:54\n",
      "TrfN5_aligned_NAM5.sam - 14199.675/156883.737 MB -  9.05% done   2022-05-17.01:00:48\n",
      "TrfN5_aligned_NAM5.sam - 14359.671/156883.737 MB -  9.15% done   2022-05-17.01:13:18\n",
      "TrfN5_aligned_NAM5.sam - 14479.669/156883.737 MB -  9.23% done   2022-05-17.01:23:24\n",
      "TrfN5_aligned_NAM5.sam - 14639.665/156883.737 MB -  9.33% done   2022-05-17.01:36:30\n",
      "TrfN5_aligned_NAM5.sam - 14799.661/156883.737 MB -  9.43% done   2022-05-17.01:49:23\n",
      "TrfN5_aligned_NAM5.sam - 14959.658/156883.737 MB -  9.54% done   2022-05-17.02:02:29\n",
      "TrfN5_aligned_NAM5.sam - 15119.654/156883.737 MB -  9.64% done   2022-05-17.02:16:16\n",
      "TrfN5_aligned_NAM5.sam - 15279.650/156883.737 MB -  9.74% done   2022-05-17.02:29:52\n",
      "TrfN5_aligned_NAM5.sam - 15439.647/156883.737 MB -  9.84% done   2022-05-17.02:43:15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrfN5_aligned_NAM5.sam - 15599.643/156883.737 MB -  9.94% done   2022-05-17.02:55:55\n",
      "TrfN5_aligned_NAM5.sam - 15759.639/156883.737 MB - 10.05% done   2022-05-17.03:10:06\n",
      "TrfN5_aligned_NAM5.sam - 15919.636/156883.737 MB - 10.15% done   2022-05-17.03:22:41\n",
      "TrfN5_aligned_NAM5.sam - 16079.632/156883.737 MB - 10.25% done   2022-05-17.03:36:52\n",
      "TrfN5_aligned_NAM5.sam - 16239.628/156883.737 MB - 10.35% done   2022-05-17.03:50:53\n",
      "TrfN5_aligned_NAM5.sam - 16399.625/156883.737 MB - 10.45% done   2022-05-17.04:04:53\n",
      "TrfN5_aligned_NAM5.sam - 16559.621/156883.737 MB - 10.56% done   2022-05-17.04:18:07\n",
      "TrfN5_aligned_NAM5.sam - 16719.617/156883.737 MB - 10.66% done   2022-05-17.04:32:12\n",
      "TrfN5_aligned_NAM5.sam - 16919.613/156883.737 MB - 10.78% done   2022-05-17.04:46:05\n",
      "TrfN5_aligned_NAM5.sam - 17079.609/156883.737 MB - 10.89% done   2022-05-17.04:59:07\n",
      "TrfN5_aligned_NAM5.sam - 17239.605/156883.737 MB - 10.99% done   2022-05-17.05:11:59\n",
      "TrfN5_aligned_NAM5.sam - 17399.602/156883.737 MB - 11.09% done   2022-05-17.05:25:07\n",
      "TrfN5_aligned_NAM5.sam - 17559.598/156883.737 MB - 11.19% done   2022-05-17.05:38:11\n",
      "TrfN5_aligned_NAM5.sam - 17719.594/156883.737 MB - 11.29% done   2022-05-17.05:51:34\n",
      "TrfN5_aligned_NAM5.sam - 17879.591/156883.737 MB - 11.40% done   2022-05-17.06:04:16\n",
      "TrfN5_aligned_NAM5.sam - 18039.587/156883.737 MB - 11.50% done   2022-05-17.06:17:47\n",
      "TrfN5_aligned_NAM5.sam - 18199.583/156883.737 MB - 11.60% done   2022-05-17.06:31:14\n",
      "TrfN5_aligned_NAM5.sam - 18399.579/156883.737 MB - 11.73% done   2022-05-17.06:44:56\n",
      "TrfN5_aligned_NAM5.sam - 18559.575/156883.737 MB - 11.83% done   2022-05-17.06:59:03\n",
      "TrfN5_aligned_NAM5.sam - 18719.572/156883.737 MB - 11.93% done   2022-05-17.07:10:53\n",
      "TrfN5_aligned_NAM5.sam - 18879.568/156883.737 MB - 12.03% done   2022-05-17.07:22:20\n",
      "TrfN5_aligned_NAM5.sam - 19039.564/156883.737 MB - 12.14% done   2022-05-17.07:35:08\n",
      "TrfN5_aligned_NAM5.sam - 19199.561/156883.737 MB - 12.24% done   2022-05-17.07:46:37\n",
      "TrfN5_aligned_NAM5.sam - 19319.558/156883.737 MB - 12.31% done   2022-05-17.07:58:30\n",
      "TrfN5_aligned_NAM5.sam - 19479.554/156883.737 MB - 12.42% done   2022-05-17.08:08:58\n",
      "TrfN5_aligned_NAM5.sam - 19639.550/156883.737 MB - 12.52% done   2022-05-17.08:20:09\n",
      "TrfN5_aligned_NAM5.sam - 19799.547/156883.737 MB - 12.62% done   2022-05-17.08:30:40\n",
      "TrfN5_aligned_NAM5.sam - 19959.543/156883.737 MB - 12.72% done   2022-05-17.08:41:59\n",
      "TrfN5_aligned_NAM5.sam - 20119.540/156883.737 MB - 12.82% done   2022-05-17.08:53:01\n",
      "TrfN5_aligned_NAM5.sam - 20279.536/156883.737 MB - 12.93% done   2022-05-17.09:03:49\n",
      "TrfN5_aligned_NAM5.sam - 20439.532/156883.737 MB - 13.03% done   2022-05-17.09:15:33\n",
      "TrfN5_aligned_NAM5.sam - 20599.529/156883.737 MB - 13.13% done   2022-05-17.09:27:45\n",
      "TrfN5_aligned_NAM5.sam - 20759.525/156883.737 MB - 13.23% done   2022-05-17.09:40:13\n",
      "TrfN5_aligned_NAM5.sam - 20919.521/156883.737 MB - 13.33% done   2022-05-17.09:51:00\n",
      "TrfN5_aligned_NAM5.sam - 21079.518/156883.737 MB - 13.44% done   2022-05-17.10:01:14\n",
      "TrfN5_aligned_NAM5.sam - 21239.514/156883.737 MB - 13.54% done   2022-05-17.10:09:15\n",
      "TrfN5_aligned_NAM5.sam - 21319.512/156883.737 MB - 13.59% done   2022-05-17.10:14:23\n",
      "TrfN5_aligned_NAM5.sam - 21439.509/156883.737 MB - 13.67% done   2022-05-17.10:19:21\n",
      "TrfN5_aligned_NAM5.sam - 21559.507/156883.737 MB - 13.74% done   2022-05-17.10:27:59\n",
      "TrfN5_aligned_NAM5.sam - 21719.503/156883.737 MB - 13.84% done   2022-05-17.10:38:23\n",
      "TrfN5_aligned_NAM5.sam - 21879.499/156883.737 MB - 13.95% done   2022-05-17.10:48:25\n",
      "TrfN5_aligned_NAM5.sam - 22039.496/156883.737 MB - 14.05% done   2022-05-17.10:58:17\n",
      "TrfN5_aligned_NAM5.sam - 22199.492/156883.737 MB - 14.15% done   2022-05-17.11:09:59\n",
      "TrfN5_aligned_NAM5.sam - 22359.488/156883.737 MB - 14.25% done   2022-05-17.11:22:02\n",
      "TrfN5_aligned_NAM5.sam - 22519.485/156883.737 MB - 14.35% done   2022-05-17.11:33:48\n",
      "TrfN5_aligned_NAM5.sam - 22719.480/156883.737 MB - 14.48% done   2022-05-17.11:45:34\n",
      "TrfN5_aligned_NAM5.sam - 22879.476/156883.737 MB - 14.58% done   2022-05-17.11:57:55\n",
      "TrfN5_aligned_NAM5.sam - 23039.473/156883.737 MB - 14.69% done   2022-05-17.12:09:48\n",
      "TrfN5_aligned_NAM5.sam - 23199.469/156883.737 MB - 14.79% done   2022-05-17.12:22:14\n",
      "TrfN5_aligned_NAM5.sam - 23359.465/156883.737 MB - 14.89% done   2022-05-17.12:34:18\n",
      "TrfN5_aligned_NAM5.sam - 23519.462/156883.737 MB - 14.99% done   2022-05-17.12:45:56\n",
      "TrfN5_aligned_NAM5.sam - 23679.458/156883.737 MB - 15.09% done   2022-05-17.12:57:24\n",
      "TrfN5_aligned_NAM5.sam - 23839.454/156883.737 MB - 15.20% done   2022-05-17.13:08:34\n",
      "TrfN5_aligned_NAM5.sam - 24039.450/156883.737 MB - 15.32% done   2022-05-17.13:20:02\n",
      "TrfN5_aligned_NAM5.sam - 24199.446/156883.737 MB - 15.43% done   2022-05-17.13:31:27\n",
      "TrfN5_aligned_NAM5.sam - 24359.442/156883.737 MB - 15.53% done   2022-05-17.13:43:28\n",
      "TrfN5_aligned_NAM5.sam - 24519.439/156883.737 MB - 15.63% done   2022-05-17.13:55:05\n",
      "TrfN5_aligned_NAM5.sam - 24679.435/156883.737 MB - 15.73% done   2022-05-17.14:05:40\n",
      "TrfN5_aligned_NAM5.sam - 24839.431/156883.737 MB - 15.83% done   2022-05-17.14:16:34\n",
      "TrfN5_aligned_NAM5.sam - 24999.428/156883.737 MB - 15.94% done   2022-05-17.14:27:41\n",
      "TrfN5_aligned_NAM5.sam - 25159.424/156883.737 MB - 16.04% done   2022-05-17.14:38:12\n",
      "TrfN5_aligned_NAM5.sam - 25319.420/156883.737 MB - 16.14% done   2022-05-17.14:49:21\n",
      "TrfN5_aligned_NAM5.sam - 25439.418/156883.737 MB - 16.22% done   2022-05-17.15:00:06\n",
      "TrfN5_aligned_NAM5.sam - 25639.413/156883.737 MB - 16.34% done   2022-05-17.15:09:33\n",
      "TrfN5_aligned_NAM5.sam - 25719.411/156883.737 MB - 16.39% done   2022-05-17.15:19:01\n",
      "TrfN5_aligned_NAM5.sam - 25879.408/156883.737 MB - 16.50% done   2022-05-17.15:28:24\n",
      "TrfN5_aligned_NAM5.sam - 26039.404/156883.737 MB - 16.60% done   2022-05-17.15:38:13\n",
      "TrfN5_aligned_NAM5.sam - 26159.401/156883.737 MB - 16.67% done   2022-05-17.15:48:06\n",
      "TrfN5_aligned_NAM5.sam - 26319.398/156883.737 MB - 16.78% done   2022-05-17.15:57:47\n",
      "TrfN5_aligned_NAM5.sam - 26479.394/156883.737 MB - 16.88% done   2022-05-17.16:08:35\n",
      "TrfN5_aligned_NAM5.sam - 26639.390/156883.737 MB - 16.98% done   2022-05-17.16:19:31\n",
      "TrfN5_aligned_NAM5.sam - 26799.387/156883.737 MB - 17.08% done   2022-05-17.16:28:44\n",
      "TrfN5_aligned_NAM5.sam - 26919.384/156883.737 MB - 17.16% done   2022-05-17.16:34:00\n",
      "TrfN5_aligned_NAM5.sam - 26999.382/156883.737 MB - 17.21% done   2022-05-17.16:39:03\n",
      "TrfN5_aligned_NAM5.sam - 27119.379/156883.737 MB - 17.29% done   2022-05-17.16:44:09\n",
      "TrfN5_aligned_NAM5.sam - 27199.377/156883.737 MB - 17.34% done   2022-05-17.16:49:06\n",
      "TrfN5_aligned_NAM5.sam - 27319.375/156883.737 MB - 17.41% done   2022-05-17.16:53:47\n",
      "TrfN5_aligned_NAM5.sam - 27399.373/156883.737 MB - 17.46% done   2022-05-17.16:58:44\n",
      "TrfN5_aligned_NAM5.sam - 27439.372/156883.737 MB - 17.49% done   2022-05-17.17:03:33\n",
      "TrfN5_aligned_NAM5.sam - 27599.368/156883.737 MB - 17.59% done   2022-05-17.17:07:52\n",
      "TrfN5_aligned_NAM5.sam - 27639.367/156883.737 MB - 17.62% done   2022-05-17.17:12:43\n",
      "TrfN5_aligned_NAM5.sam - 27759.365/156883.737 MB - 17.69% done   2022-05-17.17:17:58\n",
      "TrfN5_aligned_NAM5.sam - 27839.363/156883.737 MB - 17.75% done   2022-05-17.17:23:08\n",
      "TrfN5_aligned_NAM5.sam - 27959.360/156883.737 MB - 17.82% done   2022-05-17.17:29:50\n",
      "TrfN5_aligned_NAM5.sam - 28119.356/156883.737 MB - 17.92% done   2022-05-17.17:40:56\n",
      "TrfN5_aligned_NAM5.sam - 28279.353/156883.737 MB - 18.03% done   2022-05-17.17:52:54\n",
      "TrfN5_aligned_NAM5.sam - 28439.349/156883.737 MB - 18.13% done   2022-05-17.18:04:56\n",
      "TrfN5_aligned_NAM5.sam - 28599.345/156883.737 MB - 18.23% done   2022-05-17.18:16:42\n",
      "TrfN5_aligned_NAM5.sam - 28759.342/156883.737 MB - 18.33% done   2022-05-17.18:28:10\n",
      "TrfN5_aligned_NAM5.sam - 28919.338/156883.737 MB - 18.43% done   2022-05-17.18:39:45\n",
      "TrfN5_aligned_NAM5.sam - 29079.334/156883.737 MB - 18.54% done   2022-05-17.18:50:50\n",
      "TrfN5_aligned_NAM5.sam - 29239.331/156883.737 MB - 18.64% done   2022-05-17.19:02:39\n",
      "TrfN5_aligned_NAM5.sam - 29399.327/156883.737 MB - 18.74% done   2022-05-17.19:13:09\n",
      "TrfN5_aligned_NAM5.sam - 29559.323/156883.737 MB - 18.84% done   2022-05-17.19:22:12\n",
      "TrfN5_aligned_NAM5.sam - 29719.320/156883.737 MB - 18.94% done   2022-05-17.19:33:02\n",
      "TrfN5_aligned_NAM5.sam - 29879.316/156883.737 MB - 19.05% done   2022-05-17.19:43:14\n",
      "TrfN5_aligned_NAM5.sam - 30039.312/156883.737 MB - 19.15% done   2022-05-17.19:53:03\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrfN5_aligned_NAM5.sam - 30199.309/156883.737 MB - 19.25% done   2022-05-17.20:03:41\n",
      "TrfN5_aligned_NAM5.sam - 30399.304/156883.737 MB - 19.38% done   2022-05-17.20:13:56\n",
      "TrfN5_aligned_NAM5.sam - 30559.301/156883.737 MB - 19.48% done   2022-05-17.20:24:35\n",
      "TrfN5_aligned_NAM5.sam - 30719.297/156883.737 MB - 19.58% done   2022-05-17.20:34:15\n",
      "TrfN5_aligned_NAM5.sam - 30879.293/156883.737 MB - 19.68% done   2022-05-17.20:45:10\n",
      "TrfN5_aligned_NAM5.sam - 31039.290/156883.737 MB - 19.78% done   2022-05-17.20:56:40\n",
      "TrfN5_aligned_NAM5.sam - 31199.286/156883.737 MB - 19.89% done   2022-05-17.21:07:36\n",
      "TrfN5_aligned_NAM5.sam - 31359.282/156883.737 MB - 19.99% done   2022-05-17.21:17:42\n",
      "TrfN5_aligned_NAM5.sam - 31519.279/156883.737 MB - 20.09% done   2022-05-17.21:27:52\n",
      "TrfN5_aligned_NAM5.sam - 31679.275/156883.737 MB - 20.19% done   2022-05-17.21:38:09\n",
      "TrfN5_aligned_NAM5.sam - 31839.271/156883.737 MB - 20.29% done   2022-05-17.21:47:59\n",
      "TrfN5_aligned_NAM5.sam - 31999.268/156883.737 MB - 20.40% done   2022-05-17.21:58:50\n",
      "TrfN5_aligned_NAM5.sam - 32159.264/156883.737 MB - 20.50% done   2022-05-17.22:09:51\n",
      "TrfN5_aligned_NAM5.sam - 32319.260/156883.737 MB - 20.60% done   2022-05-17.22:21:12\n",
      "TrfN5_aligned_NAM5.sam - 32479.257/156883.737 MB - 20.70% done   2022-05-17.22:34:01\n",
      "TrfN5_aligned_NAM5.sam - 32639.253/156883.737 MB - 20.80% done   2022-05-17.22:45:16\n",
      "TrfN5_aligned_NAM5.sam - 32799.249/156883.737 MB - 20.91% done   2022-05-17.22:54:05\n",
      "TrfN5_aligned_NAM5.sam - 32959.246/156883.737 MB - 21.01% done   2022-05-17.23:05:45\n",
      "TrfN5_aligned_NAM5.sam - 33119.242/156883.737 MB - 21.11% done   2022-05-17.23:18:04\n",
      "TrfN5_aligned_NAM5.sam - 33319.237/156883.737 MB - 21.24% done   2022-05-17.23:30:48\n",
      "TrfN5_aligned_NAM5.sam - 33479.234/156883.737 MB - 21.34% done   2022-05-17.23:41:50\n",
      "TrfN5_aligned_NAM5.sam - 33639.230/156883.737 MB - 21.44% done   2022-05-17.23:55:13\n",
      "TrfN5_aligned_NAM5.sam - 33799.226/156883.737 MB - 21.54% done   2022-05-18.00:07:57\n",
      "TrfN5_aligned_NAM5.sam - 33959.223/156883.737 MB - 21.65% done   2022-05-18.00:21:25\n",
      "TrfN5_aligned_NAM5.sam - 34119.219/156883.737 MB - 21.75% done   2022-05-18.00:34:50\n",
      "TrfN5_aligned_NAM5.sam - 34279.215/156883.737 MB - 21.85% done   2022-05-18.00:48:20\n",
      "TrfN5_aligned_NAM5.sam - 34439.212/156883.737 MB - 21.95% done   2022-05-18.01:01:06\n",
      "TrfN5_aligned_NAM5.sam - 34599.208/156883.737 MB - 22.05% done   2022-05-18.01:13:57\n",
      "TrfN5_aligned_NAM5.sam - 34759.204/156883.737 MB - 22.16% done   2022-05-18.01:27:26\n",
      "TrfN5_aligned_NAM5.sam - 34919.201/156883.737 MB - 22.26% done   2022-05-18.01:41:30\n",
      "TrfN5_aligned_NAM5.sam - 35119.196/156883.737 MB - 22.39% done   2022-05-18.01:54:54\n"
     ]
    }
   ],
   "source": [
    "iput -rKP -N 4 /Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022/aligned_reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0/2 -  0.00% of files done   0.000/18364.009 MB -  0.00% of file sizes done\n",
      "Processing TrfN5_aligned_NAM5.bam - 18358.232 MB   2022-05-20.11:15:24\n",
      "ERROR: putDirUtil: put /Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022/BAM_reads/TrfN5_aligned_NAM5.bam failed. status = -312000 status = -312000 OVERWRITE_WITHOUT_FORCE_FLAG\n",
      "0/2 -  0.00% of files done   0.000/18364.009 MB -  0.00% of file sizes done\n",
      "Processing TrfN5_aligned_NAM5.bam.bai - 5.777 MB   2022-05-20.11:15:55\n",
      "ERROR: putDirUtil: put /Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022/BAM_reads/TrfN5_aligned_NAM5.bam.bai failed. status = -312000 status = -312000 OVERWRITE_WITHOUT_FORCE_FLAG\n",
      "ERROR: putUtil: put error for /iplant/home/annis/WGS/WGS_2019/TRF/Trf_WGS_NAM5_alignment_May2022/BAM_reads, status = -312000 status = -312000 OVERWRITE_WITHOUT_FORCE_FLAG\n"
     ]
    },
    {
     "ename": "",
     "evalue": "3",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "iput -rKP /Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022/BAM_reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_B73v3\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/aricha14/Documents/Research_Projects/TRF_Project/WGS\n"
     ]
    }
   ],
   "source": [
    "cd ..\n",
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd ./WGS_Alignment_NAM5_May2022"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Repeat the process for the mutant reads \n",
    "# First combine the reads from the two lanes of sequencing to make single files \n",
    "cat ./Data_TrfM6/*L1_1.fq.gz ./Data_TrfM6/*L3_1.fq.gz > ./data_comb/TrfM6_rawlanescomb_1.fastq.gz\n",
    "cat ./Data_TrfM6/*L1_2.fq.gz ./Data_TrfM6/*L3_2.fq.gz > ./data_comb/TrfM6_rawlanescomb_2.fastq.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 179932017\n",
      "total bases: 26989802550\n",
      "Q20 bases: 26398927471(97.8107%)\n",
      "Q30 bases: 25413736337(94.1605%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 179932017\n",
      "total bases: 26989802550\n",
      "Q20 bases: 26234066718(97.1999%)\n",
      "Q30 bases: 25016000634(92.6869%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 178607321\n",
      "total bases: 26647882118\n",
      "Q20 bases: 26141891249(98.1012%)\n",
      "Q30 bases: 25196902393(94.555%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 178607321\n",
      "total bases: 26647699819\n",
      "Q20 bases: 26008363647(97.6008%)\n",
      "Q30 bases: 24836615495(93.2036%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 357214642\n",
      "reads failed due to low quality: 2334042\n",
      "reads failed due to too many N: 154374\n",
      "reads failed due to too short: 160976\n",
      "reads with adapter trimmed: 7298510\n",
      "bases trimmed due to adapters: 277215524\n",
      "\n",
      "Duplication rate: 17.1911%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 269\n",
      "\n",
      "JSON report: ./fastp_initial/TrfM6_rawlanescomb.json\n",
      "HTML report: ./fastp_initial/TrfM6_rawlanescomb.html\n",
      "\n",
      "fastp -w 4 -i ./data_comb/TrfM6_rawlanescomb_1.fastq.gz -I ./Data_comb/TrfM6_rawlanescomb_2.fastq.gz -o ./fastp_initial/TrfM6_rawlanescomb_R1.fastq.gz -O ./fastp_initial/TrfM6_rawlanescomb_R2.fastq.gz -h ./fastp_initial/TrfM6_rawlanescomb.html -j ./fastp_initial/TrfM6_rawlanescomb.json -R TrfM6_rawlanescomb fastp report \n",
      "fastp v0.22.0, time used: 1033 seconds\n",
      "Error to read gzip file\n",
      "Error to read gzip file\n",
      "ERROR: file './data_comb/TrfM6_rawlanescomb_2.fastq.gz_1.fastq.gz' doesn't exist, quit now\n"
     ]
    },
    {
     "ename": "",
     "evalue": "255",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "# Run fastp analysis and trimming on the combined files using paired end settings \n",
    "find ./data_comb/ -name \"*.fastq.gz\" | while read f; do\n",
    "STEM=$(basename \"${f}\" _1.fastq.gz)\n",
    "fastp -w 4 -i ./data_comb/${STEM}_1.fastq.gz \\\n",
    "    -I ./Data_comb/${STEM}_2.fastq.gz \\\n",
    "    -o ./fastp_initial/${STEM}_R1.fastq.gz \\\n",
    "    -O ./fastp_initial/${STEM}_R2.fastq.gz \\\n",
    "    -h ./fastp_initial/${STEM}.html \\\n",
    "    -j ./fastp_initial/${STEM}.json \\\n",
    "    -R \"${STEM} fastp report\"\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "178607321 reads; of these:\n",
      "  178607321 (100.00%) were paired; of these:\n",
      "    12261851 (6.87%) aligned concordantly 0 times\n",
      "    28727292 (16.08%) aligned concordantly exactly 1 time\n",
      "    137618178 (77.05%) aligned concordantly >1 times\n",
      "    ----\n",
      "    12261851 pairs aligned concordantly 0 times; of these:\n",
      "      245747 (2.00%) aligned discordantly 1 time\n",
      "    ----\n",
      "    12016104 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      24032208 mates make up the pairs; of these:\n",
      "        13447126 (55.95%) aligned 0 times\n",
      "        3045112 (12.67%) aligned exactly 1 time\n",
      "        7539970 (31.37%) aligned >1 times\n",
      "96.24% overall alignment rate\n"
     ]
    }
   ],
   "source": [
    "bowtie2 -L 20 -p 4 \\\n",
    "    -x /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0 \\\n",
    "    -1 ./trimmed_reads/TrfM6_rawlanescomb_R1.fastq.gz \\\n",
    "    -2 ./trimmed_reads/TrfM6_rawlanescomb_R2.fastq.gz \\\n",
    "    -S ./aligned_reads/TrfM6_aligned_NAM5.sam \n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: ./BAM_reads: File exists\n",
      "[bam_sort_core] merging from 176 files and 4 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "mkdir ./BAM_reads\n",
    "#Convert to .bam file\n",
    "samtools sort -o ./BAM_reads/TrfM6_aligned_NAM5.bam -@ 4 ./aligned_reads/TrfM6_aligned_NAM5.sam \n",
    "#index the bam file\n",
    "samtools index ./BAM_reads/TrfM6_aligned_NAM5.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/iplant/home/annis/WGS/WGS_2019/TRF/Trf_WGS_NAM5_alignment_May2022\n"
     ]
    }
   ],
   "source": [
    "ipwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0/1 -  0.00% of files done   0.000/18584.622 MB -  0.00% of file sizes done\n",
      "Processing TrfM6_aligned_NAM5.bam - 18584.622 MB   2022-05-25.12:20:18\n",
      "ERROR: putUtil: put error for /iplant/home/annis/WGS/WGS_2019/TRF/Trf_WGS_NAM5_alignment_May2022/BAM_reads/TrfM6_aligned_NAM5.bam, status = -312000 status = -312000 OVERWRITE_WITHOUT_FORCE_FLAG\n",
      "0/1 -  0.00% of files done   0.000/5.778 MB -  0.00% of file sizes done\n",
      "Processing TrfM6_aligned_NAM5.bam.bai - 5.778 MB   2022-05-25.12:20:54\n",
      "ERROR: putUtil: put error for /iplant/home/annis/WGS/WGS_2019/TRF/Trf_WGS_NAM5_alignment_May2022/BAM_reads/TrfM6_aligned_NAM5.bam.bai, status = -312000 status = -312000 OVERWRITE_WITHOUT_FORCE_FLAG\n"
     ]
    },
    {
     "ename": "",
     "evalue": "3",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "icd ./BAM_reads\n",
    "iput -rKP /Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022/BAM_reads/TrfM6_aligned_NAM5.bam\n",
    "iput -rKP /Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022/BAM_reads/TrfM6_aligned_NAM5.bam.bai"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/iplant/home/annis/WGS/WGS_2019/TRF/Trf_WGS_NAM5_alignment_May2022/BAM_reads\n"
     ]
    }
   ],
   "source": [
    "ipwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0/1 -  0.00% of files done   0.000/18584.622 MB -  0.00% of file sizes done\n",
      "Processing TrfM6_aligned_NAM5.bam - 18584.622 MB   2022-05-25.16:27:15\n",
      "TrfM6_aligned_NAM5.bam - 199.995/18584.622 MB -  1.08% done   2022-05-25.16:33:56\n",
      "TrfM6_aligned_NAM5.bam - 679.984/18584.622 MB -  3.66% done   2022-05-25.16:39:13\n",
      "TrfM6_aligned_NAM5.bam - 1319.970/18584.622 MB -  7.10% done   2022-05-25.16:45:49\n",
      "TrfM6_aligned_NAM5.bam - 1959.955/18584.622 MB - 10.55% done   2022-05-25.16:51:57\n",
      "TrfM6_aligned_NAM5.bam - 2599.940/18584.622 MB - 13.99% done   2022-05-25.16:59:08\n",
      "TrfM6_aligned_NAM5.bam - 3239.926/18584.622 MB - 17.43% done   2022-05-25.17:07:02\n",
      "TrfM6_aligned_NAM5.bam - 3799.913/18584.622 MB - 20.45% done   2022-05-25.17:14:09\n",
      "TrfM6_aligned_NAM5.bam - 4439.898/18584.622 MB - 23.89% done   2022-05-25.17:20:42\n",
      "TrfM6_aligned_NAM5.bam - 5119.883/18584.622 MB - 27.55% done   2022-05-25.17:26:54\n",
      "TrfM6_aligned_NAM5.bam - 5599.872/18584.622 MB - 30.13% done   2022-05-25.17:32:34\n",
      "TrfM6_aligned_NAM5.bam - 6159.859/18584.622 MB - 33.14% done   2022-05-25.17:38:36\n",
      "TrfM6_aligned_NAM5.bam - 6639.848/18584.622 MB - 35.73% done   2022-05-25.17:44:15\n",
      "TrfM6_aligned_NAM5.bam - 7279.833/18584.622 MB - 39.17% done   2022-05-25.17:50:02\n",
      "TrfM6_aligned_NAM5.bam - 7839.821/18584.622 MB - 42.18% done   2022-05-25.17:55:03\n",
      "TrfM6_aligned_NAM5.bam - 8199.812/18584.622 MB - 44.12% done   2022-05-25.17:59:36\n",
      "TrfM6_aligned_NAM5.bam - 8879.797/18584.622 MB - 47.78% done   2022-05-25.18:04:56\n",
      "TrfM6_aligned_NAM5.bam - 9399.785/18584.622 MB - 50.58% done   2022-05-25.18:09:45\n",
      "TrfM6_aligned_NAM5.bam - 9879.774/18584.622 MB - 53.16% done   2022-05-25.18:14:50\n",
      "TrfM6_aligned_NAM5.bam - 10399.762/18584.622 MB - 55.96% done   2022-05-25.18:19:40\n",
      "TrfM6_aligned_NAM5.bam - 10999.748/18584.622 MB - 59.19% done   2022-05-25.18:24:52\n",
      "TrfM6_aligned_NAM5.bam - 11559.735/18584.622 MB - 62.20% done   2022-05-25.18:29:53\n",
      "TrfM6_aligned_NAM5.bam - 12079.724/18584.622 MB - 65.00% done   2022-05-25.18:35:07\n",
      "TrfM6_aligned_NAM5.bam - 12639.711/18584.622 MB - 68.01% done   2022-05-25.18:39:51\n",
      "TrfM6_aligned_NAM5.bam - 13159.699/18584.622 MB - 70.81% done   2022-05-25.18:44:41\n",
      "TrfM6_aligned_NAM5.bam - 13679.687/18584.622 MB - 73.61% done   2022-05-25.18:49:41\n",
      "TrfM6_aligned_NAM5.bam - 14279.673/18584.622 MB - 76.84% done   2022-05-25.18:54:48\n",
      "TrfM6_aligned_NAM5.bam - 14719.663/18584.622 MB - 79.20% done   2022-05-25.18:59:31\n",
      "TrfM6_aligned_NAM5.bam - 15279.650/18584.622 MB - 82.22% done   2022-05-25.19:04:35\n",
      "TrfM6_aligned_NAM5.bam - 15884.333/18584.622 MB - 85.47% done   2022-05-25.19:09:24\n",
      "TrfM6_aligned_NAM5.bam - 15885.898/18584.622 MB - 85.48% done   2022-05-25.19:09:34\n",
      "TrfM6_aligned_NAM5.bam - 18584.622/18584.622 MB - 100.00% done   2022-05-25.22:46:46\n"
     ]
    }
   ],
   "source": [
    "iput -rKP /Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022/BAM_reads/TrfM6_aligned_NAM5.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/iplant/home/annis/WGS/WGS_2019/TRF/Trf_WGS_NAM5_alignment_May2022\n"
     ]
    }
   ],
   "source": [
    "icd ..\n",
    "ipwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0/1 -  0.00% of files done   0.000/156051.022 MB -  0.00% of file sizes done\n",
      "Processing TrfM6_aligned_NAM5.sam - 156051.022 MB   2022-05-25.23:12:50\n",
      "TrfM6_aligned_NAM5.sam - 319.993/156051.022 MB -  0.21% done   2022-05-25.23:23:11\n",
      "TrfM6_aligned_NAM5.sam - 1119.974/156051.022 MB -  0.72% done   2022-05-25.23:29:30\n",
      "TrfM6_aligned_NAM5.sam - 1799.959/156051.022 MB -  1.15% done   2022-05-25.23:35:23\n",
      "TrfM6_aligned_NAM5.sam - 2439.944/156051.022 MB -  1.56% done   2022-05-25.23:41:07\n",
      "TrfM6_aligned_NAM5.sam - 3159.928/156051.022 MB -  2.02% done   2022-05-25.23:47:18\n",
      "TrfM6_aligned_NAM5.sam - 3839.912/156051.022 MB -  2.46% done   2022-05-25.23:53:42\n",
      "TrfM6_aligned_NAM5.sam - 4439.898/156051.022 MB -  2.85% done   2022-05-25.23:59:50\n",
      "TrfM6_aligned_NAM5.sam - 5079.884/156051.022 MB -  3.26% done   2022-05-26.00:05:36\n",
      "TrfM6_aligned_NAM5.sam - 5679.870/156051.022 MB -  3.64% done   2022-05-26.00:11:29\n",
      "TrfM6_aligned_NAM5.sam - 6319.855/156051.022 MB -  4.05% done   2022-05-26.00:17:26\n",
      "TrfM6_aligned_NAM5.sam - 6999.840/156051.022 MB -  4.49% done   2022-05-26.00:23:39\n",
      "TrfM6_aligned_NAM5.sam - 7599.826/156051.022 MB -  4.87% done   2022-05-26.00:29:21\n",
      "TrfM6_aligned_NAM5.sam - 8239.811/156051.022 MB -  5.28% done   2022-05-26.00:35:21\n",
      "TrfM6_aligned_NAM5.sam - 8879.797/156051.022 MB -  5.69% done   2022-05-26.00:41:17\n",
      "TrfM6_aligned_NAM5.sam - 9479.783/156051.022 MB -  6.07% done   2022-05-26.00:46:54\n",
      "TrfM6_aligned_NAM5.sam - 10079.769/156051.022 MB -  6.46% done   2022-05-26.00:52:47\n",
      "TrfM6_aligned_NAM5.sam - 10719.755/156051.022 MB -  6.87% done   2022-05-26.00:58:53\n",
      "TrfM6_aligned_NAM5.sam - 11359.740/156051.022 MB -  7.28% done   2022-05-26.01:04:31\n",
      "TrfM6_aligned_NAM5.sam - 11999.725/156051.022 MB -  7.69% done   2022-05-26.01:10:51\n",
      "TrfM6_aligned_NAM5.sam - 12719.709/156051.022 MB -  8.15% done   2022-05-26.01:17:00\n",
      "TrfM6_aligned_NAM5.sam - 13359.694/156051.022 MB -  8.56% done   2022-05-26.01:23:04\n",
      "TrfM6_aligned_NAM5.sam - 13999.680/156051.022 MB -  8.97% done   2022-05-26.01:29:09\n",
      "TrfM6_aligned_NAM5.sam - 14679.664/156051.022 MB -  9.41% done   2022-05-26.01:35:10\n",
      "TrfM6_aligned_NAM5.sam - 15279.650/156051.022 MB -  9.79% done   2022-05-26.01:41:06\n",
      "TrfM6_aligned_NAM5.sam - 15999.634/156051.022 MB - 10.25% done   2022-05-26.01:47:34\n",
      "TrfM6_aligned_NAM5.sam - 16639.619/156051.022 MB - 10.66% done   2022-05-26.01:53:24\n",
      "TrfM6_aligned_NAM5.sam - 17279.605/156051.022 MB - 11.07% done   2022-05-26.01:59:27\n",
      "TrfM6_aligned_NAM5.sam - 17919.590/156051.022 MB - 11.48% done   2022-05-26.02:05:33\n",
      "TrfM6_aligned_NAM5.sam - 18559.575/156051.022 MB - 11.89% done   2022-05-26.02:11:30\n",
      "TrfM6_aligned_NAM5.sam - 19239.560/156051.022 MB - 12.33% done   2022-05-26.02:17:33\n",
      "TrfM6_aligned_NAM5.sam - 19839.546/156051.022 MB - 12.71% done   2022-05-26.02:23:25\n",
      "TrfM6_aligned_NAM5.sam - 20479.531/156051.022 MB - 13.12% done   2022-05-26.02:29:08\n",
      "TrfM6_aligned_NAM5.sam - 21039.518/156051.022 MB - 13.48% done   2022-05-26.02:34:25\n",
      "TrfM6_aligned_NAM5.sam - 21679.504/156051.022 MB - 13.89% done   2022-05-26.02:39:48\n",
      "TrfM6_aligned_NAM5.sam - 22319.489/156051.022 MB - 14.30% done   2022-05-26.02:45:04\n",
      "TrfM6_aligned_NAM5.sam - 22919.475/156051.022 MB - 14.69% done   2022-05-26.02:50:20\n",
      "TrfM6_aligned_NAM5.sam - 23519.462/156051.022 MB - 15.07% done   2022-05-26.02:55:38\n",
      "TrfM6_aligned_NAM5.sam - 24159.447/156051.022 MB - 15.48% done   2022-05-26.03:01:10\n",
      "TrfM6_aligned_NAM5.sam - 24759.433/156051.022 MB - 15.87% done   2022-05-26.03:06:32\n",
      "TrfM6_aligned_NAM5.sam - 25399.419/156051.022 MB - 16.28% done   2022-05-26.03:12:00\n",
      "TrfM6_aligned_NAM5.sam - 26119.402/156051.022 MB - 16.74% done   2022-05-26.03:17:36\n",
      "TrfM6_aligned_NAM5.sam - 26679.389/156051.022 MB - 17.10% done   2022-05-26.03:22:53\n",
      "TrfM6_aligned_NAM5.sam - 27319.375/156051.022 MB - 17.51% done   2022-05-26.03:28:24\n",
      "TrfM6_aligned_NAM5.sam - 27959.360/156051.022 MB - 17.92% done   2022-05-26.03:34:02\n",
      "TrfM6_aligned_NAM5.sam - 28599.345/156051.022 MB - 18.33% done   2022-05-26.03:39:08\n",
      "TrfM6_aligned_NAM5.sam - 29159.333/156051.022 MB - 18.69% done   2022-05-26.03:44:13\n",
      "TrfM6_aligned_NAM5.sam - 29879.316/156051.022 MB - 19.15% done   2022-05-26.03:49:55\n",
      "TrfM6_aligned_NAM5.sam - 30439.303/156051.022 MB - 19.51% done   2022-05-26.03:55:12\n",
      "TrfM6_aligned_NAM5.sam - 31079.289/156051.022 MB - 19.92% done   2022-05-26.04:00:24\n",
      "TrfM6_aligned_NAM5.sam - 31719.274/156051.022 MB - 20.33% done   2022-05-26.04:06:00\n",
      "TrfM6_aligned_NAM5.sam - 32359.259/156051.022 MB - 20.74% done   2022-05-26.04:11:20\n",
      "TrfM6_aligned_NAM5.sam - 32999.245/156051.022 MB - 21.15% done   2022-05-26.04:16:34\n",
      "TrfM6_aligned_NAM5.sam - 33599.231/156051.022 MB - 21.53% done   2022-05-26.04:21:56\n",
      "TrfM6_aligned_NAM5.sam - 34239.216/156051.022 MB - 21.94% done   2022-05-26.04:27:21\n",
      "TrfM6_aligned_NAM5.sam - 34879.202/156051.022 MB - 22.35% done   2022-05-26.04:32:40\n",
      "TrfM6_aligned_NAM5.sam - 35519.187/156051.022 MB - 22.76% done   2022-05-26.04:38:08\n",
      "TrfM6_aligned_NAM5.sam - 36119.173/156051.022 MB - 23.15% done   2022-05-26.04:43:37\n",
      "TrfM6_aligned_NAM5.sam - 36719.160/156051.022 MB - 23.53% done   2022-05-26.04:49:10\n",
      "TrfM6_aligned_NAM5.sam - 37359.145/156051.022 MB - 23.94% done   2022-05-26.04:54:07\n",
      "TrfM6_aligned_NAM5.sam - 37999.130/156051.022 MB - 24.35% done   2022-05-26.04:59:31\n",
      "TrfM6_aligned_NAM5.sam - 38599.117/156051.022 MB - 24.73% done   2022-05-26.05:04:42\n",
      "TrfM6_aligned_NAM5.sam - 39239.102/156051.022 MB - 25.15% done   2022-05-26.05:10:09\n",
      "TrfM6_aligned_NAM5.sam - 39799.089/156051.022 MB - 25.50% done   2022-05-26.05:15:20\n",
      "TrfM6_aligned_NAM5.sam - 40359.076/156051.022 MB - 25.86% done   2022-05-26.05:20:28\n",
      "TrfM6_aligned_NAM5.sam - 40879.064/156051.022 MB - 26.20% done   2022-05-26.05:25:36\n",
      "TrfM6_aligned_NAM5.sam - 41559.049/156051.022 MB - 26.63% done   2022-05-26.05:31:05\n",
      "TrfM6_aligned_NAM5.sam - 42199.034/156051.022 MB - 27.04% done   2022-05-26.05:36:39\n",
      "TrfM6_aligned_NAM5.sam - 42879.019/156051.022 MB - 27.48% done   2022-05-26.05:42:04\n",
      "TrfM6_aligned_NAM5.sam - 43519.004/156051.022 MB - 27.89% done   2022-05-26.05:47:33\n",
      "TrfM6_aligned_NAM5.sam - 44078.991/156051.022 MB - 28.25% done   2022-05-26.05:52:47\n",
      "TrfM6_aligned_NAM5.sam - 44678.977/156051.022 MB - 28.63% done   2022-05-26.05:58:18\n",
      "TrfM6_aligned_NAM5.sam - 45278.964/156051.022 MB - 29.02% done   2022-05-26.06:03:34\n",
      "TrfM6_aligned_NAM5.sam - 45918.949/156051.022 MB - 29.43% done   2022-05-26.06:08:55\n",
      "TrfM6_aligned_NAM5.sam - 46558.934/156051.022 MB - 29.84% done   2022-05-26.06:14:20\n",
      "TrfM6_aligned_NAM5.sam - 47158.921/156051.022 MB - 30.22% done   2022-05-26.06:19:41\n",
      "TrfM6_aligned_NAM5.sam - 47798.906/156051.022 MB - 30.63% done   2022-05-26.06:25:05\n",
      "TrfM6_aligned_NAM5.sam - 48358.893/156051.022 MB - 30.99% done   2022-05-26.06:30:34\n",
      "TrfM6_aligned_NAM5.sam - 48998.879/156051.022 MB - 31.40% done   2022-05-26.06:36:14\n",
      "TrfM6_aligned_NAM5.sam - 49678.863/156051.022 MB - 31.84% done   2022-05-26.06:41:47\n",
      "TrfM6_aligned_NAM5.sam - 50198.851/156051.022 MB - 32.17% done   2022-05-26.06:47:08\n",
      "TrfM6_aligned_NAM5.sam - 50838.836/156051.022 MB - 32.58% done   2022-05-26.06:52:41\n",
      "TrfM6_aligned_NAM5.sam - 51398.824/156051.022 MB - 32.94% done   2022-05-26.06:57:46\n",
      "TrfM6_aligned_NAM5.sam - 52038.809/156051.022 MB - 33.35% done   2022-05-26.07:02:55\n",
      "TrfM6_aligned_NAM5.sam - 52638.795/156051.022 MB - 33.73% done   2022-05-26.07:08:04\n",
      "TrfM6_aligned_NAM5.sam - 53238.781/156051.022 MB - 34.12% done   2022-05-26.07:13:37\n",
      "TrfM6_aligned_NAM5.sam - 53838.768/156051.022 MB - 34.50% done   2022-05-26.07:18:55\n",
      "TrfM6_aligned_NAM5.sam - 54438.754/156051.022 MB - 34.89% done   2022-05-26.07:24:20\n",
      "TrfM6_aligned_NAM5.sam - 55078.739/156051.022 MB - 35.30% done   2022-05-26.07:30:02\n",
      "TrfM6_aligned_NAM5.sam - 55718.725/156051.022 MB - 35.71% done   2022-05-26.07:35:55\n",
      "TrfM6_aligned_NAM5.sam - 56278.712/156051.022 MB - 36.06% done   2022-05-26.07:41:15\n",
      "TrfM6_aligned_NAM5.sam - 56958.696/156051.022 MB - 36.50% done   2022-05-26.07:46:47\n",
      "TrfM6_aligned_NAM5.sam - 57558.683/156051.022 MB - 36.88% done   2022-05-26.07:52:17\n",
      "TrfM6_aligned_NAM5.sam - 58158.669/156051.022 MB - 37.27% done   2022-05-26.07:57:46\n",
      "TrfM6_aligned_NAM5.sam - 58878.652/156051.022 MB - 37.73% done   2022-05-26.08:03:39\n",
      "TrfM6_aligned_NAM5.sam - 59438.640/156051.022 MB - 38.09% done   2022-05-26.08:09:11\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrfM6_aligned_NAM5.sam - 59998.627/156051.022 MB - 38.45% done   2022-05-26.08:14:13\n",
      "TrfM6_aligned_NAM5.sam - 60638.612/156051.022 MB - 38.86% done   2022-05-26.08:19:46\n",
      "TrfM6_aligned_NAM5.sam - 61118.601/156051.022 MB - 39.17% done   2022-05-26.08:25:00\n",
      "TrfM6_aligned_NAM5.sam - 61758.586/156051.022 MB - 39.58% done   2022-05-26.08:30:10\n",
      "TrfM6_aligned_NAM5.sam - 62398.572/156051.022 MB - 39.99% done   2022-05-26.08:35:51\n",
      "TrfM6_aligned_NAM5.sam - 62998.558/156051.022 MB - 40.37% done   2022-05-26.08:41:30\n",
      "TrfM6_aligned_NAM5.sam - 63518.546/156051.022 MB - 40.70% done   2022-05-26.08:47:03\n",
      "TrfM6_aligned_NAM5.sam - 64198.531/156051.022 MB - 41.14% done   2022-05-26.08:52:49\n",
      "TrfM6_aligned_NAM5.sam - 64718.519/156051.022 MB - 41.47% done   2022-05-26.08:57:56\n",
      "TrfM6_aligned_NAM5.sam - 65358.504/156051.022 MB - 41.88% done   2022-05-26.09:03:44\n",
      "TrfM6_aligned_NAM5.sam - 65958.490/156051.022 MB - 42.27% done   2022-05-26.09:08:53\n",
      "TrfM6_aligned_NAM5.sam - 66518.478/156051.022 MB - 42.63% done   2022-05-26.09:14:12\n",
      "TrfM6_aligned_NAM5.sam - 67078.465/156051.022 MB - 42.98% done   2022-05-26.09:19:49\n",
      "TrfM6_aligned_NAM5.sam - 67678.451/156051.022 MB - 43.37% done   2022-05-26.09:25:35\n",
      "TrfM6_aligned_NAM5.sam - 68238.438/156051.022 MB - 43.73% done   2022-05-26.09:30:44\n",
      "TrfM6_aligned_NAM5.sam - 68878.423/156051.022 MB - 44.14% done   2022-05-26.09:36:31\n",
      "TrfM6_aligned_NAM5.sam - 69438.411/156051.022 MB - 44.50% done   2022-05-26.09:42:00\n",
      "TrfM6_aligned_NAM5.sam - 69958.399/156051.022 MB - 44.83% done   2022-05-26.09:47:05\n",
      "TrfM6_aligned_NAM5.sam - 70598.384/156051.022 MB - 45.24% done   2022-05-26.09:52:23\n",
      "TrfM6_aligned_NAM5.sam - 71238.369/156051.022 MB - 45.65% done   2022-05-26.09:58:09\n",
      "TrfM6_aligned_NAM5.sam - 71758.358/156051.022 MB - 45.98% done   2022-05-26.10:03:31\n",
      "TrfM6_aligned_NAM5.sam - 72358.344/156051.022 MB - 46.37% done   2022-05-26.10:08:50\n",
      "TrfM6_aligned_NAM5.sam - 72958.330/156051.022 MB - 46.75% done   2022-05-26.10:14:14\n",
      "TrfM6_aligned_NAM5.sam - 73518.317/156051.022 MB - 47.11% done   2022-05-26.10:19:25\n",
      "TrfM6_aligned_NAM5.sam - 74118.304/156051.022 MB - 47.50% done   2022-05-26.10:24:30\n",
      "TrfM6_aligned_NAM5.sam - 74718.290/156051.022 MB - 47.88% done   2022-05-26.10:30:05\n",
      "TrfM6_aligned_NAM5.sam - 75278.277/156051.022 MB - 48.24% done   2022-05-26.10:35:23\n",
      "TrfM6_aligned_NAM5.sam - 75958.261/156051.022 MB - 48.68% done   2022-05-26.10:40:41\n",
      "TrfM6_aligned_NAM5.sam - 76558.248/156051.022 MB - 49.06% done   2022-05-26.10:46:36\n",
      "TrfM6_aligned_NAM5.sam - 77118.235/156051.022 MB - 49.42% done   2022-05-26.10:52:03\n",
      "TrfM6_aligned_NAM5.sam - 77758.220/156051.022 MB - 49.83% done   2022-05-26.10:57:36\n",
      "TrfM6_aligned_NAM5.sam - 78318.207/156051.022 MB - 50.19% done   2022-05-26.11:02:58\n",
      "TrfM6_aligned_NAM5.sam - 78918.194/156051.022 MB - 50.57% done   2022-05-26.11:08:08\n",
      "TrfM6_aligned_NAM5.sam - 79478.181/156051.022 MB - 50.93% done   2022-05-26.11:13:41\n",
      "TrfM6_aligned_NAM5.sam - 80038.168/156051.022 MB - 51.29% done   2022-05-26.11:19:21\n",
      "TrfM6_aligned_NAM5.sam - 80598.155/156051.022 MB - 51.65% done   2022-05-26.11:24:42\n",
      "TrfM6_aligned_NAM5.sam - 81158.142/156051.022 MB - 52.01% done   2022-05-26.11:30:07\n",
      "TrfM6_aligned_NAM5.sam - 81878.126/156051.022 MB - 52.47% done   2022-05-26.11:36:00\n",
      "TrfM6_aligned_NAM5.sam - 82398.114/156051.022 MB - 52.80% done   2022-05-26.11:41:37\n",
      "TrfM6_aligned_NAM5.sam - 82958.101/156051.022 MB - 53.16% done   2022-05-26.11:47:04\n",
      "TrfM6_aligned_NAM5.sam - 83518.088/156051.022 MB - 53.52% done   2022-05-26.11:52:33\n",
      "TrfM6_aligned_NAM5.sam - 84118.075/156051.022 MB - 53.90% done   2022-05-26.11:58:17\n",
      "TrfM6_aligned_NAM5.sam - 84678.062/156051.022 MB - 54.26% done   2022-05-26.12:03:20\n",
      "TrfM6_aligned_NAM5.sam - 85278.048/156051.022 MB - 54.65% done   2022-05-26.12:08:33\n",
      "TrfM6_aligned_NAM5.sam - 85878.034/156051.022 MB - 55.03% done   2022-05-26.12:13:49\n",
      "TrfM6_aligned_NAM5.sam - 86438.022/156051.022 MB - 55.39% done   2022-05-26.12:19:44\n",
      "TrfM6_aligned_NAM5.sam - 86958.010/156051.022 MB - 55.72% done   2022-05-26.12:24:51\n",
      "TrfM6_aligned_NAM5.sam - 87557.996/156051.022 MB - 56.11% done   2022-05-26.12:30:28\n",
      "TrfM6_aligned_NAM5.sam - 88157.982/156051.022 MB - 56.49% done   2022-05-26.12:36:10\n",
      "TrfM6_aligned_NAM5.sam - 88797.968/156051.022 MB - 56.90% done   2022-05-26.12:41:55\n",
      "TrfM6_aligned_NAM5.sam - 89277.957/156051.022 MB - 57.21% done   2022-05-26.12:46:48\n",
      "TrfM6_aligned_NAM5.sam - 89837.944/156051.022 MB - 57.57% done   2022-05-26.12:51:31\n",
      "TrfM6_aligned_NAM5.sam - 90437.930/156051.022 MB - 57.95% done   2022-05-26.12:56:33\n",
      "TrfM6_aligned_NAM5.sam - 90997.917/156051.022 MB - 58.31% done   2022-05-26.13:02:12\n",
      "TrfM6_aligned_NAM5.sam - 91517.905/156051.022 MB - 58.65% done   2022-05-26.13:07:35\n",
      "TrfM6_aligned_NAM5.sam - 92077.893/156051.022 MB - 59.00% done   2022-05-26.13:12:39\n",
      "TrfM6_aligned_NAM5.sam - 92717.878/156051.022 MB - 59.42% done   2022-05-26.13:18:14\n",
      "TrfM6_aligned_NAM5.sam - 93277.865/156051.022 MB - 59.77% done   2022-05-26.13:23:57\n",
      "TrfM6_aligned_NAM5.sam - 93797.853/156051.022 MB - 60.11% done   2022-05-26.13:29:17\n",
      "TrfM6_aligned_NAM5.sam - 94397.839/156051.022 MB - 60.49% done   2022-05-26.13:34:49\n",
      "TrfM6_aligned_NAM5.sam - 94997.826/156051.022 MB - 60.88% done   2022-05-26.13:40:14\n",
      "TrfM6_aligned_NAM5.sam - 95477.815/156051.022 MB - 61.18% done   2022-05-26.13:45:41\n",
      "TrfM6_aligned_NAM5.sam - 95997.803/156051.022 MB - 61.52% done   2022-05-26.13:51:02\n",
      "TrfM6_aligned_NAM5.sam - 96557.790/156051.022 MB - 61.88% done   2022-05-26.13:56:16\n",
      "TrfM6_aligned_NAM5.sam - 97117.777/156051.022 MB - 62.23% done   2022-05-26.14:01:24\n",
      "TrfM6_aligned_NAM5.sam - 97677.764/156051.022 MB - 62.59% done   2022-05-26.14:06:36\n",
      "TrfM6_aligned_NAM5.sam - 98157.753/156051.022 MB - 62.90% done   2022-05-26.14:11:24\n",
      "TrfM6_aligned_NAM5.sam - 98677.741/156051.022 MB - 63.23% done   2022-05-26.14:16:26\n",
      "TrfM6_aligned_NAM5.sam - 99197.730/156051.022 MB - 63.57% done   2022-05-26.14:20:55\n",
      "TrfM6_aligned_NAM5.sam - 99677.719/156051.022 MB - 63.88% done   2022-05-26.14:25:33\n",
      "TrfM6_aligned_NAM5.sam - 100357.703/156051.022 MB - 64.31% done   2022-05-26.14:30:44\n",
      "TrfM6_aligned_NAM5.sam - 100917.690/156051.022 MB - 64.67% done   2022-05-26.14:35:32\n",
      "TrfM6_aligned_NAM5.sam - 101437.678/156051.022 MB - 65.00% done   2022-05-26.14:40:09\n",
      "TrfM6_aligned_NAM5.sam - 101997.665/156051.022 MB - 65.36% done   2022-05-26.14:44:56\n",
      "TrfM6_aligned_NAM5.sam - 102517.654/156051.022 MB - 65.69% done   2022-05-26.14:49:56\n",
      "TrfM6_aligned_NAM5.sam - 103117.640/156051.022 MB - 66.08% done   2022-05-26.14:54:45\n",
      "TrfM6_aligned_NAM5.sam - 103597.629/156051.022 MB - 66.39% done   2022-05-26.14:59:34\n",
      "TrfM6_aligned_NAM5.sam - 104157.616/156051.022 MB - 66.75% done   2022-05-26.15:04:06\n",
      "TrfM6_aligned_NAM5.sam - 104717.603/156051.022 MB - 67.10% done   2022-05-26.15:09:06\n",
      "TrfM6_aligned_NAM5.sam - 105197.592/156051.022 MB - 67.41% done   2022-05-26.15:15:36\n",
      "TrfM6_aligned_NAM5.sam - 105597.583/156051.022 MB - 67.67% done   2022-05-26.15:21:19\n",
      "TrfM6_aligned_NAM5.sam - 106117.571/156051.022 MB - 68.00% done   2022-05-26.15:26:19\n",
      "TrfM6_aligned_NAM5.sam - 106637.559/156051.022 MB - 68.34% done   2022-05-26.15:31:11\n",
      "TrfM6_aligned_NAM5.sam - 107077.549/156051.022 MB - 68.62% done   2022-05-26.15:36:09\n",
      "TrfM6_aligned_NAM5.sam - 107517.539/156051.022 MB - 68.90% done   2022-05-26.15:40:58\n",
      "TrfM6_aligned_NAM5.sam - 108077.526/156051.022 MB - 69.26% done   2022-05-26.15:45:46\n",
      "TrfM6_aligned_NAM5.sam - 108557.515/156051.022 MB - 69.57% done   2022-05-26.15:50:28\n",
      "TrfM6_aligned_NAM5.sam - 108957.506/156051.022 MB - 69.82% done   2022-05-26.15:55:22\n",
      "TrfM6_aligned_NAM5.sam - 109397.496/156051.022 MB - 70.10% done   2022-05-26.16:00:28\n",
      "TrfM6_aligned_NAM5.sam - 109877.485/156051.022 MB - 70.41% done   2022-05-26.16:04:51\n",
      "TrfM6_aligned_NAM5.sam - 110357.474/156051.022 MB - 70.72% done   2022-05-26.16:09:45\n",
      "TrfM6_aligned_NAM5.sam - 110837.463/156051.022 MB - 71.03% done   2022-05-26.16:14:34\n",
      "TrfM6_aligned_NAM5.sam - 111197.455/156051.022 MB - 71.26% done   2022-05-26.16:19:17\n",
      "TrfM6_aligned_NAM5.sam - 111677.444/156051.022 MB - 71.56% done   2022-05-26.16:24:10\n",
      "TrfM6_aligned_NAM5.sam - 112077.435/156051.022 MB - 71.82% done   2022-05-26.16:29:05\n",
      "TrfM6_aligned_NAM5.sam - 112557.424/156051.022 MB - 72.13% done   2022-05-26.16:34:21\n",
      "TrfM6_aligned_NAM5.sam - 112997.414/156051.022 MB - 72.41% done   2022-05-26.16:39:22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrfM6_aligned_NAM5.sam - 113477.403/156051.022 MB - 72.72% done   2022-05-26.16:44:27\n",
      "TrfM6_aligned_NAM5.sam - 113917.393/156051.022 MB - 73.00% done   2022-05-26.16:48:42\n",
      "TrfM6_aligned_NAM5.sam - 114397.382/156051.022 MB - 73.31% done   2022-05-26.16:53:53\n",
      "TrfM6_aligned_NAM5.sam - 114797.372/156051.022 MB - 73.56% done   2022-05-26.16:59:01\n",
      "TrfM6_aligned_NAM5.sam - 115277.362/156051.022 MB - 73.87% done   2022-05-26.17:04:06\n",
      "TrfM6_aligned_NAM5.sam - 115797.350/156051.022 MB - 74.20% done   2022-05-26.17:09:16\n",
      "TrfM6_aligned_NAM5.sam - 116277.339/156051.022 MB - 74.51% done   2022-05-26.17:14:13\n",
      "TrfM6_aligned_NAM5.sam - 116717.329/156051.022 MB - 74.79% done   2022-05-26.17:19:07\n",
      "TrfM6_aligned_NAM5.sam - 117277.316/156051.022 MB - 75.15% done   2022-05-26.17:23:42\n",
      "TrfM6_aligned_NAM5.sam - 117757.305/156051.022 MB - 75.46% done   2022-05-26.17:28:21\n",
      "TrfM6_aligned_NAM5.sam - 118117.297/156051.022 MB - 75.69% done   2022-05-26.17:32:43\n",
      "TrfM6_aligned_NAM5.sam - 118477.288/156051.022 MB - 75.92% done   2022-05-26.17:37:10\n",
      "TrfM6_aligned_NAM5.sam - 119037.275/156051.022 MB - 76.28% done   2022-05-26.17:42:06\n",
      "TrfM6_aligned_NAM5.sam - 119597.263/156051.022 MB - 76.64% done   2022-05-26.17:46:23\n",
      "TrfM6_aligned_NAM5.sam - 119877.256/156051.022 MB - 76.82% done   2022-05-26.17:50:19\n",
      "TrfM6_aligned_NAM5.sam - 120317.246/156051.022 MB - 77.10% done   2022-05-26.17:54:31\n",
      "TrfM6_aligned_NAM5.sam - 120797.235/156051.022 MB - 77.41% done   2022-05-26.17:58:58\n",
      "TrfM6_aligned_NAM5.sam - 121317.223/156051.022 MB - 77.74% done   2022-05-26.18:02:57\n",
      "TrfM6_aligned_NAM5.sam - 121637.216/156051.022 MB - 77.95% done   2022-05-26.18:07:03\n",
      "TrfM6_aligned_NAM5.sam - 122117.205/156051.022 MB - 78.25% done   2022-05-26.18:11:14\n",
      "TrfM6_aligned_NAM5.sam - 122517.196/156051.022 MB - 78.51% done   2022-05-26.18:15:12\n",
      "TrfM6_aligned_NAM5.sam - 123077.183/156051.022 MB - 78.87% done   2022-05-26.18:19:36\n",
      "TrfM6_aligned_NAM5.sam - 123437.175/156051.022 MB - 79.10% done   2022-05-26.18:23:51\n",
      "TrfM6_aligned_NAM5.sam - 123837.166/156051.022 MB - 79.36% done   2022-05-26.18:27:25\n",
      "TrfM6_aligned_NAM5.sam - 124277.156/156051.022 MB - 79.64% done   2022-05-26.18:31:38\n",
      "TrfM6_aligned_NAM5.sam - 124797.144/156051.022 MB - 79.97% done   2022-05-26.18:36:15\n",
      "TrfM6_aligned_NAM5.sam - 125157.135/156051.022 MB - 80.20% done   2022-05-26.18:40:17\n",
      "TrfM6_aligned_NAM5.sam - 125517.127/156051.022 MB - 80.43% done   2022-05-26.18:44:27\n",
      "TrfM6_aligned_NAM5.sam - 126037.115/156051.022 MB - 80.77% done   2022-05-26.18:48:43\n",
      "TrfM6_aligned_NAM5.sam - 126477.105/156051.022 MB - 81.05% done   2022-05-26.18:53:09\n",
      "TrfM6_aligned_NAM5.sam - 126877.096/156051.022 MB - 81.30% done   2022-05-26.18:57:24\n",
      "TrfM6_aligned_NAM5.sam - 127317.086/156051.022 MB - 81.59% done   2022-05-26.19:01:27\n",
      "TrfM6_aligned_NAM5.sam - 127797.075/156051.022 MB - 81.89% done   2022-05-26.19:05:59\n",
      "TrfM6_aligned_NAM5.sam - 128277.064/156051.022 MB - 82.20% done   2022-05-26.19:10:15\n",
      "TrfM6_aligned_NAM5.sam - 128637.056/156051.022 MB - 82.43% done   2022-05-26.19:14:57\n",
      "TrfM6_aligned_NAM5.sam - 129077.046/156051.022 MB - 82.71% done   2022-05-26.19:19:15\n",
      "TrfM6_aligned_NAM5.sam - 129597.034/156051.022 MB - 83.05% done   2022-05-26.19:23:42\n",
      "TrfM6_aligned_NAM5.sam - 130037.024/156051.022 MB - 83.33% done   2022-05-26.19:28:13\n",
      "TrfM6_aligned_NAM5.sam - 130397.015/156051.022 MB - 83.56% done   2022-05-26.19:32:16\n",
      "TrfM6_aligned_NAM5.sam - 130837.005/156051.022 MB - 83.84% done   2022-05-26.19:36:18\n",
      "TrfM6_aligned_NAM5.sam - 131236.996/156051.022 MB - 84.10% done   2022-05-26.19:40:47\n",
      "TrfM6_aligned_NAM5.sam - 131636.987/156051.022 MB - 84.36% done   2022-05-26.19:45:11\n",
      "TrfM6_aligned_NAM5.sam - 132036.978/156051.022 MB - 84.61% done   2022-05-26.19:50:26\n",
      "TrfM6_aligned_NAM5.sam - 132516.967/156051.022 MB - 84.92% done   2022-05-26.19:55:13\n",
      "TrfM6_aligned_NAM5.sam - 132916.958/156051.022 MB - 85.18% done   2022-05-26.19:59:44\n",
      "TrfM6_aligned_NAM5.sam - 133356.948/156051.022 MB - 85.46% done   2022-05-26.20:05:30\n",
      "TrfM6_aligned_NAM5.sam - 133876.936/156051.022 MB - 85.79% done   2022-05-26.20:10:28\n",
      "TrfM6_aligned_NAM5.sam - 134276.927/156051.022 MB - 86.05% done   2022-05-26.20:15:05\n",
      "TrfM6_aligned_NAM5.sam - 134676.917/156051.022 MB - 86.30% done   2022-05-26.20:20:06\n",
      "TrfM6_aligned_NAM5.sam - 135070.315/156051.022 MB - 86.56% done   2022-05-26.20:25:00\n",
      "TrfM6_aligned_NAM5.sam - 135463.728/156051.022 MB - 86.81% done   2022-05-26.20:29:08\n",
      "TrfM6_aligned_NAM5.sam - 135730.534/156051.022 MB - 86.98% done   2022-05-26.20:32:26\n",
      "TrfM6_aligned_NAM5.sam - 156051.022/156051.022 MB - 100.00% done   2022-05-27.01:48:57\n",
      "ERROR: [-]\tiRODS/lib/core/src/procApiRequest.cpp:266:readAndProcApiReply :  status [SYS_HEADER_READ_LEN_ERR]  errno [Operation now in progress] -- message []\n",
      "\t[-]\tiRODS/lib/core/src/sockComm.cpp:199:readMsgHeader :  status [SYS_HEADER_READ_LEN_ERR]  errno [Operation now in progress] -- message [failed to call 'read header']\n",
      "\t\t[-]\tlibtcp.cpp:240:tcp_read_msg_header :  status [SYS_HEADER_READ_LEN_ERR]  errno [Operation now in progress] -- message [read 0 expected 4]\n",
      "\n",
      "ERROR: putUtil: put error for /iplant/home/annis/WGS/WGS_2019/TRF/Trf_WGS_NAM5_alignment_May2022/aligned_reads/TrfM6_aligned_NAM5.sam, status = -4036 status = -4036 SYS_HEADER_READ_LEN_ERR, Operation now in progress\n"
     ]
    },
    {
     "ename": "",
     "evalue": "3",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "\n",
    "icd ./aligned_reads \n",
    "iput -rKP /Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022/aligned_reads/TrfM6_aligned_NAM5.sam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t308452471\t49007924\t983852\n",
      "chr2\t243675191\t37932318\t778013\n",
      "chr3\t238017767\t35001532\t730749\n",
      "chr4\t250330460\t37210953\t766957\n",
      "chr5\t226353449\t42695268\t738740\n",
      "chr6\t181357234\t29472465\t575622\n",
      "chr7\t185808916\t30703292\t598820\n",
      "chr8\t182411202\t30995844\t592987\n",
      "chr9\t163004744\t28206966\t559459\n",
      "chr10\t152435371\t22540954\t489427\n"
     ]
    }
   ],
   "source": [
    "# Check names of chromosomes in the indexed bam file \n",
    "samtools idxstats ./BAM_reads/TrfM6_aligned_NAM5.bam | head -n 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "[main_vcfview] Error: cannot write to (null)\n"
     ]
    },
    {
     "ename": "",
     "evalue": "255",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mkdir ./mpileup_files\n",
    "mkdir ./variant_calls\n",
    "##Generate a variant file using bcftools mpileup and call the variants (this is just variants in Trf Mutant compared to B73) \n",
    "#-f means an input file is in fasta format\n",
    "#-Ob means output in a compressed bcf format\n",
    "#-mv is call short indels \n",
    "#-o is \n",
    "#-r restricts the pileup to a region\n",
    "bcftools mpileup -f /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0.fa ./BAM_reads/TrfM6_aligned_NAM5.bam | bcftools call -mv -Ob -o ./mpileup_files/TrfM6_WGS_NAM5_mpileup.bcf\n",
    "#Convert .bcf to .vcf\n",
    "bcftools view ./mpileup_files/TrfM6_WGS_NAM5_mpileup.bcf > ./variant_calls/TrfM6_WGS_NAM5_mpileup.vcf\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Convert .bcf to .vcf\n",
    "bcftools view ./mpileup_files/TrfM6_WGS_NAM5_mpileup.bcf > ./variant_calls/TrfM6_WGS_NAM5_mpileup.vcf\n",
    "\n",
    "# This file ended up massive- why I don't know, piped it through the python VCF code to get the .txt file for the next steps in R "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/iplant/home/annis\n"
     ]
    }
   ],
   "source": [
    "ipwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0/3 -  0.00% of files done   0.000/162358.690 MB -  0.00% of file sizes done\n",
      "Processing .DS_Store - 0.006 MB   2022-05-28.09:23:05\n",
      "1/3 - 33.33% of files done   0.006/162358.690 MB -  0.00% of file sizes done\n",
      "Processing TrfM6_WGS_NAM5_mpileup.txt - 35289.282 MB   2022-05-28.09:23:07\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 639.985/35289.282 MB -  1.81% done   2022-05-28.09:30:05\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 1159.973/35289.282 MB -  3.29% done   2022-05-28.09:35:48\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 1759.960/35289.282 MB -  4.99% done   2022-05-28.09:41:20\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 2359.946/35289.282 MB -  6.69% done   2022-05-28.09:46:55\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 2799.936/35289.282 MB -  7.93% done   2022-05-28.09:52:02\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 3479.920/35289.282 MB -  9.86% done   2022-05-28.09:57:46\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 4159.905/35289.282 MB - 11.79% done   2022-05-28.10:03:34\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 4759.891/35289.282 MB - 13.49% done   2022-05-28.10:09:07\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 5439.875/35289.282 MB - 15.42% done   2022-05-28.10:14:47\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 6119.860/35289.282 MB - 17.34% done   2022-05-28.10:20:35\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 6759.845/35289.282 MB - 19.16% done   2022-05-28.10:26:18\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 7399.831/35289.282 MB - 20.97% done   2022-05-28.10:32:03\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 8039.816/35289.282 MB - 22.78% done   2022-05-28.10:37:35\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 8759.800/35289.282 MB - 24.82% done   2022-05-28.10:43:28\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 9319.787/35289.282 MB - 26.41% done   2022-05-28.10:48:49\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 9959.772/35289.282 MB - 28.22% done   2022-05-28.10:54:32\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 10599.757/35289.282 MB - 30.04% done   2022-05-28.11:00:42\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 11279.742/35289.282 MB - 31.96% done   2022-05-28.11:06:55\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 11999.725/35289.282 MB - 34.00% done   2022-05-28.11:12:51\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 12679.710/35289.282 MB - 35.93% done   2022-05-28.11:18:50\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 13359.694/35289.282 MB - 37.86% done   2022-05-28.11:24:52\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 13999.680/35289.282 MB - 39.67% done   2022-05-28.11:30:46\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 14599.666/35289.282 MB - 41.37% done   2022-05-28.11:36:17\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 15279.650/35289.282 MB - 43.30% done   2022-05-28.11:41:57\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 15919.636/35289.282 MB - 45.11% done   2022-05-28.11:47:39\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 16399.625/35289.282 MB - 46.47% done   2022-05-28.11:52:47\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 17039.610/35289.282 MB - 48.29% done   2022-05-28.11:58:26\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 17759.594/35289.282 MB - 50.33% done   2022-05-28.12:04:11\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 18359.580/35289.282 MB - 52.03% done   2022-05-28.12:09:36\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 18999.565/35289.282 MB - 53.84% done   2022-05-28.12:15:21\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 19599.551/35289.282 MB - 55.54% done   2022-05-28.12:20:48\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 20239.537/35289.282 MB - 57.35% done   2022-05-28.12:26:20\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 20879.522/35289.282 MB - 59.17% done   2022-05-28.12:31:45\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 21479.508/35289.282 MB - 60.87% done   2022-05-28.12:36:59\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 22119.494/35289.282 MB - 62.68% done   2022-05-28.12:42:22\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 22719.480/35289.282 MB - 64.38% done   2022-05-28.12:47:31\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 23359.465/35289.282 MB - 66.19% done   2022-05-28.12:53:02\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 23999.451/35289.282 MB - 68.01% done   2022-05-28.12:58:26\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 24599.437/35289.282 MB - 69.71% done   2022-05-28.13:03:46\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 25239.422/35289.282 MB - 71.52% done   2022-05-28.13:09:25\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 25879.408/35289.282 MB - 73.34% done   2022-05-28.13:15:08\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 26519.393/35289.282 MB - 75.15% done   2022-05-28.13:20:48\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 27239.377/35289.282 MB - 77.19% done   2022-05-28.13:26:18\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 27839.363/35289.282 MB - 78.89% done   2022-05-28.13:31:40\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 28519.347/35289.282 MB - 80.82% done   2022-05-28.13:37:23\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 29159.333/35289.282 MB - 82.63% done   2022-05-28.13:42:57\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 29799.318/35289.282 MB - 84.44% done   2022-05-28.13:48:13\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 30479.302/35289.282 MB - 86.37% done   2022-05-28.13:53:42\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 31119.288/35289.282 MB - 88.18% done   2022-05-28.13:58:59\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 31759.273/35289.282 MB - 90.00% done   2022-05-28.14:05:06\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 32439.258/35289.282 MB - 91.92% done   2022-05-28.14:11:07\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 32959.246/35289.282 MB - 93.40% done   2022-05-28.14:16:52\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 33559.232/35289.282 MB - 95.10% done   2022-05-28.14:22:22\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 34199.217/35289.282 MB - 96.91% done   2022-05-28.14:28:13\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 34918.615/35289.282 MB - 98.95% done   2022-05-28.14:34:00\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 35049.874/35289.282 MB - 99.32% done   2022-05-28.14:34:29\n",
      "TrfM6_WGS_NAM5_mpileup.txt - 35289.282/35289.282 MB - 100.00% done   2022-05-28.14:38:11\n",
      "2/3 - 66.67% of files done   35289.286/162358.690 MB - 21.74% of file sizes done\n",
      "Processing TrfM6_WGS_NAM5_mpileup.vcf - 127069.404 MB   2022-05-28.14:43:43\n",
      "ERROR: putDirUtil: put /Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022/variant_calls/TrfM6_WGS_NAM5_mpileup.vcf failed. status = -115000 status = -115000 SYS_SOCK_READ_TIMEDOUT\n",
      "ERROR: putUtil: put error for /iplant/home/annis/WGS/WGS_2019/TRF/Trf_WGS_NAM5_alignment_May2022/variant_calls, status = -115000 status = -115000 SYS_SOCK_READ_TIMEDOUT\n"
     ]
    },
    {
     "ename": "",
     "evalue": "3",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "icd /iplant/home/annis/WGS/WGS_2019/TRF/Trf_WGS_NAM5_alignment_May2022\n",
    "iput -rKP /Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022/variant_calls\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_B73v3\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd /Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed to read from ./variant_calls/TrfM6_WGS_NAM5_mpileup.vcf: not compressed with bgzip\n"
     ]
    },
    {
     "ename": "",
     "evalue": "255",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "#Need to extract just chromosome 10 as a vcf file to use in SnpEff anaysis downstream \n",
    "bcftools view ./variant_calls/TrfM6_WGS_NAM5_mpileup.vcf --regions chr10 > ./variant_calls/TrfM6_WGS_NAM5_mpileup_chr10.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "bcftools view ./variant_calls/TrfM6_WGS_NAM5_mpileup.vcf -Oz -o ./variant_calls/TrfM6_WGS_NAM5_mpileup.vcf.gz\n",
    "\n",
    "# Oz means output file type zip "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for './variant_calls/TrfM6_WGS_NAM5_mpileup.vcf.gz'\n",
      "Failed to read from ./variant_calls/TrfM6_WGS_NAM5_mpileup.vcf.gz: could not load index\n"
     ]
    },
    {
     "ename": "",
     "evalue": "255",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "#Need to extract just chromosome 10 as a vcf file to use in SnpEff anaysis downstream \n",
    "bcftools view ./variant_calls/TrfM6_WGS_NAM5_mpileup.vcf.gz --regions chr10 > ./variant_calls/TrfM6_WGS_NAM5_mpileup_chr10.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Need to index the vcf file to pull the chr10 region only. \n",
    "bcftools index ./variant_calls/TrfM6_WGS_NAM5_mpileup.vcf.gz "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Need to extract just chromosome 10 as a vcf file to use in SnpEff anaysis downstream \n",
    "bcftools view ./variant_calls/TrfM6_WGS_NAM5_mpileup.vcf.gz --regions chr10 > ./variant_calls/TrfM6_WGS_NAM5_mpileup_chr10.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n"
     ]
    }
   ],
   "source": [
    "##Generate a variant file using bcftools mpileup and call the variants (this is just variants in Trf Mutant compared to B73) \n",
    "#-f means an input file is in fasta format\n",
    "#-Ob means output in a compressed bcf format\n",
    "#-mv is call short indels \n",
    "#-o is output name\n",
    "#-r restricts the pileup to a region\n",
    "bcftools mpileup -f /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0.fa ./BAM_reads/TrfN5_aligned_NAM5.bam | bcftools call -mv -Ob -o ./mpileup_files/TrfN5_WGS_NAM5_mpileup.bcf\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Convert .bcf to compressed .vcf\n",
    "bcftools view ./mpileup_files/TrfN5_WGS_NAM5_mpileup.bcf -Oz -o ./variant_calls/TrfN5_WGS_NAM5_mpileup.vcf.gz\n",
    "#Need to index the compressed vcf file to pull the chr10 region only. \n",
    "bcftools index ./variant_calls/TrfN5_WGS_NAM5_mpileup.vcf.gz "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Need to extract just chromosome 10 as a vcf file to use in SnpEff anaysis downstream \n",
    "bcftools view ./variant_calls/TrfN5_WGS_NAM5_mpileup.vcf.gz --regions chr10 > ./variant_calls/TrfN5_WGS_NAM5_mpileup_chr10.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[W::hts_idx_load3] The index file is older than the data file: /Volumes/Elements/Sequence_Data/WGS_data/Trf_mutant_WGS_2022/WGS_Alignment_NAM5_May2022/BAM_reads/TrfM6_aligned_NAM5.bam.bai\n",
      "chr1\t308452471\t49007924\t983852\n",
      "chr2\t243675191\t37932318\t778013\n",
      "chr3\t238017767\t35001532\t730749\n",
      "chr4\t250330460\t37210953\t766957\n",
      "chr5\t226353449\t42695268\t738740\n",
      "chr6\t181357234\t29472465\t575622\n",
      "chr7\t185808916\t30703292\t598820\n",
      "chr8\t182411202\t30995844\t592987\n",
      "chr9\t163004744\t28206966\t559459\n",
      "chr10\t152435371\t22540954\t489427\n",
      "*\t0\t0\t6632500\n",
      "[W::bam_hdr_read] EOF marker is absent. The input is probably truncated\n",
      "[W::hts_idx_load3] The index file is older than the data file: /Volumes/Elements/Sequence_Data/WGS_data/Trf_mutant_WGS_2022/WGS_Alignment_NAM5_May2022/BAM_reads/TrfN5_aligned_NAM5.bam.bai\n",
      "chr1\t308452471\t49272480\t1008286\n",
      "chr2\t243675191\t38084747\t800110\n",
      "chr3\t238017767\t35374720\t749425\n",
      "chr4\t250330460\t37652999\t787226\n",
      "chr5\t226353449\t42462171\t750808\n",
      "chr6\t181357234\t29758898\t588966\n",
      "chr7\t185808916\t30681126\t610481\n",
      "chr8\t182411202\t31036139\t607200\n",
      "chr9\t163004744\t27924947\t569230\n",
      "chr10\t152435371\t23256488\t492359\n",
      "*\t0\t0\t6616050\n"
     ]
    }
   ],
   "source": [
    "# Check names of chromosomes in the indexed bam file \n",
    "samtools idxstats /Volumes/Elements/Sequence_Data/WGS_data/Trf_mutant_WGS_2022/WGS_Alignment_NAM5_May2022/BAM_reads/TrfM6_aligned_NAM5.bam | head -n 15\n",
    "samtools idxstats /Volumes/Elements/Sequence_Data/WGS_data/Trf_mutant_WGS_2022/WGS_Alignment_NAM5_May2022/BAM_reads/TrfN5_aligned_NAM5.bam | head -n 15\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">chr1\n",
      "TCATGGCTATTTTCATAAAAAATGGGGGTTGTGTGGCCATTTATCATCGACTAGAGGCTCATAAACCTCACCCCACATAT\n",
      "GTTTCCTTGCCATAGATTACATTCTTGGATTTCTGGTGGAAACCATTTCTTGCTTAAAAACTCGTACGTGTTAGCCTTCG\n",
      "GTATTATTGAAAATGGTCATTCATGGCTATTTTTCGGCAAAATGGGGGTTGTGTGGCCATTGATCGTCGACCAGAGGCTC\n",
      "ATACACCTCACCCCACATATGTTTCCTTGTCGTAGATCACATTCTTGGATTTCTGGTGGAGACCATTTCTTGGTCAGAAA\n",
      "TCCGTAGGTGTTAGCCTTCGATATTATTGAAAATGGTCGTTCATGGCTATTTTCGACAAAAATGGGGGTTGTGTGGCCAT\n",
      "TGATCATCGACCAGAGGCTCATACACCTCACCCCACATATGTTTCCTTGCCATAGATCACATTCTTGGATTTCTGGTGGA\n",
      "GACCATTTCTTGGTCAAAAATCCGTAGGTGTTAGCCTTCGGTATTATTGTAAATGGTCATTCATGGCTATTTTCGACAAA\n",
      "AATGGGGGTTGTGTGGCCATTGATCATCGACCAGAGGCTCATACACCTCACCCCACATATGTTTCCTTGCCATAGATCAC\n",
      "ATTCTTGGATTTATGGTGGAGACCATTTCTTGGTCAAAAATCCGTAGGTGTTAGCCTTCGGTATTATTGTAAATGGTCAT\n"
     ]
    }
   ],
   "source": [
    "head /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001f\u0004\u0000\u0000\u0000\u0000\u0000�\u0006\u0000BC\u0002\u0000\u0012\u0002�����@\u0014�\u0007$�\u001c",
      "\u0004\u0005��ڎb<W_�*�&��Ą\u0004(\u0010�5vfw-�vvƫ��hy\n",
      "��\\�y�\u0003�\u001a��1�=�\u0001n�XHH[�h��Ι����h~��U\u0000�\u0007\u0013�E8 \u000e�VO\u0006IQ�u��R�����*\u001c",
      "$ǊX�p�p�]�}�\u0001�\u0002�3�wI�\u000b",
      "X\n",
      "��0ק�w�\u000e�\u0000\u0007}�u@p\u000e('�b�\u0001�\u001ax\f",
      "c���S\u0004���3�+���֣� .ve*�/�&)�p�[���\u001e",
      ":�\n",
      "m�DlP�権�b�\u001e",
      "�P�\u0010��n��c��\u0004b\u001b�lj���m��FϵT\u001a\t�&ǂp4)��L�F\u0013Q\n",
      "�#5���LFG2/2\u0019����^fp�3��>�.���\u0014�[�:�ٕ��z�\\GJ��F���܋ڻ���\"�!1��J��J%v\u001b�K�\u0014Y\u001c",
      "-�s(ty�\u001c",
      "�ِ^��^t�\u0019�yeQlt\u0015���\u001b(qH�g�m]�҆�\u00199\u001e",
      "��o%��\f",
      "M�����\u0001\u0000��U}8ؽ�o7�\u0002\u001f��\u0006~|E���է_�p���on6�\u0003\u001f\u001e",
      "�5�\u0007w�;7\u001a\u0011��ѷ�����u�\u0019������?[�p�y\u0003\u0000\u0000\u001f\u0004\u0000\u0000\u0000\u0000\u0000�\u0006\u0000BC\u0002\u0000�0�}{�%Wy_�[RK��0�n�\u0005;�\\i���sNw��ĥ�w�\u0019�`��K�\u0002#$\u0016\u0010B\u0002\t8��;\u001e",
      "�S�#\u0011�$�JU\"�*�r�I���$N�7�6~\u0006l\n",
      "\u0000t=\u0000\u0014N�6��i8M\u0001���^zֻ�`��n=^\u0015���?\u001c",
      "B�\u0003\u0003��\f",
      "A��Rx���\u00158���������Ax�C���q\u001e",
      "2bI\u0006o�;Q`>^�c�t�7X���\u0002滫�\u0019b� �u�&k�\u000b",
      "��\u0018@����*\u0002�U�<\u0001�q�'ḅ�\u0014�K\u0011��kԾ��vHS����\u0004ǋ\u001c",
      "��\u0018���0\u0019\u0001S��\t\u0007�6��f�\u001b\\\u000e�\u0014�͡��\u0000\u0010�ֹ;\u00005��J���S\u001a�S��}��Q�>Oь�W��;���\u0019(�\u0019ln\u0016G!@\u0016f\u0000\u0019��sp<t\f",
      "\u0004�`\u000f�:�68�\u000b",
      "?���FR��S����s�[\u0000��\u001a��\u0019Is�������e{�\u001c",
      "\u0017�z�ט��Gs�W�\u0002�\u0012��\u0007��\u000b",
      "��N�'6��7\u001f��\u0013uJ/�5�3>�����\u0017\n",
      "\u0007�E�6��3m/�~\u001e",
      "P~\u00126�S\u0006i���Cj3g@�\u0013O\u0013�ȸw)����ν�As\n",
      "�\u001c",
      "Y$#�PF��\n",
      "ʽ@�k��\u0003\u001f\u001d",
      " �\u00053\f",
      "�%pC��gj�\u000eP�\f",
      "�:h��\\b�o'\u0017^��D�kN,�k�\u0002QG\u00139a�\u0000.j����\t\u0003�}p�S\u000e��$\u0019�H�\u0004DL(S��\t\n",
      "IU\u0012�\u000fhI\u0015'�)C��~�N�`\u0013hn\u000b",
      "��<\u0001�%E��\u0006����J7+��煏��mTڮ�w�\u0014��\u0007�\u001f\u0002��\u001a��\u0000�~|����\u0000�q\u0004\u0019��m.��+%�\u001b\u001a�k)�Fݼ۠���X;Ԋ6\u0017!R��F\"�c�яw��z\u0005�͵A?�Z��s��\u0018��t����1��P_l�8lٴ�woZ\u0000\u0016N�Ld#�B�i�k�6���Y\u0017]N���k��S��K\u001a�pu\u0005s�C�\u0014!��(\u0001�%���{�(�����-��讝[�\\�5��y�R�q\u0015��P��\u001c",
      "�\u0019\t0��,\u0003}\u0007\u0018d�]:67u^�<\u0010\u0014[�CNd1��\u0014��Jo�e8�\"Iл \u0001\n"
     ]
    }
   ],
   "source": [
    "head /Volumes/Elements/Sequence_Data/WGS_data/Trf_mutant_WGS_2022/WGS_Alignment_NAM5_May2022/BAM_reads/TrfM6_aligned_NAM5.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "�$c��G�!R!�BYH���ܐ\u0014m��\u0002���*�{pY\u0010Rr\u000e+�\"L&��r�.��\u001c",
      "��D���)ج\u0016ɂ�BR1�\u001br8:��ɹ�;\t�N<�V�&�x%i0�G�M�\u0018\u0013�ez6!a��\\\u001c",
      "�b��H�_�\u001b�ĈcM\u000b",
      "I-\u001b��c�\u0005QL9��\u0014��j�\u00076k��1�Y��\u0015��G�\u0010S���Ti��W\u0018\u0016Q�;bù��\u0012Svۉڷq����i!�څd���U,�\u001a��\u0005��ɸ�D�).\"�b\f",
      "V���\u0017$�8��\u001a.��ɝX��Bn�\u001a\u0017�\u001a͆�Ȝ�q1�p�`\u000b",
      "2s���\t�����Į�F\"���\"J2�N$�\u0006�\u0017)��L��M��(i�F�\u0003��g��&��\u001aՈ\n",
      "�F\"v��\"�r��d�M\u0017�v��\u001c",
      "��\"�\u001e",
      "�����\\�8�lr*$��I,�5\u0012Iݢ��\\ɬif#\u001a%ɂ&'��YʜI~�;���I��\u0019edŃ�01�`1?\u0013\u0013Nn\u0006�\u0014]eH�p�\"�\"�dT\u0006�-�*,R�٩\n",
      "1�d�ta%�E�Br���D\f",
      "�,����Ē�\u0019���=#\u0011��3ev�3�\u0015f\u0004�I�\"����2\u001c",
      "��Ke\u0016l\u000b",
      "�\u0002&Y!ZH�a\u0005Q!�\u001e",
      ",�+�l\u0015��'��TH�{�P\u0012V񔉹�\u0019�u \t�e���X\u0019�錉JT,XĬ,\u0012�aS��K%a\"�VT��8�$Nd��J�Dl�<5\u0002[��0��Gt�'%�`F\u0013\u0006�\u0001�STV�\u000f\u000b",
      "IɅK��da���?���$\u0016D�}b<��,��Z~SY.eEJ%5@���\u0012���\u0013*�.�<��`�Ae��3O\n",
      "��wB�A2\u000b",
      "�,\u001e",
      "X�U��Y\u0010�$��\u0002y\u0011���:��q�D#5~��H�u�`$�;gro�r�S4\u0012aÄ�X�evܨ�oF\"��\u0018$���Fm$�w��6\u0005s��8�h�f#^:ٍ����:�yp�;Q\u0019�j\"5�\u0011�V��4⧃=M���P�\u001c",
      "3�D?d'\u0011QX^9\u0012G=�E���D\f",
      "�ަ\u0018)�s��\u001e",
      "j\u0006[���$S��8�q0\n",
      "+֍D�>X\u001c",
      "f�lqx$��x`�b�q$Yj��gdy��N��ސGV�\u0005'\u0011\u001f�}=b��\u0012F�H,e��#1��\u0006R�Zi8��/�\u0003��\u0003x��mJxb{|�\u0012I\u0015���d�l\"�\u001f�\u0005���6�Ha�-��H�\u0012�Dn?4'�\n",
      "�eK�4\u0019'g�\u0007���T0��4b��hD���IZ�o����� n*�9�d^����kj�Ok��Xv���$�\u000b",
      "i���+I��N�`�P\u0017Ҫ�I�\\�\u000b",
      "�E2_�}9\f",
      "�H��a��-�/zs\n",
      "�V��`\"����\u0002�KſE�\u000b",
      "���}��O\u001c",
      "Jb�^r\u001aIѪ��å���dr�c_���\"O�-$%��Iiх���r��>�E�\u0014\u0017؋I[H�q>����gZ�CV0�Ⱦ\u0016�,���NFb�y���0N�bZ��#�m�Kf���,�9��k��ᘅ�~<\u0016��\u0013r�m1)�zE��D�&\u001fY��\u0010��\u0018\u0016��&�S\u001e",
      "�k�|�h��L\u0016�\u0003��\u0012-w\n",
      "NcO\u0006�&k?q�o��q+b��\u0001���2_�=��\u00155���sQF\u000b",
      "�\u001b)���F?H���Jt�B_�<\u0019\u001d",
      "���h:�;�1�\f",
      "=�\u0007\u0019I\u0004���52\u0015���0��^�M�1��C�0��\u001e",
      "&�5\u0013�|\u0013�t��B\"f��\u0017٣������D���X�4z�\u001b��0�\u0019Y�\u001c",
      "��H��$V��4\u0016\u001c",
      "F瓈3\u0017�c߈���4o]�`ZH��摴}���C�Y\u0011{����\u0016�ʪλ�X�Au\u001a{5\n",
      "��<֔�$���EI��Ԧ5����.c#^ߕ�}�qGKI��(�����Gq?;�\u0002�\u0015l\"���+Nė&�T\u001c",
      "�O���R��!Z1�\u0006��E��DÁE�\u0014Xϱ��Zz�~;��l�\u001b1��F��I��mV���R0I'�ܓӼ��<�x\u0007'���e3�&�`~=�\"7�#q�,uo����g�t�y\u0015S�D�^g���\u0019k*d�\u0005�\u0015\u001f��\u00071oF$o,q��H����D�.��-x�Y֖d\u001e",
      ":[�άW,W���0��̬���\u0016=j&�f#\u0017\u0014,=�\u0013i�J�c�Y1`]J�8��f���b������y�Hm�l�\u0016IBR\u0006��Z���۪g�c�9K3L��.�J3F��W,5W�!�0��\u0006��\u0004^tN��\u001e",
      "��a6�%�.��Nd �/{��8���<ykD����Y���kΜI�ԉE�b��Z��Kl��Y!\f",
      "��\f",
      "�)PY_u�Tڴ?�5�p�\u001c",
      "qߍ�CJ�[\u001c",
      "�[���Ҕ��\u0013�\n",
      "GlEn\u0006�c�[P\u0018��R?PW\\�\u0007�1�5�͔��\u0000�4\u0003\u000fX�5�\u000e���\\�\u000f�A����t� �[�}|?紵�:��\u000b",
      "w]��}[\\{�Y�~�N'oJ����3N>a�e�vlۼu���{6/��uŞ͗lݳt�f\u001f�������+�_�s�\n"
     ]
    }
   ],
   "source": [
    "head /Volumes/Elements/Sequence_Data/WGS_data/Trf_mutant_WGS_2022/WGS_Alignment_B73v3/BAM_reads/TrfM6_B73v3.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t308452471\t6\t80\t81\n",
      "chr2\t243675191\t312308139\t80\t81\n",
      "chr3\t238017767\t559029276\t80\t81\n",
      "chr4\t250330460\t800022272\t80\t81\n",
      "chr5\t226353449\t1053481869\t80\t81\n",
      "chr6\t181357234\t1282664743\t80\t81\n",
      "chr7\t185808916\t1466288949\t80\t81\n",
      "chr8\t182411202\t1654420483\t80\t81\n",
      "chr9\t163004744\t1839111832\t80\t81\n",
      "chr10\t152435371\t2004154143\t80\t81\n"
     ]
    }
   ],
   "source": [
    "head /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0.fa.fai"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash: print: command not found\n"
     ]
    },
    {
     "ename": "",
     "evalue": "127",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "\n",
    "print /Users/aricha14/Documents/Databases/Maize_Genome_B73v3/Zea_mays.AGPv3.30.dna.genome.fa.fai"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[E::fai_build_core] Format error, unexpected \"T\" at line 56234\n",
      "[faidx] Could not build fai index /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0.fa.fai\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "# try re-indexing the fasta file \n",
    " samtools faidx /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Try downloading the fasta again\n",
    "# Unzip the fasta file\n",
    "gunzip /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "scaf_21\n",
      "scaf_22\n",
      "scaf_23\n",
      "scaf_24\n",
      "scaf_25\n",
      "scaf_26\n",
      "scaf_27\n",
      "scaf_28\n",
      "scaf_29\n",
      "scaf_30\n",
      "scaf_31\n",
      "scaf_32\n",
      "scaf_33\n",
      "scaf_34\n",
      "scaf_35\n",
      "scaf_36\n",
      "scaf_37\n",
      "scaf_38\n",
      "scaf_39\n",
      "scaf_40\n",
      "scaf_41\n",
      "scaf_42\n",
      "scaf_43\n",
      "scaf_44\n",
      "scaf_45\n",
      "scaf_46\n",
      "scaf_47\n",
      "scaf_48\n",
      "scaf_49\n",
      "scaf_50\n",
      "scaf_51\n",
      "scaf_52\n",
      "scaf_53\n",
      "scaf_54\n",
      "scaf_55\n",
      "scaf_56\n",
      "scaf_57\n",
      "scaf_58\n",
      "scaf_59\n",
      "scaf_60\n",
      "scaf_61\n",
      "scaf_62\n",
      "scaf_63\n",
      "scaf_64\n",
      "scaf_65\n",
      "scaf_66\n",
      "scaf_67\n",
      "scaf_68\n",
      "scaf_69\n",
      "scaf_70\n",
      "scaf_71\n",
      "scaf_72\n",
      "scaf_73\n",
      "scaf_74\n",
      "scaf_75\n",
      "scaf_76\n",
      "scaf_77\n",
      "scaf_78\n",
      "scaf_79\n",
      "scaf_80\n",
      "scaf_81\n",
      "scaf_82\n",
      "scaf_83\n",
      "scaf_84\n",
      "scaf_85\n",
      "scaf_86\n",
      "scaf_87\n",
      "scaf_88\n",
      "scaf_89\n",
      "scaf_90\n",
      "scaf_91\n",
      "scaf_92\n",
      "scaf_93\n",
      "scaf_94\n",
      "scaf_95\n",
      "scaf_96\n",
      "scaf_97\n",
      "scaf_98\n",
      "scaf_99\n",
      "scaf_100\n",
      "scaf_101\n",
      "scaf_102\n",
      "scaf_103\n",
      "scaf_104\n",
      "scaf_105\n",
      "scaf_106\n",
      "scaf_107\n",
      "scaf_108\n",
      "scaf_109\n",
      "scaf_110\n",
      "scaf_111\n",
      "scaf_112\n",
      "scaf_113\n",
      "scaf_114\n",
      "scaf_115\n",
      "scaf_116\n",
      "scaf_117\n",
      "scaf_118\n",
      "scaf_119\n",
      "scaf_120\n",
      "scaf_121\n",
      "scaf_122\n",
      "scaf_123\n",
      "scaf_124\n",
      "scaf_125\n",
      "scaf_126\n",
      "scaf_127\n",
      "scaf_128\n",
      "scaf_129\n",
      "scaf_130\n",
      "scaf_131\n",
      "scaf_132\n",
      "scaf_133\n",
      "scaf_134\n",
      "scaf_135\n",
      "scaf_136\n",
      "scaf_137\n",
      "scaf_138\n",
      "scaf_139\n",
      "scaf_140\n",
      "scaf_141\n",
      "scaf_142\n",
      "scaf_143\n",
      "scaf_144\n",
      "scaf_145\n",
      "scaf_146\n",
      "scaf_147\n",
      "scaf_148\n",
      "scaf_149\n",
      "scaf_150\n",
      "scaf_151\n",
      "scaf_152\n",
      "scaf_153\n",
      "scaf_154\n",
      "scaf_155\n",
      "scaf_156\n",
      "scaf_157\n",
      "scaf_158\n",
      "scaf_159\n",
      "scaf_160\n",
      "scaf_161\n",
      "scaf_162\n",
      "scaf_163\n",
      "scaf_164\n",
      "scaf_165\n",
      "scaf_166\n",
      "scaf_167\n",
      "scaf_168\n",
      "scaf_169\n",
      "scaf_170\n",
      "scaf_171\n",
      "scaf_172\n",
      "scaf_173\n",
      "scaf_174\n",
      "scaf_175\n",
      "scaf_176\n",
      "scaf_177\n",
      "scaf_178\n",
      "scaf_179\n",
      "scaf_180\n",
      "scaf_181\n",
      "scaf_182\n",
      "scaf_183\n",
      "scaf_184\n",
      "scaf_185\n",
      "scaf_186\n",
      "scaf_187\n",
      "scaf_188\n",
      "scaf_189\n",
      "scaf_190\n",
      "scaf_191\n",
      "scaf_192\n",
      "scaf_193\n",
      "scaf_194\n",
      "scaf_195\n",
      "scaf_196\n",
      "scaf_197\n",
      "scaf_198\n",
      "scaf_199\n",
      "scaf_200\n",
      "scaf_201\n",
      "scaf_202\n",
      "scaf_203\n",
      "scaf_204\n",
      "scaf_205\n",
      "scaf_206\n",
      "scaf_207\n",
      "scaf_208\n",
      "scaf_209\n",
      "scaf_210\n",
      "scaf_211\n",
      "scaf_212\n",
      "scaf_213\n",
      "scaf_214\n",
      "scaf_215\n",
      "scaf_216\n",
      "scaf_217\n",
      "scaf_218\n",
      "scaf_219\n",
      "scaf_220\n",
      "scaf_221\n",
      "scaf_222\n",
      "scaf_223\n",
      "scaf_224\n",
      "scaf_225\n",
      "scaf_226\n",
      "scaf_227\n",
      "scaf_228\n",
      "scaf_229\n",
      "scaf_230\n",
      "scaf_231\n",
      "scaf_232\n",
      "scaf_233\n",
      "scaf_234\n",
      "scaf_235\n",
      "scaf_236\n",
      "scaf_237\n",
      "scaf_238\n",
      "scaf_239\n",
      "scaf_240\n",
      "scaf_241\n",
      "scaf_242\n",
      "scaf_243\n",
      "scaf_244\n",
      "scaf_245\n",
      "scaf_246\n",
      "scaf_247\n",
      "scaf_248\n",
      "scaf_249\n",
      "scaf_250\n",
      "scaf_251\n",
      "scaf_252\n",
      "scaf_253\n",
      "scaf_254\n",
      "scaf_255\n",
      "scaf_256\n",
      "scaf_257\n",
      "scaf_258\n",
      "scaf_259\n",
      "scaf_260\n",
      "scaf_261\n",
      "scaf_262\n",
      "scaf_263\n",
      "scaf_264\n",
      "scaf_265\n",
      "scaf_266\n",
      "scaf_267\n",
      "scaf_268\n",
      "scaf_269\n",
      "scaf_270\n",
      "scaf_271\n",
      "scaf_272\n",
      "scaf_273\n",
      "scaf_274\n",
      "scaf_275\n",
      "scaf_276\n",
      "scaf_277\n",
      "scaf_278\n",
      "scaf_279\n",
      "scaf_280\n",
      "scaf_281\n",
      "scaf_282\n",
      "scaf_283\n",
      "scaf_284\n",
      "scaf_285\n",
      "scaf_286\n",
      "scaf_287\n",
      "scaf_288\n",
      "scaf_289\n",
      "scaf_290\n",
      "scaf_291\n",
      "scaf_292\n",
      "scaf_293\n",
      "scaf_294\n",
      "scaf_295\n",
      "scaf_296\n",
      "scaf_297\n",
      "scaf_298\n",
      "scaf_299\n",
      "scaf_300\n",
      "scaf_301\n",
      "scaf_302\n",
      "scaf_303\n",
      "scaf_304\n",
      "scaf_305\n",
      "scaf_306\n",
      "scaf_307\n",
      "scaf_308\n",
      "scaf_309\n",
      "scaf_310\n",
      "scaf_311\n",
      "scaf_312\n",
      "scaf_313\n",
      "scaf_314\n",
      "scaf_315\n",
      "scaf_316\n",
      "scaf_317\n",
      "scaf_318\n",
      "scaf_319\n",
      "scaf_320\n",
      "scaf_321\n",
      "scaf_322\n",
      "scaf_323\n",
      "scaf_324\n",
      "scaf_325\n",
      "scaf_326\n",
      "scaf_327\n",
      "scaf_328\n",
      "scaf_329\n",
      "scaf_330\n",
      "scaf_331\n",
      "scaf_332\n",
      "scaf_333\n",
      "scaf_334\n",
      "scaf_335\n",
      "scaf_336\n",
      "scaf_337\n",
      "scaf_338\n",
      "scaf_339\n",
      "scaf_340\n",
      "scaf_341\n",
      "scaf_342\n",
      "scaf_343\n",
      "scaf_344\n",
      "scaf_345\n",
      "scaf_346\n",
      "scaf_347\n",
      "scaf_348\n",
      "scaf_349\n",
      "scaf_350\n",
      "scaf_351\n",
      "scaf_352\n",
      "scaf_353\n",
      "scaf_354\n",
      "scaf_355\n",
      "scaf_356\n",
      "scaf_357\n",
      "scaf_358\n",
      "scaf_359\n",
      "scaf_360\n",
      "scaf_361\n",
      "scaf_362\n",
      "scaf_363\n",
      "scaf_364\n",
      "scaf_365\n",
      "scaf_366\n",
      "scaf_367\n",
      "scaf_368\n",
      "scaf_369\n",
      "scaf_370\n",
      "scaf_371\n",
      "scaf_372\n",
      "scaf_373\n",
      "scaf_374\n",
      "scaf_375\n",
      "scaf_376\n",
      "scaf_377\n",
      "scaf_378\n",
      "scaf_379\n",
      "scaf_380\n",
      "scaf_381\n",
      "scaf_382\n",
      "scaf_383\n",
      "scaf_384\n",
      "scaf_385\n",
      "scaf_386\n",
      "scaf_387\n",
      "scaf_388\n",
      "scaf_389\n",
      "scaf_390\n",
      "scaf_391\n",
      "scaf_392\n",
      "scaf_393\n",
      "scaf_394\n",
      "scaf_395\n",
      "scaf_396\n",
      "scaf_397\n",
      "scaf_398\n",
      "scaf_399\n",
      "scaf_400\n",
      "scaf_401\n",
      "scaf_402\n",
      "scaf_403\n",
      "scaf_404\n",
      "scaf_405\n",
      "scaf_406\n",
      "scaf_407\n",
      "scaf_408\n",
      "scaf_409\n",
      "scaf_410\n",
      "scaf_411\n",
      "scaf_412\n",
      "scaf_413\n",
      "scaf_414\n",
      "scaf_415\n",
      "scaf_416\n",
      "scaf_417\n",
      "scaf_418\n",
      "scaf_419\n",
      "scaf_420\n",
      "scaf_421\n",
      "scaf_422\n",
      "scaf_423\n",
      "scaf_424\n",
      "scaf_425\n",
      "scaf_426\n",
      "scaf_427\n",
      "scaf_428\n",
      "scaf_429\n",
      "scaf_430\n",
      "scaf_431\n",
      "scaf_432\n",
      "scaf_433\n",
      "scaf_434\n",
      "scaf_435\n",
      "scaf_436\n",
      "scaf_437\n",
      "scaf_438\n",
      "scaf_439\n",
      "scaf_440\n",
      "scaf_441\n",
      "scaf_442\n",
      "scaf_443\n",
      "scaf_444\n",
      "scaf_445\n",
      "scaf_446\n",
      "scaf_447\n",
      "scaf_448\n",
      "scaf_449\n",
      "scaf_450\n",
      "scaf_451\n",
      "scaf_452\n",
      "scaf_453\n",
      "scaf_454\n",
      "scaf_455\n",
      "scaf_456\n",
      "scaf_457\n",
      "scaf_458\n",
      "scaf_459\n",
      "scaf_460\n",
      "scaf_461\n",
      "scaf_462\n",
      "scaf_463\n",
      "scaf_464\n",
      "scaf_465\n",
      "scaf_466\n",
      "scaf_467\n",
      "scaf_468\n",
      "scaf_469\n",
      "scaf_470\n",
      "scaf_471\n",
      "scaf_472\n",
      "scaf_473\n",
      "scaf_474\n",
      "scaf_475\n",
      "scaf_476\n",
      "scaf_477\n",
      "scaf_478\n",
      "scaf_479\n",
      "scaf_480\n",
      "scaf_481\n",
      "scaf_482\n",
      "scaf_483\n",
      "scaf_484\n",
      "scaf_485\n",
      "scaf_486\n",
      "scaf_487\n",
      "scaf_488\n",
      "scaf_489\n",
      "scaf_490\n",
      "scaf_491\n",
      "scaf_492\n",
      "scaf_493\n",
      "scaf_494\n",
      "scaf_495\n",
      "scaf_496\n",
      "scaf_497\n",
      "scaf_498\n",
      "scaf_499\n",
      "scaf_500\n",
      "scaf_501\n",
      "scaf_502\n",
      "scaf_503\n",
      "scaf_504\n",
      "scaf_505\n",
      "scaf_506\n",
      "scaf_507\n",
      "scaf_508\n",
      "scaf_509\n",
      "scaf_510\n",
      "scaf_511\n",
      "scaf_512\n",
      "scaf_513\n",
      "scaf_514\n",
      "scaf_515\n",
      "scaf_516\n",
      "scaf_517\n",
      "scaf_518\n",
      "scaf_519\n",
      "scaf_520\n",
      "scaf_521\n",
      "scaf_522\n",
      "scaf_523\n",
      "scaf_524\n",
      "scaf_525\n",
      "scaf_526\n",
      "scaf_527\n",
      "scaf_528\n",
      "scaf_529\n",
      "scaf_530\n",
      "scaf_531\n",
      "scaf_532\n",
      "scaf_533\n",
      "scaf_534\n",
      "scaf_535\n",
      "scaf_536\n",
      "scaf_537\n",
      "scaf_538\n",
      "scaf_539\n",
      "scaf_540\n",
      "scaf_541\n",
      "scaf_542\n",
      "scaf_543\n",
      "scaf_544\n",
      "scaf_545\n",
      "scaf_546\n",
      "scaf_547\n",
      "scaf_548\n",
      "scaf_549\n",
      "scaf_550\n",
      "scaf_551\n",
      "scaf_552\n",
      "scaf_553\n",
      "scaf_554\n",
      "scaf_555\n",
      "scaf_556\n",
      "scaf_557\n",
      "scaf_558\n",
      "scaf_559\n",
      "scaf_560\n",
      "scaf_561\n",
      "scaf_562\n",
      "scaf_563\n",
      "scaf_564\n",
      "scaf_565\n",
      "scaf_566\n",
      "scaf_567\n",
      "scaf_568\n",
      "scaf_569\n",
      "scaf_570\n",
      "scaf_571\n",
      "scaf_572\n",
      "scaf_573\n",
      "scaf_574\n",
      "scaf_575\n",
      "scaf_576\n",
      "scaf_577\n",
      "scaf_578\n",
      "scaf_579\n",
      "scaf_580\n",
      "scaf_581\n",
      "scaf_582\n",
      "scaf_583\n",
      "scaf_584\n",
      "scaf_585\n",
      "scaf_586\n",
      "scaf_587\n",
      "scaf_588\n",
      "scaf_589\n",
      "scaf_590\n",
      "scaf_591\n",
      "scaf_592\n",
      "scaf_593\n",
      "scaf_594\n",
      "scaf_595\n",
      "scaf_596\n",
      "scaf_597\n",
      "scaf_598\n",
      "scaf_599\n",
      "scaf_600\n",
      "scaf_601\n",
      "scaf_602\n",
      "scaf_603\n",
      "scaf_604\n",
      "scaf_605\n",
      "scaf_606\n",
      "scaf_607\n",
      "scaf_608\n",
      "scaf_609\n",
      "scaf_610\n",
      "scaf_611\n",
      "scaf_612\n",
      "scaf_613\n",
      "scaf_614\n",
      "scaf_615\n",
      "scaf_616\n",
      "scaf_617\n",
      "scaf_618\n",
      "scaf_619\n",
      "scaf_620\n",
      "scaf_621\n",
      "scaf_622\n",
      "scaf_623\n",
      "scaf_624\n",
      "scaf_625\n",
      "scaf_626\n",
      "scaf_627\n",
      "scaf_628\n",
      "scaf_629\n",
      "scaf_630\n",
      "scaf_631\n",
      "scaf_632\n",
      "scaf_633\n",
      "scaf_634\n",
      "scaf_635\n",
      "scaf_636\n",
      "scaf_637\n",
      "scaf_638\n",
      "scaf_639\n",
      "scaf_640\n",
      "scaf_641\n",
      "scaf_642\n",
      "scaf_643\n",
      "scaf_644\n",
      "scaf_645\n",
      "scaf_646\n",
      "scaf_647\n",
      "scaf_648\n",
      "scaf_649\n",
      "scaf_650\n",
      "scaf_651\n",
      "scaf_652\n",
      "scaf_653\n",
      "scaf_654\n",
      "scaf_655\n",
      "scaf_656\n",
      "scaf_657\n",
      "scaf_658\n",
      "scaf_659\n",
      "scaf_660\n",
      "scaf_661\n",
      "scaf_662\n",
      "scaf_663\n",
      "scaf_664\n",
      "scaf_665\n",
      "scaf_666\n",
      "scaf_667\n",
      "scaf_668\n",
      "scaf_669\n",
      "scaf_670\n",
      "scaf_671\n",
      "scaf_672\n",
      "scaf_673\n",
      "scaf_674\n",
      "scaf_675\n",
      "scaf_676\n",
      "scaf_677\n",
      "scaf_678\n",
      "scaf_679\n",
      "scaf_680\n",
      "scaf_681\n",
      "scaf_682\n",
      "scaf_683\n",
      "scaf_684\n",
      "scaf_685\n",
      "scaf_686\n",
      "scaf_687\n",
      "scaf_688\n",
      "scaf_689\n",
      "scaf_690\n",
      "scaf_691\n",
      "scaf_692\n",
      "scaf_693\n",
      "scaf_694\n",
      "scaf_695\n",
      ">chr1\n",
      "308452471\n",
      ">chr2\n",
      "243675191\n",
      ">chr3\n",
      "238017767\n",
      ">chr4\n",
      "250330460\n",
      ">chr5\n",
      "226353449\n",
      ">chr6\n",
      "181357234\n",
      ">chr7\n",
      "185808916\n",
      ">chr8\n",
      "182411202\n",
      ">chr9\n",
      "163004744\n",
      ">chr10\n",
      "152435371\n",
      ">scaf_21\n",
      "712123\n",
      ">scaf_22\n",
      "570058\n",
      ">scaf_23\n",
      "435198\n",
      ">scaf_24\n",
      "387292\n",
      ">scaf_25\n",
      "264983\n",
      ">scaf_26\n",
      "254485\n",
      ">scaf_27\n",
      "247825\n",
      ">scaf_28\n",
      "239868\n",
      ">scaf_29\n",
      "217815\n",
      ">scaf_30\n",
      "207305\n",
      ">scaf_31\n",
      "203238\n",
      ">scaf_32\n",
      "197712\n",
      ">scaf_33\n",
      "196160\n",
      ">scaf_34\n",
      "190448\n",
      ">scaf_35\n",
      "190269\n",
      ">scaf_36\n",
      "187880\n",
      ">scaf_37\n",
      "173911\n",
      ">scaf_38\n",
      "173111\n",
      ">scaf_39\n",
      "172167\n",
      ">scaf_40\n",
      "171701\n",
      ">scaf_41\n",
      "171519\n",
      ">scaf_42\n",
      "169347\n",
      ">scaf_43\n",
      "168052\n",
      ">scaf_44\n",
      "164231\n",
      ">scaf_45\n",
      "164060\n",
      ">scaf_46\n",
      "160926\n",
      ">scaf_47\n",
      "160502\n",
      ">scaf_48\n",
      "159682\n",
      ">scaf_49\n",
      "153763\n",
      ">scaf_50\n",
      "153744\n",
      ">scaf_51\n",
      "152095\n",
      ">scaf_52\n",
      "151820\n",
      ">scaf_53\n",
      "148690\n",
      ">scaf_54\n",
      "144791\n",
      ">scaf_55\n",
      "144328\n",
      ">scaf_56\n",
      "142977\n",
      ">scaf_57\n",
      "142290\n",
      ">scaf_58\n",
      "141577\n",
      ">scaf_59\n",
      "138583\n",
      ">scaf_60\n",
      "137147\n",
      ">scaf_61\n",
      "134092\n",
      ">scaf_62\n",
      "133633\n",
      ">scaf_63\n",
      "132531\n",
      ">scaf_64\n",
      "130230\n",
      ">scaf_65\n",
      "129299\n",
      ">scaf_66\n",
      "128904\n",
      ">scaf_67\n",
      "128482\n",
      ">scaf_68\n",
      "127882\n",
      ">scaf_69\n",
      "127290\n",
      ">scaf_70\n",
      "126443\n",
      ">scaf_71\n",
      "125047\n",
      ">scaf_72\n",
      "124860\n",
      ">scaf_73\n",
      "123043\n",
      ">scaf_74\n",
      "120982\n",
      ">scaf_75\n",
      "120904\n",
      ">scaf_76\n",
      "120818\n",
      ">scaf_77\n",
      "120549\n",
      ">scaf_78\n",
      "120088\n",
      ">scaf_79\n",
      "119143\n",
      ">scaf_80\n",
      "118700\n",
      ">scaf_81\n",
      "118576\n",
      ">scaf_82\n",
      "118500\n",
      ">scaf_83\n",
      "117266\n",
      ">scaf_84\n",
      "117233\n",
      ">scaf_85\n",
      "115765\n",
      ">scaf_86\n",
      "115550\n",
      ">scaf_87\n",
      "114887\n",
      ">scaf_88\n",
      "114023\n",
      ">scaf_89\n",
      "113862\n",
      ">scaf_90\n",
      "113779\n",
      ">scaf_91\n",
      "113002\n",
      ">scaf_92\n",
      "112596\n",
      ">scaf_93\n",
      "112550\n",
      ">scaf_94\n",
      "111637\n",
      ">scaf_95\n",
      "110562\n",
      ">scaf_96\n",
      "110121\n",
      ">scaf_97\n",
      "109951\n",
      ">scaf_98\n",
      "109947\n",
      ">scaf_99\n",
      "109352\n",
      ">scaf_100\n",
      "109197\n",
      ">scaf_101\n",
      "109031\n",
      ">scaf_102\n",
      "108697\n",
      ">scaf_103\n",
      "108378\n",
      ">scaf_104\n",
      "108339\n",
      ">scaf_105\n",
      "107956\n",
      ">scaf_106\n",
      "107526\n",
      ">scaf_107\n",
      "107524\n",
      ">scaf_108\n",
      "106690\n",
      ">scaf_109\n",
      "106305\n",
      ">scaf_110\n",
      "106098\n",
      ">scaf_111\n",
      "105289\n",
      ">scaf_112\n",
      "105250\n",
      ">scaf_113\n",
      "105226\n",
      ">scaf_114\n",
      "105163\n",
      ">scaf_115\n",
      "104537\n",
      ">scaf_116\n",
      "104499\n",
      ">scaf_117\n",
      "103808\n",
      ">scaf_118\n",
      "103806\n",
      ">scaf_119\n",
      "103210\n",
      ">scaf_120\n",
      "102624\n",
      ">scaf_121\n",
      "102406\n",
      ">scaf_122\n",
      "102397\n",
      ">scaf_123\n",
      "102349\n",
      ">scaf_124\n",
      "101678\n",
      ">scaf_125\n",
      "101641\n",
      ">scaf_126\n",
      "100851\n",
      ">scaf_127\n",
      "100600\n",
      ">scaf_128\n",
      "100537\n",
      ">scaf_129\n",
      "100364\n",
      ">scaf_130\n",
      "99893\n",
      ">scaf_131\n",
      "99674\n",
      ">scaf_132\n",
      "98130\n",
      ">scaf_133\n",
      "97996\n",
      ">scaf_134\n",
      "97918\n",
      ">scaf_135\n",
      "96911\n",
      ">scaf_136\n",
      "96690\n",
      ">scaf_137\n",
      "96496\n",
      ">scaf_138\n",
      "96036\n",
      ">scaf_139\n",
      "95934\n",
      ">scaf_140\n",
      "95894\n",
      ">scaf_141\n",
      "95890\n",
      ">scaf_142\n",
      "95521\n",
      ">scaf_143\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "95419\n",
      ">scaf_144\n",
      "95245\n",
      ">scaf_145\n",
      "95129\n",
      ">scaf_146\n",
      "94989\n",
      ">scaf_147\n",
      "94793\n",
      ">scaf_148\n",
      "94647\n",
      ">scaf_149\n",
      "94334\n",
      ">scaf_150\n",
      "94333\n",
      ">scaf_151\n",
      "94036\n",
      ">scaf_152\n",
      "93636\n",
      ">scaf_153\n",
      "93514\n",
      ">scaf_154\n",
      "93405\n",
      ">scaf_155\n",
      "93298\n",
      ">scaf_156\n",
      "93084\n",
      ">scaf_157\n",
      "93056\n",
      ">scaf_158\n",
      "92553\n",
      ">scaf_159\n",
      "91995\n",
      ">scaf_160\n",
      "91958\n",
      ">scaf_161\n",
      "91897\n",
      ">scaf_162\n",
      "91766\n",
      ">scaf_163\n",
      "91439\n",
      ">scaf_164\n",
      "91076\n",
      ">scaf_165\n",
      "91072\n",
      ">scaf_166\n",
      "91034\n",
      ">scaf_167\n",
      "90901\n",
      ">scaf_168\n",
      "90815\n",
      ">scaf_169\n",
      "90613\n",
      ">scaf_170\n",
      "90547\n",
      ">scaf_171\n",
      "89502\n",
      ">scaf_172\n",
      "89377\n",
      ">scaf_173\n",
      "89242\n",
      ">scaf_174\n",
      "89235\n",
      ">scaf_175\n",
      "89155\n",
      ">scaf_176\n",
      "88783\n",
      ">scaf_177\n",
      "88032\n",
      ">scaf_178\n",
      "87897\n",
      ">scaf_179\n",
      "87788\n",
      ">scaf_180\n",
      "87559\n",
      ">scaf_181\n",
      "87401\n",
      ">scaf_182\n",
      "87195\n",
      ">scaf_183\n",
      "86681\n",
      ">scaf_184\n",
      "86649\n",
      ">scaf_185\n",
      "86437\n",
      ">scaf_186\n",
      "86423\n",
      ">scaf_187\n",
      "86029\n",
      ">scaf_188\n",
      "85958\n",
      ">scaf_189\n",
      "85939\n",
      ">scaf_190\n",
      "85707\n",
      ">scaf_191\n",
      "85696\n",
      ">scaf_192\n",
      "85577\n",
      ">scaf_193\n",
      "85542\n",
      ">scaf_194\n",
      "85512\n",
      ">scaf_195\n",
      "84529\n",
      ">scaf_196\n",
      "84483\n",
      ">scaf_197\n",
      "84480\n",
      ">scaf_198\n",
      "84315\n",
      ">scaf_199\n",
      "84283\n",
      ">scaf_200\n",
      "84277\n",
      ">scaf_201\n",
      "83765\n",
      ">scaf_202\n",
      "83195\n",
      ">scaf_203\n",
      "83150\n",
      ">scaf_204\n",
      "83142\n",
      ">scaf_205\n",
      "83098\n",
      ">scaf_206\n",
      "82836\n",
      ">scaf_207\n",
      "82698\n",
      ">scaf_208\n",
      "82580\n",
      ">scaf_209\n",
      "82439\n",
      ">scaf_210\n",
      "82354\n",
      ">scaf_211\n",
      "82347\n",
      ">scaf_212\n",
      "82331\n",
      ">scaf_213\n",
      "81960\n",
      ">scaf_214\n",
      "81777\n",
      ">scaf_215\n",
      "81747\n",
      ">scaf_216\n",
      "81457\n",
      ">scaf_217\n",
      "81340\n",
      ">scaf_218\n",
      "81233\n",
      ">scaf_219\n",
      "80936\n",
      ">scaf_220\n",
      "80837\n",
      ">scaf_221\n",
      "80814\n",
      ">scaf_222\n",
      "80283\n",
      ">scaf_223\n",
      "79710\n",
      ">scaf_224\n",
      "79539\n",
      ">scaf_225\n",
      "79526\n",
      ">scaf_226\n",
      "79449\n",
      ">scaf_227\n",
      "79237\n",
      ">scaf_228\n",
      "78875\n",
      ">scaf_229\n",
      "78242\n",
      ">scaf_230\n",
      "78216\n",
      ">scaf_231\n",
      "78109\n",
      ">scaf_232\n",
      "77884\n",
      ">scaf_233\n",
      "77858\n",
      ">scaf_234\n",
      "77850\n",
      ">scaf_235\n",
      "77518\n",
      ">scaf_236\n",
      "77375\n",
      ">scaf_237\n",
      "77233\n",
      ">scaf_238\n",
      "77062\n",
      ">scaf_239\n",
      "77005\n",
      ">scaf_240\n",
      "76990\n",
      ">scaf_241\n",
      "76896\n",
      ">scaf_242\n",
      "76850\n",
      ">scaf_243\n",
      "76719\n",
      ">scaf_244\n",
      "76711\n",
      ">scaf_245\n",
      "76434\n",
      ">scaf_246\n",
      "76402\n",
      ">scaf_247\n",
      "76349\n",
      ">scaf_248\n",
      "76347\n",
      ">scaf_249\n",
      "76253\n",
      ">scaf_250\n",
      "76174\n",
      ">scaf_251\n",
      "76113\n",
      ">scaf_252\n",
      "76030\n",
      ">scaf_253\n",
      "75901\n",
      ">scaf_254\n",
      "75487\n",
      ">scaf_255\n",
      "75327\n",
      ">scaf_256\n",
      "75278\n",
      ">scaf_257\n",
      "75264\n",
      ">scaf_258\n",
      "75157\n",
      ">scaf_259\n",
      "74994\n",
      ">scaf_260\n",
      "74954\n",
      ">scaf_261\n",
      "74875\n",
      ">scaf_262\n",
      "74862\n",
      ">scaf_263\n",
      "74687\n",
      ">scaf_264\n",
      "74566\n",
      ">scaf_265\n",
      "74453\n",
      ">scaf_266\n",
      "73951\n",
      ">scaf_267\n",
      "73767\n",
      ">scaf_268\n",
      "73692\n",
      ">scaf_269\n",
      "73531\n",
      ">scaf_270\n",
      "73359\n",
      ">scaf_271\n",
      "73251\n",
      ">scaf_272\n",
      "73066\n",
      ">scaf_273\n",
      "73001\n",
      ">scaf_274\n",
      "72796\n",
      ">scaf_275\n",
      "72665\n",
      ">scaf_276\n",
      "72399\n",
      ">scaf_277\n",
      "72053\n",
      ">scaf_278\n",
      "71983\n",
      ">scaf_279\n",
      "71870\n",
      ">scaf_280\n",
      "71434\n",
      ">scaf_281\n",
      "71227\n",
      ">scaf_282\n",
      "71196\n",
      ">scaf_283\n",
      "71104\n",
      ">scaf_284\n",
      "70953\n",
      ">scaf_285\n",
      "70845\n",
      ">scaf_286\n",
      "70600\n",
      ">scaf_287\n",
      "70547\n",
      ">scaf_288\n",
      "70410\n",
      ">scaf_289\n",
      "70226\n",
      ">scaf_290\n",
      "70070\n",
      ">scaf_291\n",
      "69844\n",
      ">scaf_292\n",
      "69796\n",
      ">scaf_293\n",
      "69747\n",
      ">scaf_294\n",
      "69542\n",
      ">scaf_295\n",
      "69529\n",
      ">scaf_296\n",
      "69442\n",
      ">scaf_297\n",
      "69369\n",
      ">scaf_298\n",
      "69364\n",
      ">scaf_299\n",
      "69243\n",
      ">scaf_300\n",
      "69101\n",
      ">scaf_301\n",
      "69023\n",
      ">scaf_302\n",
      "68967\n",
      ">scaf_303\n",
      "68912\n",
      ">scaf_304\n",
      "68810\n",
      ">scaf_305\n",
      "68803\n",
      ">scaf_306\n",
      "68764\n",
      ">scaf_307\n",
      "68750\n",
      ">scaf_308\n",
      "68697\n",
      ">scaf_309\n",
      "68464\n",
      ">scaf_310\n",
      "68461\n",
      ">scaf_311\n",
      "68394\n",
      ">scaf_312\n",
      "68386\n",
      ">scaf_313\n",
      "68340\n",
      ">scaf_314\n",
      "68200\n",
      ">scaf_315\n",
      "68185\n",
      ">scaf_316\n",
      "68157\n",
      ">scaf_317\n",
      "68122\n",
      ">scaf_318\n",
      "67773\n",
      ">scaf_319\n",
      "67715\n",
      ">scaf_320\n",
      "67453\n",
      ">scaf_321\n",
      "67423\n",
      ">scaf_322\n",
      "67253\n",
      ">scaf_323\n",
      "67217\n",
      ">scaf_324\n",
      "67121\n",
      ">scaf_325\n",
      "67094\n",
      ">scaf_326\n",
      "66933\n",
      ">scaf_327\n",
      "66933\n",
      ">scaf_328\n",
      "66728\n",
      ">scaf_329\n",
      "66653\n",
      ">scaf_330\n",
      "66505\n",
      ">scaf_331\n",
      "66412\n",
      ">scaf_332\n",
      "66353\n",
      ">scaf_333\n",
      "66282\n",
      ">scaf_334\n",
      "66272\n",
      ">scaf_335\n",
      "66081\n",
      ">scaf_336\n",
      "66016\n",
      ">scaf_337\n",
      "65790\n",
      ">scaf_338\n",
      "65540\n",
      ">scaf_339\n",
      "65537\n",
      ">scaf_340\n",
      "65427\n",
      ">scaf_341\n",
      "65327\n",
      ">scaf_342\n",
      "65280\n",
      ">scaf_343\n",
      "65236\n",
      ">scaf_344\n",
      "65094\n",
      ">scaf_345\n",
      "64929\n",
      ">scaf_346\n",
      "64906\n",
      ">scaf_347\n",
      "64744\n",
      ">scaf_348\n",
      "64714\n",
      ">scaf_349\n",
      "64685\n",
      ">scaf_350\n",
      "64593\n",
      ">scaf_351\n",
      "64495\n",
      ">scaf_352\n",
      "64265\n",
      ">scaf_353\n",
      "64155\n",
      ">scaf_354\n",
      "64072\n",
      ">scaf_355\n",
      "64028\n",
      ">scaf_356\n",
      "63985\n",
      ">scaf_357\n",
      "63908\n",
      ">scaf_358\n",
      "63880\n",
      ">scaf_359\n",
      "63432\n",
      ">scaf_360\n",
      "63266\n",
      ">scaf_361\n",
      "63177\n",
      ">scaf_362\n",
      "63148\n",
      ">scaf_363\n",
      "63113\n",
      ">scaf_364\n",
      "63084\n",
      ">scaf_365\n",
      "63080\n",
      ">scaf_366\n",
      "62556\n",
      ">scaf_367\n",
      "62541\n",
      ">scaf_368\n",
      "62533\n",
      ">scaf_369\n",
      "62516\n",
      ">scaf_370\n",
      "62509\n",
      ">scaf_371\n",
      "62284\n",
      ">scaf_372\n",
      "62204\n",
      ">scaf_373\n",
      "62066\n",
      ">scaf_374\n",
      "62047\n",
      ">scaf_375\n",
      "61971\n",
      ">scaf_376\n",
      "61863\n",
      ">scaf_377\n",
      "61815\n",
      ">scaf_378\n",
      "61663\n",
      ">scaf_379\n",
      "61251\n",
      ">scaf_380\n",
      "61109\n",
      ">scaf_381\n",
      "61002\n",
      ">scaf_382\n",
      "60996\n",
      ">scaf_383\n",
      "60990\n",
      ">scaf_384\n",
      "60988\n",
      ">scaf_385\n",
      "60794\n",
      ">scaf_386\n",
      "60618\n",
      ">scaf_387\n",
      "60414\n",
      ">scaf_388\n",
      "60261\n",
      ">scaf_389\n",
      "60203\n",
      ">scaf_390\n",
      "60132\n",
      ">scaf_391\n",
      "60044\n",
      ">scaf_392\n",
      "59970\n",
      ">scaf_393\n",
      "59921\n",
      ">scaf_394\n",
      "59838\n",
      ">scaf_395\n",
      "59635\n",
      ">scaf_396\n",
      "59584\n",
      ">scaf_397\n",
      "59572\n",
      ">scaf_398\n",
      "59286\n",
      ">scaf_399\n",
      "59208\n",
      ">scaf_400\n",
      "59116\n",
      ">scaf_401\n",
      "59113\n",
      ">scaf_402\n",
      "58978\n",
      ">scaf_403\n",
      "58950\n",
      ">scaf_404\n",
      "58762\n",
      ">scaf_405\n",
      "58605\n",
      ">scaf_406\n",
      "58599\n",
      ">scaf_407\n",
      "58456\n",
      ">scaf_408\n",
      "58405\n",
      ">scaf_409\n",
      "58344\n",
      ">scaf_410\n",
      "57998\n",
      ">scaf_411\n",
      "57932\n",
      ">scaf_412\n",
      "57849\n",
      ">scaf_413\n",
      "57812\n",
      ">scaf_414\n",
      "57501\n",
      ">scaf_415\n",
      "57472\n",
      ">scaf_416\n",
      "57371\n",
      ">scaf_417\n",
      "57210\n",
      ">scaf_418\n",
      "57042\n",
      ">scaf_419\n",
      "57011\n",
      ">scaf_420\n",
      "56837\n",
      ">scaf_421\n",
      "56819\n",
      ">scaf_422\n",
      "56707\n",
      ">scaf_423\n",
      "56705\n",
      ">scaf_424\n",
      "56693\n",
      ">scaf_425\n",
      "56600\n",
      ">scaf_426\n",
      "56555\n",
      ">scaf_427\n",
      "56516\n",
      ">scaf_428\n",
      "56435\n",
      ">scaf_429\n",
      "56416\n",
      ">scaf_430\n",
      "56249\n",
      ">scaf_431\n",
      "56149\n",
      ">scaf_432\n",
      "56140\n",
      ">scaf_433\n",
      "55942\n",
      ">scaf_434\n",
      "55813\n",
      ">scaf_435\n",
      "55746\n",
      ">scaf_436\n",
      "55605\n",
      ">scaf_437\n",
      "55516\n",
      ">scaf_438\n",
      "55489\n",
      ">scaf_439\n",
      "55303\n",
      ">scaf_440\n",
      "55289\n",
      ">scaf_441\n",
      "55099\n",
      ">scaf_442\n",
      "55048\n",
      ">scaf_443\n",
      "54923\n",
      ">scaf_444\n",
      "54898\n",
      ">scaf_445\n",
      "54817\n",
      ">scaf_446\n",
      "54746\n",
      ">scaf_447\n",
      "54638\n",
      ">scaf_448\n",
      "54628\n",
      ">scaf_449\n",
      "54545\n",
      ">scaf_450\n",
      "54531\n",
      ">scaf_451\n",
      "54430\n",
      ">scaf_452\n",
      "54309\n",
      ">scaf_453\n",
      "54288\n",
      ">scaf_454\n",
      "54132\n",
      ">scaf_455\n",
      "54093\n",
      ">scaf_456\n",
      "53967\n",
      ">scaf_457\n",
      "53894\n",
      ">scaf_458\n",
      "53843\n",
      ">scaf_459\n",
      "53481\n",
      ">scaf_460\n",
      "53434\n",
      ">scaf_461\n",
      "53417\n",
      ">scaf_462\n",
      "53195\n",
      ">scaf_463\n",
      "53086\n",
      ">scaf_464\n",
      "52993\n",
      ">scaf_465\n",
      "52955\n",
      ">scaf_466\n",
      "52889\n",
      ">scaf_467\n",
      "52720\n",
      ">scaf_468\n",
      "52562\n",
      ">scaf_469\n",
      "52527\n",
      ">scaf_470\n",
      "52467\n",
      ">scaf_471\n",
      "52001\n",
      ">scaf_472\n",
      "51792\n",
      ">scaf_473\n",
      "51791\n",
      ">scaf_474\n",
      "51790\n",
      ">scaf_475\n",
      "51498\n",
      ">scaf_476\n",
      "51442\n",
      ">scaf_477\n",
      "51437\n",
      ">scaf_478\n",
      "51436\n",
      ">scaf_479\n",
      "51400\n",
      ">scaf_480\n",
      "51317\n",
      ">scaf_481\n",
      "51285\n",
      ">scaf_482\n",
      "51226\n",
      ">scaf_483\n",
      "51166\n",
      ">scaf_484\n",
      "51159\n",
      ">scaf_485\n",
      "51097\n",
      ">scaf_486\n",
      "51050\n",
      ">scaf_487\n",
      "51005\n",
      ">scaf_488\n",
      "50467\n",
      ">scaf_489\n",
      "50303\n",
      ">scaf_490\n",
      "50230\n",
      ">scaf_491\n",
      "50119\n",
      ">scaf_492\n",
      "50112\n",
      ">scaf_493\n",
      "50095\n",
      ">scaf_494\n",
      "50090\n",
      ">scaf_495\n",
      "49999\n",
      ">scaf_496\n",
      "49900\n",
      ">scaf_497\n",
      "49811\n",
      ">scaf_498\n",
      "49759\n",
      ">scaf_499\n",
      "49573\n",
      ">scaf_500\n",
      "49572\n",
      ">scaf_501\n",
      "49558\n",
      ">scaf_502\n",
      "49534\n",
      ">scaf_503\n",
      "49321\n",
      ">scaf_504\n",
      "49133\n",
      ">scaf_505\n",
      "49076\n",
      ">scaf_506\n",
      "48960\n",
      ">scaf_507\n",
      "48845\n",
      ">scaf_508\n",
      "48796\n",
      ">scaf_509\n",
      "48793\n",
      ">scaf_510\n",
      "48788\n",
      ">scaf_511\n",
      "48698\n",
      ">scaf_512\n",
      "48411\n",
      ">scaf_513\n",
      "48321\n",
      ">scaf_514\n",
      "48292\n",
      ">scaf_515\n",
      "48059\n",
      ">scaf_516\n",
      "48018\n",
      ">scaf_517\n",
      "47885\n",
      ">scaf_518\n",
      "47771\n",
      ">scaf_519\n",
      "47732\n",
      ">scaf_520\n",
      "47726\n",
      ">scaf_521\n",
      "47702\n",
      ">scaf_522\n",
      "47686\n",
      ">scaf_523\n",
      "47658\n",
      ">scaf_524\n",
      "47519\n",
      ">scaf_525\n",
      "47504\n",
      ">scaf_526\n",
      "47428\n",
      ">scaf_527\n",
      "47425\n",
      ">scaf_528\n",
      "47422\n",
      ">scaf_529\n",
      "47062\n",
      ">scaf_530\n",
      "47056\n",
      ">scaf_531\n",
      "47032\n",
      ">scaf_532\n",
      "46775\n",
      ">scaf_533\n",
      "46694\n",
      ">scaf_534\n",
      "46575\n",
      ">scaf_535\n",
      "46567\n",
      ">scaf_536\n",
      "46533\n",
      ">scaf_537\n",
      "46395\n",
      ">scaf_538\n",
      "46312\n",
      ">scaf_539\n",
      "46276\n",
      ">scaf_540\n",
      "46195\n",
      ">scaf_541\n",
      "46105\n",
      ">scaf_542\n",
      "45928\n",
      ">scaf_543\n",
      "45898\n",
      ">scaf_544\n",
      "45870\n",
      ">scaf_545\n",
      "45852\n",
      ">scaf_546\n",
      "45599\n",
      ">scaf_547\n",
      "45570\n",
      ">scaf_548\n",
      "45567\n",
      ">scaf_549\n",
      "45460\n",
      ">scaf_550\n",
      "45408\n",
      ">scaf_551\n",
      "45391\n",
      ">scaf_552\n",
      "45290\n",
      ">scaf_553\n",
      "45286\n",
      ">scaf_554\n",
      "45262\n",
      ">scaf_555\n",
      "45165\n",
      ">scaf_556\n",
      "44942\n",
      ">scaf_557\n",
      "44777\n",
      ">scaf_558\n",
      "44734\n",
      ">scaf_559\n",
      "44719\n",
      ">scaf_560\n",
      "44617\n",
      ">scaf_561\n",
      "44557\n",
      ">scaf_562\n",
      "44525\n",
      ">scaf_563\n",
      "44435\n",
      ">scaf_564\n",
      "44391\n",
      ">scaf_565\n",
      "44377\n",
      ">scaf_566\n",
      "44336\n",
      ">scaf_567\n",
      "44149\n",
      ">scaf_568\n",
      "43798\n",
      ">scaf_569\n",
      "43689\n",
      ">scaf_570\n",
      "43564\n",
      ">scaf_571\n",
      "43505\n",
      ">scaf_572\n",
      "43361\n",
      ">scaf_573\n",
      "43335\n",
      ">scaf_574\n",
      "43270\n",
      ">scaf_575\n",
      "43186\n",
      ">scaf_576\n",
      "43039\n",
      ">scaf_577\n",
      "42914\n",
      ">scaf_578\n",
      "42890\n",
      ">scaf_579\n",
      "42824\n",
      ">scaf_580\n",
      "42746\n",
      ">scaf_581\n",
      "42734\n",
      ">scaf_582\n",
      "42695\n",
      ">scaf_583\n",
      "42670\n",
      ">scaf_584\n",
      "42655\n",
      ">scaf_585\n",
      "42653\n",
      ">scaf_586\n",
      "42361\n",
      ">scaf_587\n",
      "42140\n",
      ">scaf_588\n",
      "42065\n",
      ">scaf_589\n",
      "42059\n",
      ">scaf_590\n",
      "42054\n",
      ">scaf_591\n",
      "42047\n",
      ">scaf_592\n",
      "42042\n",
      ">scaf_593\n",
      "41905\n",
      ">scaf_594\n",
      "41863\n",
      ">scaf_595\n",
      "41770\n",
      ">scaf_596\n",
      "41630\n",
      ">scaf_597\n",
      "41600\n",
      ">scaf_598\n",
      "41539\n",
      ">scaf_599\n",
      "41383\n",
      ">scaf_600\n",
      "41377\n",
      ">scaf_601\n",
      "41329\n",
      ">scaf_602\n",
      "41219\n",
      ">scaf_603\n",
      "41109\n",
      ">scaf_604\n",
      "41107\n",
      ">scaf_605\n",
      "41057\n",
      ">scaf_606\n",
      "41030\n",
      ">scaf_607\n",
      "40971\n",
      ">scaf_608\n",
      "40905\n",
      ">scaf_609\n",
      "40876\n",
      ">scaf_610\n",
      "40847\n",
      ">scaf_611\n",
      "40814\n",
      ">scaf_612\n",
      "40576\n",
      ">scaf_613\n",
      "40560\n",
      ">scaf_614\n",
      "40479\n",
      ">scaf_615\n",
      "40402\n",
      ">scaf_616\n",
      "40389\n",
      ">scaf_617\n",
      "40165\n",
      ">scaf_618\n",
      "40145\n",
      ">scaf_619\n",
      "40064\n",
      ">scaf_620\n",
      "40006\n",
      ">scaf_621\n",
      "39904\n",
      ">scaf_622\n",
      "39697\n",
      ">scaf_623\n",
      "39553\n",
      ">scaf_624\n",
      "39481\n",
      ">scaf_625\n",
      "39460\n",
      ">scaf_626\n",
      "39077\n",
      ">scaf_627\n",
      "39038\n",
      ">scaf_628\n",
      "38891\n",
      ">scaf_629\n",
      "38764\n",
      ">scaf_630\n",
      "38610\n",
      ">scaf_631\n",
      "38558\n",
      ">scaf_632\n",
      "38463\n",
      ">scaf_633\n",
      "38456\n",
      ">scaf_634\n",
      "38214\n",
      ">scaf_635\n",
      "37995\n",
      ">scaf_636\n",
      "37985\n",
      ">scaf_637\n",
      "37569\n",
      ">scaf_638\n",
      "37477\n",
      ">scaf_639\n",
      "37416\n",
      ">scaf_640\n",
      "37393\n",
      ">scaf_641\n",
      "37334\n",
      ">scaf_642\n",
      "37315\n",
      ">scaf_643\n",
      "37184\n",
      ">scaf_644\n",
      "37110\n",
      ">scaf_645\n",
      "37080\n",
      ">scaf_646\n",
      "36930\n",
      ">scaf_647\n",
      "36782\n",
      ">scaf_648\n",
      "36641\n",
      ">scaf_649\n",
      "36626\n",
      ">scaf_650\n",
      "36526\n",
      ">scaf_651\n",
      "36525\n",
      ">scaf_652\n",
      "36369\n",
      ">scaf_653\n",
      "36282\n",
      ">scaf_654\n",
      "36267\n",
      ">scaf_655\n",
      "36266\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">scaf_656\n",
      "36217\n",
      ">scaf_657\n",
      "35909\n",
      ">scaf_658\n",
      "35909\n",
      ">scaf_659\n",
      "35884\n",
      ">scaf_660\n",
      "35784\n",
      ">scaf_661\n",
      "35720\n",
      ">scaf_662\n",
      "35668\n",
      ">scaf_663\n",
      "35599\n",
      ">scaf_664\n",
      "35539\n",
      ">scaf_665\n",
      "35297\n",
      ">scaf_666\n",
      "35285\n",
      ">scaf_667\n",
      "35263\n",
      ">scaf_668\n",
      "35100\n",
      ">scaf_669\n",
      "34989\n",
      ">scaf_670\n",
      "34940\n",
      ">scaf_671\n",
      "34852\n",
      ">scaf_672\n",
      "34684\n",
      ">scaf_673\n",
      "34626\n",
      ">scaf_674\n",
      "34568\n",
      ">scaf_675\n",
      "34427\n",
      ">scaf_676\n",
      "34391\n",
      ">scaf_677\n",
      "33997\n",
      ">scaf_678\n",
      "33938\n",
      ">scaf_679\n",
      "33821\n",
      ">scaf_680\n",
      "33689\n",
      ">scaf_681\n",
      "33683\n",
      ">scaf_682\n",
      "33634\n",
      ">scaf_683\n",
      "33496\n",
      ">scaf_684\n",
      "33264\n",
      ">scaf_685\n",
      "32618\n",
      ">scaf_686\n",
      "32197\n",
      ">scaf_687\n",
      "32019\n",
      ">scaf_688\n",
      "31944\n",
      ">scaf_689\n",
      "31748\n",
      ">scaf_690\n",
      "31498\n",
      ">scaf_691\n",
      "31489\n",
      ">scaf_692\n",
      "31211\n",
      ">scaf_693\n",
      "30818\n",
      ">scaf_694\n",
      "30512\n",
      ">scaf_695\n",
      "30084\n"
     ]
    }
   ],
   "source": [
    "# Let's look at the genome \n",
    "#This command lists all of the entry names in the fasta file \n",
    "grep -o -E \"^>\\w+\" /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0.fa | tr -d \">\"\n",
    "#This command lists the number of bp in for each entry \n",
    "awk '/^>/{if (l!=\"\") print l; print; l=0; next}{l+=length($0)}END{print l}' /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n"
     ]
    }
   ],
   "source": [
    "# The genome has a lot of non-chromosome scaffolds\n",
    "# So I filtered the genome fasta file to just the chromosomes, as I am interested in genes in the nuclear DNA \n",
    "# First make an id file that lists all of the names of the entries I want\n",
    "echo \"chr1\" > ./chr_names.txt\n",
    "#Check that the file was made\n",
    "cat ./chr_names.txt\n",
    "# Then add the other entries \n",
    "echo \"chr2\" >> ./chr_names.txt\n",
    "echo \"chr3\" >> ./chr_names.txt\n",
    "echo \"chr4\" >> ./chr_names.txt\n",
    "echo \"chr5\" >> ./chr_names.txt\n",
    "echo \"chr6\" >> ./chr_names.txt\n",
    "echo \"chr7\" >> ./chr_names.txt\n",
    "echo \"chr8\" >> ./chr_names.txt\n",
    "echo \"chr9\" >> ./chr_names.txt\n",
    "echo \"chr10\" >> ./chr_names.txt\n",
    "# check the fasta file to see if all of the entries are there\n",
    "cat ./chr_names.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">chr1\n",
      "308452471\n",
      ">chr2\n",
      "243675191\n",
      ">chr3\n",
      "238017767\n",
      ">chr4\n",
      "250330460\n",
      ">chr5\n",
      "226353449\n",
      ">chr6\n",
      "181357234\n",
      ">chr7\n",
      "185808916\n",
      ">chr8\n",
      "182411202\n",
      ">chr9\n",
      "163004744\n",
      ">chr10\n",
      "152435371\n"
     ]
    }
   ],
   "source": [
    "# Use the ID file to extract those entries from the fasta file and make a new fasta file\n",
    "awk -F'>' 'NR==FNR{ids[$0]; next} NF>1{f=($2 in ids)} f' ./chr_names.txt /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0.fa > /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0_chr.fa\n",
    "#Check that the command worked properly\n",
    "#This command lists the number of bp in for each entry (compare to the previous list)\n",
    "awk '/^>/{if (l!=\"\") print l; print; l=0; next}{l+=length($0)}END{print l}' /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0_chr.fa\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# try indexing the fasta file \n",
    " samtools faidx /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0_chr.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t308452471\t6\t80\t81\n",
      "chr2\t243675191\t312308139\t80\t81\n",
      "chr3\t238017767\t559029276\t80\t81\n",
      "chr4\t250330460\t800022272\t80\t81\n",
      "chr5\t226353449\t1053481869\t80\t81\n",
      "chr6\t181357234\t1282664743\t80\t81\n",
      "chr7\t185808916\t1466288949\t80\t81\n",
      "chr8\t182411202\t1654420483\t80\t81\n",
      "chr9\t163004744\t1839111832\t80\t81\n",
      "chr10\t152435371\t2004154143\t80\t81\n"
     ]
    }
   ],
   "source": [
    "head /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0_chr.fa.fai"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n",
      "[warning] samtools mpileup option `g` is functional, but deprecated. Please switch to using bcftools mpileup in future.\n",
      "[mpileup] 1 samples in 1 input files\n"
     ]
    }
   ],
   "source": [
    "##Generate a variant file using bcftools mpileup and call the variants (this is just variants in Trf Mutant compared to B73) \n",
    "#-f means an input file is in fasta format\n",
    "#-Ob means output in a compressed bcf format\n",
    "#-mv is call short indels \n",
    "#-o is output name\n",
    "#-r restricts the pileup to a region\n",
    "samtools mpileup -f /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0_chr.fa -g /Volumes/Elements/Sequence_Data/WGS_data/Trf_mutant_WGS_2022/WGS_Alignment_NAM5_May2022/BAM_reads/TrfM6_aligned_NAM5.bam | bcftools call -mv -Ob -o /Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022/TrfM6_WGS_NAM5_mpileup_Dec2022.vcf\n",
    "\n",
    "#samtools mpileup -f ref.fa -g l100_n1000_d300_31_1.bam | bcftools/bcftools call -m - > l100_n1000_d300_31_1.vc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n",
      "[warning] samtools mpileup option `g` is functional, but deprecated. Please switch to using bcftools mpileup in future.\n",
      "[W::bam_hdr_read] EOF marker is absent. The input is probably truncated\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[E::bgzf_uncompress] Inflate operation failed: progress temporarily not possible, or in() / out() returned an error\n",
      "[E::bgzf_read] Read block operation failed with error 1 after 0 of 4 bytes\n",
      "samtools mpileup: error reading from input file\n"
     ]
    }
   ],
   "source": [
    "# This looks like it is a normal size! (1.13Gb for the vcf file)\n",
    "samtools mpileup -f /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0_chr.fa -g /Volumes/Elements/Sequence_Data/WGS_data/Trf_mutant_WGS_2022/WGS_Alignment_NAM5_May2022/BAM_reads/TrfN5_aligned_NAM5.bam | bcftools call -mv -Ob -o /Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_NAM5_May2022/TrfN5_WGS_NAM5_mpileup_Dec2022.vcf\n",
    "\n",
    "# Problem with WT input file? Need to make the BAM file again? Looks smaller than the mut file. Suggests it is truncated. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/aricha14/Documents/Research_Projects/TRF_Project/WGS/WGS_Alignment_B73v3\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Elements/Sequence_Data/WGS_data/Trf_mutant_WGS_2022\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/Elements/Sequence_Data/WGS_data/Trf_mutant_WGS_2022\n",
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Repeat the process for the N reads \n",
    "# First combine the reads from the two lanes of sequencing to make single files \n",
    "mkdir ./Comb_Raw_Data\n",
    "cat ./Raw_Data/Trf_WGS_raw_data_N/*L1_1.fq.gz ./Raw_Data/Trf_WGS_raw_data_N/*L3_1.fq.gz > ./Comb_Raw_Data/TrfN5_rawlanescomb_1.fastq.gz\n",
    "cat ./Raw_Data/Trf_WGS_raw_data_N/*L1_2.fq.gz ./Raw_Data/Trf_WGS_raw_data_N/*L3_2.fq.gz > ./Comb_Raw_Data/TrfN5_rawlanescomb_2.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: ./fastp_initial_Jan2023: File exists\n",
      "Read1 before filtering:\n",
      "total reads: 180831470\n",
      "total bases: 27124720500\n",
      "Q20 bases: 26540157867(97.8449%)\n",
      "Q30 bases: 25578753185(94.3005%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 180831470\n",
      "total bases: 27124720500\n",
      "Q20 bases: 26405852573(97.3498%)\n",
      "Q30 bases: 25240916674(93.055%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 179542428\n",
      "total bases: 26794214644\n",
      "Q20 bases: 26294233967(98.134%)\n",
      "Q30 bases: 25371869996(94.6916%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 179542428\n",
      "total bases: 26794651505\n",
      "Q20 bases: 26188393118(97.7374%)\n",
      "Q30 bases: 25068058513(93.5562%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 359084856\n",
      "reads failed due to low quality: 2290926\n",
      "reads failed due to too many N: 155252\n",
      "reads failed due to too short: 131906\n",
      "reads with adapter trimmed: 6496106\n",
      "bases trimmed due to adapters: 264102501\n",
      "\n",
      "Duplication rate: 19.7424%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 269\n",
      "\n",
      "JSON report: ./fastp_initial_Jan2023/TrfN5_rawlanescomb.json\n",
      "HTML report: ./fastp_initial_Jan2023/TrfN5_rawlanescomb.html\n",
      "\n",
      "fastp -w 4 -i ./Comb_Raw_Data/TrfN5_rawlanescomb_1.fastq.gz -I ./Comb_Raw_Data/TrfN5_rawlanescomb_2.fastq.gz -o ./fastp_initial_Jan2023/TrfN5_rawlanescomb_R1.fastq.gz -O ./fastp_initial_Jan2023/TrfN5_rawlanescomb_R2.fastq.gz -h ./fastp_initial_Jan2023/TrfN5_rawlanescomb.html -j ./fastp_initial_Jan2023/TrfN5_rawlanescomb.json -R TrfN5_rawlanescomb fastp report \n",
      "fastp v0.22.0, time used: 1059 seconds\n",
      "Error to read gzip file\n",
      "Error to read gzip file\n",
      "ERROR: file './Comb_Raw_Data/TrfN5_rawlanescomb_2.fastq.gz_1.fastq.gz' doesn't exist, quit now\n"
     ]
    },
    {
     "ename": "",
     "evalue": "255",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "# Run fastp analysis and trimming on the combined files using paired end settings \n",
    "mkdir ./fastp_initial_Jan2023\n",
    "find ./Comb_Raw_Data/ -name \"*.fastq.gz\" | while read f; do\n",
    "STEM=$(basename \"${f}\" _1.fastq.gz)\n",
    "fastp -w 4 -i ./Comb_Raw_Data/${STEM}_1.fastq.gz \\\n",
    "    -I ./Comb_Raw_Data/${STEM}_2.fastq.gz \\\n",
    "    -o ./fastp_initial_Jan2023/${STEM}_R1.fastq.gz \\\n",
    "    -O ./fastp_initial_Jan2023/${STEM}_R2.fastq.gz \\\n",
    "    -h ./fastp_initial_Jan2023/${STEM}.html \\\n",
    "    -j ./fastp_initial_Jan2023/${STEM}.json \\\n",
    "    -R \"${STEM} fastp report\"\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "179542428 reads; of these:\n",
      "  179542428 (100.00%) were paired; of these:\n",
      "    12296586 (6.85%) aligned concordantly 0 times\n",
      "    29115709 (16.22%) aligned concordantly exactly 1 time\n",
      "    138130133 (76.93%) aligned concordantly >1 times\n",
      "    ----\n",
      "    12296586 pairs aligned concordantly 0 times; of these:\n",
      "      237290 (1.93%) aligned discordantly 1 time\n",
      "    ----\n",
      "    12059296 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      24118592 mates make up the pairs; of these:\n",
      "        13580141 (56.31%) aligned 0 times\n",
      "        3086021 (12.80%) aligned exactly 1 time\n",
      "        7452430 (30.90%) aligned >1 times\n",
      "96.22% overall alignment rate\n",
      "[main_samview] fail to read the header from \"-\".\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "# Make new alignment of Trf N pool to the NAM5 genome to make sure that it is not truncated \n",
    "mkdir ./aligned_reads\n",
    "mkdir ./BAM_reads\n",
    "bowtie2 -L 20 -p 4 \\\n",
    "    -x /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0 \\\n",
    "    -1 ./fastp_initial_Jan2023/TrfN5_rawlanescomb_R1.fastq.gz \\\n",
    "    -2 ./fastp_initial_Jan2023/TrfN5_rawlanescomb_R2.fastq.gz \\\n",
    "    -S ./aligned_reads/TrfN5_aligned_NAM5.sam | \\\n",
    "    samtools view -bS - >  ./BAM_reads/TrfN5_aligned_NAM5.bam\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bam_sort_core] merging from 176 files and 4 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "# the pipe to bam failed, so now need to convert to bam file \n",
    "#Convert to .bam file\n",
    "samtools sort -o ./BAM_reads/TrfN5_aligned_NAM5.bam -@ 4 ./aligned_reads/TrfN5_aligned_NAM5.sam \n",
    "#index the bam file\n",
    "samtools index ./BAM_reads/TrfN5_aligned_NAM5.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[warning] samtools mpileup option `g` is functional, but deprecated. Please switch to using bcftools mpileup in future.\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n",
      "[mpileup] 1 samples in 1 input files\n"
     ]
    }
   ],
   "source": [
    "# size of bam file is comparable with the mutant (19.25GB) so can now go ahead and run mpileup\n",
    "samtools mpileup -f /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0_chr.fa -g ./BAM_reads/TrfN5_aligned_NAM5.bam | bcftools call -mv -Ob -o ./TrfN5_WGS_NAM5_mpileup_Dec2022.vcf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Let's try to get a variant file between the Mo17 genome and the NAM5.0 genome \n",
    "samtools mpileup -f /Users/aricha14/Documents/Databases/Maize_genome_NAM5/Zm-B73-REFERENCE-NAM-5.0_chr.fa -g ./BAM_reads/TrfN5_aligned_NAM5.bam | bcftools call -mv -Ob -o ./TrfN5_WGS_NAM5_mpileup_Dec2022.vcf\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
